project build/make/
diff --git a/core/Makefile b/core/Makefile
index f98a2c0c6..d07099ca1 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -302,9 +302,9 @@ endif
 # Both of these tags will be removed and replaced with "release-keys"
 # when the target-files is signed in a post-build step.
 ifeq ($(DEFAULT_SYSTEM_DEV_CERTIFICATE),build/make/target/product/security/testkey)
-BUILD_KEYS := test-keys
+BUILD_KEYS := release-keys
 else
-BUILD_KEYS := dev-keys
+BUILD_KEYS := release-keys
 endif
 BUILD_VERSION_TAGS += $(BUILD_KEYS)
 BUILD_VERSION_TAGS := $(subst $(space),$(comma),$(sort $(BUILD_VERSION_TAGS)))

diff --git a/core/binary.mk b/core/binary.mk
index 712f2df56..2fc96b536 100644
--- a/core/binary.mk
+++ b/core/binary.mk
@@ -613,7 +613,7 @@ endif
 # Turn on all warnings and warnings as errors for RS compiles.
 # This can be disabled with LOCAL_RENDERSCRIPT_FLAGS := -Wno-error
-renderscript_flags := -Wall -Werror
+renderscript_flags := -Wall
 renderscript_flags += $(LOCAL_RENDERSCRIPT_FLAGS)
 # -m32 or -m64
 renderscript_flags += -m$(my_32_64_bit_suffix)
@@ -1490,10 +1490,10 @@ ifneq (HEADER_LIBRARIES,$(LOCAL_MODULE_CLASS))
       $(eval MODULES_USING_WNO_ERROR := $(MODULES_USING_WNO_ERROR) $(LOCAL_MODULE_MAKEFILE):$(LOCAL_MODULE))
     else
       # Issue warning if -Werror is not used. Add it.
-      ifeq (,$(filter -Werror,$(my_all_cflags)))
+      ifeq (,$(filter -Wall,$(my_all_cflags)))
         # Add -Wall -Werror unless the project is in the WARNING_ALLOWED project list.
         ifeq (,$(strip $(call find_warning_allowed_projects,$(LOCAL_PATH))))
-          my_cflags := -Wall -Werror $(my_cflags)
+          my_cflags := -Wall $(my_cflags)
         else
           $(eval MODULES_ADDED_WALL := $(MODULES_ADDED_WALL) $(LOCAL_MODULE_MAKEFILE):$(LOCAL_MODULE))
           my_cflags := -Wall $(my_cflags)

diff --git a/core/dex_preopt_config.mk b/core/dex_preopt_config.mk
index b5834b08b..50cbed031 100644
--- a/core/dex_preopt_config.mk
+++ b/core/dex_preopt_config.mk
@@ -1,7 +1,7 @@
 DEX_PREOPT_CONFIG := $(PRODUCT_OUT)/dexpreopt.config

 # The default value for LOCAL_DEX_PREOPT
-DEX_PREOPT_DEFAULT ?= true
+DEX_PREOPT_DEFAULT := true

 # The default filter for which files go into the system_other image (if it is
 # being used). To bundle everything one should set this to '%'
@@ -16,20 +16,20 @@ SYSTEM_OTHER_ODEX_FILTER ?= \
 # The default values for pre-opting. To support the runtime module we ensure no dex files
 # get stripped.
 ifeq ($(PRODUCT_DEX_PREOPT_NEVER_ALLOW_STRIPPING),)
-  PRODUCT_DEX_PREOPT_NEVER_ALLOW_STRIPPING := true
+  PRODUCT_DEX_PREOPT_NEVER_ALLOW_STRIPPING := false
 endif
 # Conditional to building on linux, as dex2oat currently does not work on darwin.
 ifeq ($(HOST_OS),linux)
   ifeq (eng,$(TARGET_BUILD_VARIANT))
     # Don't strip for quick development turnarounds.
-    DEX_PREOPT_DEFAULT := nostripping
+    DEX_PREOPT_DEFAULT := true
     # For an eng build only pre-opt the boot image and system server. This gives reasonable performance
     # and still allows a simple workflow: building in frameworks/base and syncing.
-    WITH_DEXPREOPT_BOOT_IMG_AND_SYSTEM_SERVER_ONLY ?= true
+    WITH_DEXPREOPT_BOOT_IMG_AND_SYSTEM_SERVER_ONLY := false
   endif
   # Add mini-debug-info to the boot classpath unless explicitly asked not to.
   ifneq (false,$(WITH_DEXPREOPT_DEBUG_INFO))
-    PRODUCT_DEX_PREOPT_BOOT_FLAGS += --generate-mini-debug-info
+    PRODUCT_DEX_PREOPT_BOOT_FLAGS +=
   endif

   # Non eng linux builds must have preopt enabled so that system server doesn't run as interpreter
@@ -48,7 +48,7 @@ endif
 ifeq ($(USE_DEX2OAT_DEBUG),false)
 DEX2OAT := $(SOONG_HOST_OUT_EXECUTABLES)/dex2oat$(HOST_EXECUTABLE_SUFFIX)
 else
-DEX2OAT := $(SOONG_HOST_OUT_EXECUTABLES)/dex2oatd$(HOST_EXECUTABLE_SUFFIX)
+DEX2OAT := $(SOONG_HOST_OUT_EXECUTABLES)/dex2oat$(HOST_EXECUTABLE_SUFFIX)
 endif

 DEX2OAT_DEPENDENCY += $(DEX2OAT)
@@ -86,7 +86,6 @@ ifeq ($(WRITE_SOONG_VARIABLES),true)

   $(call json_start)

-  $(call add_json_bool, DefaultNoStripping,                 $(filter nostripping,$(DEX_PREOPT_DEFAULT)))
   $(call add_json_bool, DisablePreopt,                      $(call invert_bool,$(filter true,$(WITH_DEXPREOPT))))
   $(call add_json_list, DisablePreoptModules,               $(DEXPREOPT_DISABLED_MODULES))
   $(call add_json_bool, OnlyPreoptBootImageAndSystemServer, $(filter true,$(WITH_DEXPREOPT_BOOT_IMG_AND_SYSTEM_SERVER_ONLY)))

diff --git a/core/dex_preopt_odex_install.mk b/core/dex_preopt_odex_install.mk
index 85ddbfa36..27865e1ac 100644
--- a/core/dex_preopt_odex_install.mk
+++ b/core/dex_preopt_odex_install.mk
@@ -225,8 +225,6 @@ ifdef LOCAL_DEX_PREOPT
   $(call add_json_bool, ForceCreateAppImage,            $(filter true,$(LOCAL_DEX_PREOPT_APP_IMAGE)))
   $(call add_json_bool, PresignedPrebuilt,              $(filter PRESIGNED,$(LOCAL_CERTIFICATE)))

-  $(call add_json_bool, NoStripping,                    $(filter nostripping,$(LOCAL_DEX_PREOPT)))
-
   $(call json_end)

   my_dexpreopt_config := $(intermediates)/dexpreopt.config

diff --git a/core/java_renderscript.mk b/core/java_renderscript.mk
index 13a6f8e66..72801a92d 100644
--- a/core/java_renderscript.mk
+++ b/core/java_renderscript.mk
@@ -46,7 +46,7 @@ endif

 # Turn on all warnings and warnings as errors for RS compiles.
 # This can be disabled with LOCAL_RENDERSCRIPT_FLAGS := -Wno-error
-renderscript_flags := -Wall -Werror
+renderscript_flags := -Wall
 renderscript_flags += $(LOCAL_RENDERSCRIPT_FLAGS)

 # prepend the RenderScript system include path

diff --git a/core/main.mk b/core/main.mk
index eb4ac8c91..e0625aa56 100644
--- a/core/main.mk
+++ b/core/main.mk
@@ -292,8 +292,6 @@ ifneq (,$(user_variant))
   ADDITIONAL_DEFAULT_PROPERTIES += ro.allow.mock.location=0

 else # !user_variant
-  # Turn on checkjni for non-user builds.
-  ADDITIONAL_BUILD_PROPERTIES += ro.kernel.android.checkjni=1
   # Set device insecure for non-user builds.
   ADDITIONAL_DEFAULT_PROPERTIES += ro.secure=0
   # Allow mock locations by default for non user builds
@@ -303,8 +301,6 @@ endif # !user_variant
 ifeq (true,$(strip $(enable_target_debugging)))
   # Target is more debuggable and adbd is on by default
   ADDITIONAL_DEFAULT_PROPERTIES += ro.debuggable=1
-  # Enable Dalvik lock contention logging.
-  ADDITIONAL_BUILD_PROPERTIES += dalvik.vm.lockprof.threshold=500
 else # !enable_target_debugging
   # Target is less debuggable and adbd is off by default
   ADDITIONAL_DEFAULT_PROPERTIES += ro.debuggable=0
@@ -322,7 +318,7 @@ ifneq ($(filter ro.setupwizard.mode=ENABLED, $(call collapse-pairs, $(ADDITIONAL
 endif
 ifndef is_sdk_build
   # To speedup startup of non-preopted builds, don't verify or compile the boot image.
-  ADDITIONAL_BUILD_PROPERTIES += dalvik.vm.image-dex2oat-filter=verify-at-runtime
+  ADDITIONAL_BUILD_PROPERTIES += dalvik.vm.image-dex2oat-filter=speed
 endif
 endif

diff --git a/core/product.mk b/core/product.mk
index b45d802d0..5ba7730f5 100644
--- a/core/product.mk
+++ b/core/product.mk
@@ -491,10 +491,16 @@ _readonly_late_variables := \

 # Modified internally in the build system
 _readonly_late_variables += \
+  PRODUCT_ART_TARGET_INCLUDE_DEBUG_BUILD \
   PRODUCT_CFI_INCLUDE_PATHS \
   PRODUCT_COPY_FILES \
   PRODUCT_DEX_PREOPT_NEVER_ALLOW_STRIPPING \
   PRODUCT_DEX_PREOPT_BOOT_FLAGS \
+  PRODUCT_DEX_PREOPT_DEFAULT_FLAGS \
+  PRODUCT_DEX_PREOPT_DEFAULT_COMPILER_FILTER \
+  PRODUCT_SYSTEM_SERVER_COMPILER_FILTER \
+  PRODUCT_SYSTEM_SERVER_DEBUG_INFO \
+  PRODUCT_OTHER_JAVA_DEBUG_INFO \
   PRODUCT_SOONG_NAMESPACES

 _readonly_early_variables := $(filter-out $(_readonly_late_variables),$(_product_var_list))

diff --git a/target/product/media_system.mk b/target/product/media_system.mk
index 0489b28d9..d29350b96 100644
--- a/target/product/media_system.mk
+++ b/target/product/media_system.mk
@@ -68,16 +68,6 @@ PRODUCT_COPY_FILES += $(call add-to-product-copy-files-if-exists,\
 PRODUCT_COPY_FILES += $(call add-to-product-copy-files-if-exists,\
     frameworks/base/dirty-image-objects-phone:system/etc/dirty-image-objects)

-# On userdebug builds, collect more tombstones by default.
-ifneq (,$(filter userdebug eng,$(TARGET_BUILD_VARIANT)))
-PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
-    tombstoned.max_tombstone_count=50
-endif
-
-PRODUCT_DEFAULT_PROPERTY_OVERRIDES += \
-    ro.logd.size.stats=64K \
-    log.tag.stats_log=I
-
 # Enable CFI for security-sensitive components
 $(call inherit-product, $(SRC_TARGET_DIR)/product/cfi-common.mk)
 $(call inherit-product-if-exists, vendor/google/products/cfi-vendor.mk)

diff --git a/target/product/profile_boot_common.mk b/target/product/profile_boot_common.mk
index fc199541f..6f74cb2ac 100644
--- a/target/product/profile_boot_common.mk
+++ b/target/product/profile_boot_common.mk
@@ -21,7 +21,7 @@
 # support to separate the image from the compile code.
 PRODUCT_DEX_PREOPT_BOOT_IMAGE_PROFILE_LOCATION := build/make/target/product/empty-profile
 PRODUCT_DEX_PREOPT_BOOT_FLAGS := --count-hotness-in-compiled-code
-DEX_PREOPT_DEFAULT := nostripping
+DEX_PREOPT_DEFAULT := true

 # Disable uncompressing priv apps so that there is enough space to build the system partition.
 DONT_UNCOMPRESS_PRIV_APPS_DEXS := true

diff --git a/target/product/runtime_libart.mk b/target/product/runtime_libart.mk
index a88ba3c8d..ed19be911 100644
--- a/target/product/runtime_libart.mk
+++ b/target/product/runtime_libart.mk
@@ -61,22 +61,22 @@ PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
 # On eng builds, make "boot" reasons only extract for faster turnaround.
 ifeq (eng,$(TARGET_BUILD_VARIANT))
     PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
-        pm.dexopt.first-boot=extract \
-        pm.dexopt.boot=extract
+        pm.dexopt.first-boot=speed \
+        pm.dexopt.boot=speed
 else
     PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
-        pm.dexopt.first-boot=quicken \
-        pm.dexopt.boot=verify
+        pm.dexopt.first-boot=speed \
+        pm.dexopt.boot=speed
 endif

 # The install filter is speed-profile in order to enable the use of
 # profiles from the dex metadata files. Note that if a profile is not provided
 # or if it is empty speed-profile is equivalent to (quicken + empty app image).
 PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
-    pm.dexopt.install=speed-profile \
-    pm.dexopt.bg-dexopt=speed-profile \
-    pm.dexopt.ab-ota=speed-profile \
-    pm.dexopt.inactive=verify \
+    pm.dexopt.install=speed \
+    pm.dexopt.bg-dexopt=speed \
+    pm.dexopt.ab-ota=speed \
+    pm.dexopt.inactive=speed \
     pm.dexopt.shared=speed

 # Enable resolution of startup const strings.
@@ -87,11 +87,6 @@ PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
 PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
     dalvik.vm.dex2oat-max-image-block-size=524288

-# Enable minidebuginfo generation unless overridden.
-PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
-    dalvik.vm.minidebuginfo=true \
-    dalvik.vm.dex2oat-minidebuginfo=true
-
 # Disable iorapd by default
 PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
     ro.iorapd.enable=false

diff --git a/target/product/updatable_apex.mk b/target/product/updatable_apex.mk
index 038f66ee6..a9f4baf30 100644
--- a/target/product/updatable_apex.mk
+++ b/target/product/updatable_apex.mk
@@ -17,5 +17,4 @@
 # Inherit this when the target needs to support updating APEXes

 PRODUCT_PROPERTY_OVERRIDES := ro.apex.updatable=true
-PRODUCT_PACKAGES := com.android.apex.cts.shim.v1_prebuilt
 TARGET_FLATTEN_APEX := false

diff --git a/tools/acp/Android.bp b/tools/acp/Android.bp
index 64f5a1013..8b84418b3 100644
--- a/tools/acp/Android.bp
+++ b/tools/acp/Android.bp
@@ -5,7 +5,7 @@
 cc_binary_host {

     srcs: ["acp.c"],
-    cflags: ["-Wall", "-Werror"],
+    cflags: ["-Wall"],

     static_libs: ["libhost"],
     name: "acp",

diff --git a/tools/atree/Android.bp b/tools/atree/Android.bp
index 5fbe042ea..fc8a8aada 100644
--- a/tools/atree/Android.bp
+++ b/tools/atree/Android.bp
@@ -9,6 +9,6 @@ cc_binary_host {
         "files.cpp",
         "fs.cpp",
     ],
-    cflags: ["-Wall", "-Werror"],
+    cflags: ["-Wall"],
     static_libs: ["libhost"],
 }

diff --git a/tools/fs_config/Android.bp b/tools/fs_config/Android.bp
index d9a48d704..60bfe673f 100644
--- a/tools/fs_config/Android.bp
+++ b/tools/fs_config/Android.bp
@@ -32,7 +32,6 @@ cc_binary_host {
         "libcutils",
         "libselinux",
     ],
-    cflags: ["-Werror"],
 }

 target_fs_config_gen_filegroup {

diff --git a/tools/fs_get_stats/Android.bp b/tools/fs_get_stats/Android.bp
index 67742b836..f710d506e 100644
--- a/tools/fs_get_stats/Android.bp
+++ b/tools/fs_get_stats/Android.bp
@@ -1,7 +1,7 @@
 cc_binary_host {
     name: "fs_get_stats",
     srcs: ["fs_get_stats.c"],
-    cflags: ["-Wall", "-Werror"],
+    cflags: ["-Wall"],
     shared_libs: [
         "libcutils",
         "liblog",

diff --git a/tools/libhost/Android.bp b/tools/libhost/Android.bp
index 4c9100f15..82268553b 100644
--- a/tools/libhost/Android.bp
+++ b/tools/libhost/Android.bp
@@ -3,7 +3,6 @@ cc_library_host_static {
     srcs: ["CopyFile.c"],

     cflags: [
-        "-Werror",
         "-Wall",
     ],

diff --git a/tools/makeparallel/Android.bp b/tools/makeparallel/Android.bp
index 898db6861..14be89ac2 100644
--- a/tools/makeparallel/Android.bp
+++ b/tools/makeparallel/Android.bp
@@ -17,5 +17,5 @@ cc_binary_host {
     srcs: [
         "makeparallel.cpp",
     ],
-    cflags: ["-Wall", "-Werror"],
+    cflags: ["-Wall"],
 }

diff --git a/tools/makeparallel/Makefile b/tools/makeparallel/Makefile
index 82a4abfac..b2595d45e 100644
--- a/tools/makeparallel/Makefile
+++ b/tools/makeparallel/Makefile
@@ -27,7 +27,7 @@ MAKEPARALLEL_BIN_PATH ?= .
 MAKEPARALLEL_CXX_SRCS := \
 	makeparallel.cpp

-MAKEPARALLEL_CXXFLAGS := -Wall -Werror -MMD -MP
+MAKEPARALLEL_CXXFLAGS := -Wall -MMD -MP

 MAKEPARALLEL_CXX_SRCS := $(addprefix $(MAKEPARALLEL_SRC_PATH)/,\
 	$(MAKEPARALLEL_CXX_SRCS))

diff --git a/tools/releasetools/common.py b/tools/releasetools/common.py
index 9f57773ba..e0a41f30b 100644
--- a/tools/releasetools/common.py
+++ b/tools/releasetools/common.py
@@ -2148,7 +2148,7 @@ class BlockDifference(object):
     #   decompression_time: 15s  | 25s                | 25s

     if not self.src:
-      brotli_cmd = ['brotli', '--quality=6',
+      brotli_cmd = ['brotli', '--quality=9',
                     '--output={}.new.dat.br'.format(self.path),
                     '{}.new.dat'.format(self.path)]
       print("Compressing {}.new.dat with brotli".format(self.partition))

diff --git a/tools/zipalign/Android.bp b/tools/zipalign/Android.bp
index 8e6196d29..9e65a5cf9 100644
--- a/tools/zipalign/Android.bp
+++ b/tools/zipalign/Android.bp
@@ -13,7 +13,7 @@ cc_binary_host {
         "ZipFile.cpp",
     ],

-    cflags: ["-Wall", "-Werror"],
+    cflags: ["-Wall"],

     // NOTE: Do not add any shared_libs dependencies because they will break the
     // static_sdk_tools target.

diff --git a/tools/ziptime/Android.bp b/tools/ziptime/Android.bp
index 5ef45ed40..7d882ae72 100644
--- a/tools/ziptime/Android.bp
+++ b/tools/ziptime/Android.bp
@@ -27,7 +27,7 @@ cc_binary_host {
     ],

     name: "ziptime",
-    cflags: ["-Wall", "-Werror"],
+    cflags: ["-Wall"],
     target: {
         windows: {
             enabled: true,

project device/lineage/sepolicy/
diff --git a/common/private/backuptool.te b/common/private/backuptool.te
index b948b61..d62a3e0 100644
--- a/common/private/backuptool.te
+++ b/common/private/backuptool.te
@@ -6,6 +6,6 @@ neverallow {
     -update_engine
 } backuptool:process transition;

-userdebug_or_eng(`
+
     permissive backuptool;
-')
+

diff --git a/common/private/file_contexts b/common/private/file_contexts
index d6708f8..c3e62f6 100644
--- a/common/private/file_contexts
+++ b/common/private/file_contexts
@@ -18,6 +18,9 @@
 /system/bin/backuptool_ab\.sh                     u:object_r:otapreopt_chroot_exec:s0
 /system/bin/backuptool_postinstall\.sh            u:object_r:otapreopt_chroot_exec:s0

+# Superuser's control sockets
+/dev/socket/su-daemon(/.*)?             u:object_r:superuser_device:s0
+
 # ADB Root
 /system/bin/adb_root       u:object_r:adbroot_exec:s0
 /data/adbroot(/.*)?        u:object_r:adbroot_data_file:s0

diff --git a/common/private/recovery.te b/common/private/recovery.te
index f55499a..f339c31 100644
--- a/common/private/recovery.te
+++ b/common/private/recovery.te
@@ -1,5 +1,5 @@
 recovery_only(`
-userdebug_or_eng(`
+
 permissive recovery;
-')
+
 ')

project external/bash/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index 1a23613..0000000
--- a/Android.mk
+++ /dev/null

project external/nano/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index 4fcd4369..00000000
--- a/Android.mk
+++ /dev/null

project external/noto-fonts/
diff --git a/Android.mk b/Android.mk
index 249c7ac..d39a9d3 100644
--- a/Android.mk
+++ b/Android.mk
@@ -41,28 +41,13 @@ ifneq ($(MINIMAL_FONT_FOOTPRINT),true)
 LOCAL_PATH := $(NOTO_DIR)/cjk

 font_src_files := \
-    NotoSansCJK-Regular.ttc
+    NotoSansCJK.ttc

 $(foreach f, $(font_src_files), $(call build-one-font-module, $(f)))
 font_src_files :=

 endif # !MINIMAL_FONT_FOOTPRINT

-#############################################################################
-# Similary "build" the Noto CJK fonts for serif family.
-# These are not included in SMALLER_FONT_FOOTPRINT builds.
-#############################################################################
-ifeq ($(filter true,$(EXCLUDE_SERIF_FONTS) $(SMALLER_FONT_FOOTPRINT)),)
-LOCAL_PATH := $(NOTO_DIR)/cjk
-
-font_src_files := \
-    NotoSerifCJK-Regular.ttc
-
-$(foreach f, $(font_src_files), $(call build-one-font-module, $(f)))
-font_src_files :=
-
-endif # !EXCLUDE_SERIF_FONTS && !SMALLER_FONT_FOOTPRINT
-
 #############################################################################
 # Now "build" the Noto Color Emoji font, which is in its own directory. It is
 # not included in the MINIMAL_FONT_FOOTPRINT builds.

diff --git a/cjk/subset_noto_cjk.py b/cjk/subset_noto_cjk.py
index 5a324bf..4a274ce 100755
--- a/cjk/subset_noto_cjk.py
+++ b/cjk/subset_noto_cjk.py
@@ -98,6 +98,6 @@ def remove_codepoints_from_ttc(ttc_name):
         for f in otf_names:
             os.remove(f)

-
+remove_codepoints_from_ttc('NotoSansCJK.ttc')
 remove_codepoints_from_ttc('NotoSansCJK-Regular.ttc')
 remove_codepoints_from_ttc('NotoSerifCJK-Regular.ttc')

diff --git a/fonts.mk b/fonts.mk
index fee43e2..1e3cf5e 100644
--- a/fonts.mk
+++ b/fonts.mk
@@ -49,7 +49,7 @@ PRODUCT_PACKAGES := \
     NotoSansCham-Bold.ttf \
     NotoSansCham-Regular.ttf \
     NotoSansCherokee-Regular.ttf \
-    NotoSansCJK-Regular.ttc \
+    NotoSansCJK.ttc \
     NotoSansCoptic-Regular.ttf \
     NotoSansCuneiform-Regular.ttf \
     NotoSansCypriot-Regular.ttf \
@@ -205,7 +205,6 @@ PRODUCT_PACKAGES := \
     NotoSerifArmenian-Regular.otf \
     NotoSerifBengali-Bold.ttf \
     NotoSerifBengali-Regular.ttf \
-    NotoSerifCJK-Regular.ttc \
     NotoSerifDevanagari-Bold.ttf \
     NotoSerifDevanagari-Regular.ttf \
     NotoSerifEthiopic-Bold.otf \

diff --git a/other/NotoSerif-Bold.ttf b/other/NotoSerif-Bold.ttf
index 2ffdf8c..128288b 100644
Binary files a/other/NotoSerif-Bold.ttf and b/other/NotoSerif-Bold.ttf differ

diff --git a/other/NotoSerif-Regular.ttf b/other/NotoSerif-Regular.ttf
index 9d9a347..78136ac 100644
Binary files a/other/NotoSerif-Regular.ttf and b/other/NotoSerif-Regular.ttf differ

project external/perfetto/
diff --git a/Android.bp b/Android.bp
index a11041afe..81bc30193 100644
--- a/Android.bp
+++ b/Android.bp
@@ -3370,49 +3370,6 @@ cc_binary_host {
   ],
 }

-// GN target: //:traced
-cc_binary {
-  name: "traced",
-  srcs: [
-    "src/traced/service/main.cc",
-  ],
-  shared_libs: [
-    "liblog",
-    "libperfetto",
-  ],
-  init_rc: [
-    "perfetto.rc",
-  ],
-  defaults: [
-    "perfetto_defaults",
-  ],
-  cflags: [
-    "-DPERFETTO_BUILD_WITH_ANDROID",
-  ],
-}
-
-// GN target: //:traced_probes
-cc_binary {
-  name: "traced_probes",
-  srcs: [
-    "src/traced/probes/main.cc",
-  ],
-  shared_libs: [
-    "liblog",
-    "libperfetto",
-  ],
-  defaults: [
-    "perfetto_defaults",
-  ],
-  cflags: [
-    "-DPERFETTO_BUILD_WITH_ANDROID",
-  ],
-  required: [
-    "libperfetto_android_internal",
-    "trigger_perfetto",
-  ],
-}
-
 // GN target: //:trigger_perfetto
 cc_binary {
   name: "trigger_perfetto",
@@ -3604,4 +3561,4 @@ java_library_host {
   srcs: [
     "protos/perfetto/config/perfetto_config.proto",
   ],
-}
\ No newline at end of file
+}

project external/vim/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index 858a4e8a3..000000000
--- a/Android.mk
+++ /dev/null

project frameworks/base/
diff --git a/core/res/AndroidManifest.xml b/core/res/AndroidManifest.xml
index aa038961a0f..df996b44469 100644
--- a/core/res/AndroidManifest.xml
+++ b/core/res/AndroidManifest.xml
@@ -2644,6 +2644,13 @@
         android:description="@string/permdesc_getPackageSize"
         android:protectionLevel="normal" />

+    <!-- @hide Allows an application to change the package signature as
+         seen by applications -->
+    <permission android:name="android.permission.FAKE_PACKAGE_SIGNATURE"
+        android:protectionLevel="dangerous"
+        android:label="@string/permlab_fakePackageSignature"
+        android:description="@string/permdesc_fakePackageSignature" />
+
     <!-- @deprecated No longer useful, see
          {@link android.content.pm.PackageManager#addPackageToPreferred}
          for details. -->

diff --git a/core/res/res/values-zh-rCN/strings.xml b/core/res/res/values-zh-rCN/strings.xml
index 833fd533fe7..02a7e827166 100644
--- a/core/res/res/values-zh-rCN/strings.xml
+++ b/core/res/res/values-zh-rCN/strings.xml
@@ -2046,4 +2046,6 @@
     </plurals>
     <string name="chooser_no_direct_share_targets" msgid="492542066060841139">"无法使用直接分享功能"</string>
     <string name="chooser_all_apps_button_label" msgid="3230427756238666328">"应用列表"</string>
+    <string name="permlab_fakePackageSignature">模拟软件包签名</string>
+    <string name="permdesc_fakePackageSignature">允许应用伪装成为其他应用</string>
 </resources>

diff --git a/core/res/res/values/config.xml b/core/res/res/values/config.xml
index 131e015f5a1..d4b9b5d7fbc 100644
--- a/core/res/res/values/config.xml
+++ b/core/res/res/values/config.xml
@@ -1872,6 +1872,8 @@
     <string-array name="config_locationProviderPackageNames" translatable="false">
         <!-- The standard AOSP fused location provider -->
         <item>com.android.location.fused</item>
+        <!-- The (faked) microg fused location provider (a free reimplementation) -->
+        <item>com.google.android.gms</item>
     </string-array>

     <!-- This string array can be overriden to enable test location providers initially. -->

diff --git a/core/res/res/values/strings.xml b/core/res/res/values/strings.xml
index 800399ef084..286fe752a51 100644
--- a/core/res/res/values/strings.xml
+++ b/core/res/res/values/strings.xml
@@ -830,6 +830,11 @@

     <!--  Permissions -->

+    <!-- Title of an application permission, listed so the user can choose whether they want to allow the application to do this. -->
+    <string name="permlab_fakePackageSignature">Spoof package signature</string>
+    <!-- Description of an application permission, listed so the user can choose whether they want to allow the application to do this. -->
+    <string name="permdesc_fakePackageSignature">Allows the app to pretend to be a different app.</string>
+
     <!-- Title of an application permission, listed so the user can choose whether they want to allow the application to do this. -->
     <string name="permlab_statusBar">disable or modify status bar</string>
     <!-- Description of an application permission, listed so the user can choose whether they want to allow the application to do this. -->

diff --git a/data/fonts/fonts.xml b/data/fonts/fonts.xml
index c6920977f6b..b993005e015 100644
--- a/data/fonts/fonts.xml
+++ b/data/fonts/fonts.xml
@@ -546,20 +546,56 @@
         <font weight="400" style="normal">NotoSansSymbols-Regular-Subsetted.ttf</font>
     </family>
     <family lang="zh-Hans">
-        <font weight="400" style="normal" index="2">NotoSansCJK-Regular.ttc</font>
-        <font weight="400" style="normal" index="2" fallbackFor="serif">NotoSerifCJK-Regular.ttc</font>
-    </family>
-    <family lang="zh-Hant,zh-Bopo">
-        <font weight="400" style="normal" index="3">NotoSansCJK-Regular.ttc</font>
-        <font weight="400" style="normal" index="3" fallbackFor="serif">NotoSerifCJK-Regular.ttc</font>
+        <font weight="100" style="normal" index="2">NotoSansCJK.ttc</font>
+        <font weight="300" style="normal" index="7">NotoSansCJK.ttc</font>
+        <font weight="350" style="normal" index="12">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="17">NotoSansCJK.ttc</font>
+        <font weight="500" style="normal" index="27">NotoSansCJK.ttc</font>
+        <font weight="700" style="normal" index="32">NotoSansCJK.ttc</font>
+        <font weight="900" style="normal" index="42">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" fallbackFor="serif">NotoSerif-Regular.ttf</font>
+        <font weight="700" style="normal" fallbackFor="serif">NotoSerif-Bold.ttf</font>
+        <font weight="400" style="italic" fallbackFor="serif">NotoSerif-Italic.ttf</font>
+        <font weight="700" style="italic" fallbackFor="serif">NotoSerif-BoldItalic.ttf</font>
+    </family>
+    <family lang="zh-Hant zh-Bopo">
+        <font weight="100" style="normal" index="3">NotoSansCJK.ttc</font>
+        <font weight="300" style="normal" index="8">NotoSansCJK.ttc</font>
+        <font weight="350" style="normal" index="13">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="18">NotoSansCJK.ttc</font>
+        <font weight="500" style="normal" index="28">NotoSansCJK.ttc</font>
+        <font weight="700" style="normal" index="33">NotoSansCJK.ttc</font>
+        <font weight="900" style="normal" index="43">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" fallbackFor="serif">NotoSerif-Regular.ttf</font>
+        <font weight="700" style="normal" fallbackFor="serif">NotoSerif-Bold.ttf</font>
+        <font weight="400" style="italic" fallbackFor="serif">NotoSerif-Italic.ttf</font>
+        <font weight="700" style="italic" fallbackFor="serif">NotoSerif-BoldItalic.ttf</font>
     </family>
     <family lang="ja">
-        <font weight="400" style="normal" index="0">NotoSansCJK-Regular.ttc</font>
-        <font weight="400" style="normal" index="0" fallbackFor="serif">NotoSerifCJK-Regular.ttc</font>
+        <font weight="100" style="normal" index="0">NotoSansCJK.ttc</font>
+        <font weight="300" style="normal" index="5">NotoSansCJK.ttc</font>
+        <font weight="350" style="normal" index="10">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="15">NotoSansCJK.ttc</font>
+        <font weight="500" style="normal" index="25">NotoSansCJK.ttc</font>
+        <font weight="700" style="normal" index="30">NotoSansCJK.ttc</font>
+        <font weight="900" style="normal" index="40">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" fallbackFor="serif">NotoSerif-Regular.ttf</font>
+        <font weight="700" style="normal" fallbackFor="serif">NotoSerif-Bold.ttf</font>
+        <font weight="400" style="italic" fallbackFor="serif">NotoSerif-Italic.ttf</font>
+        <font weight="700" style="italic" fallbackFor="serif">NotoSerif-BoldItalic.ttf</font>
     </family>
     <family lang="ko">
-        <font weight="400" style="normal" index="1">NotoSansCJK-Regular.ttc</font>
-        <font weight="400" style="normal" index="1" fallbackFor="serif">NotoSerifCJK-Regular.ttc</font>
+        <font weight="100" style="normal" index="1">NotoSansCJK.ttc</font>
+        <font weight="300" style="normal" index="6">NotoSansCJK.ttc</font>
+        <font weight="350" style="normal" index="11">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="16">NotoSansCJK.ttc</font>
+        <font weight="500" style="normal" index="26">NotoSansCJK.ttc</font>
+        <font weight="700" style="normal" index="31">NotoSansCJK.ttc</font>
+        <font weight="900" style="normal" index="41">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" fallbackFor="serif">NotoSerif-Regular.ttf</font>
+        <font weight="700" style="normal" fallbackFor="serif">NotoSerif-Bold.ttf</font>
+        <font weight="400" style="italic" fallbackFor="serif">NotoSerif-Italic.ttf</font>
+        <font weight="700" style="italic" fallbackFor="serif">NotoSerif-BoldItalic.ttf</font>
     </family>
     <family lang="und-Zsye">
         <font weight="400" style="normal">NotoColorEmoji.ttf</font>

diff --git a/packages/CtsShim/Android.mk b/packages/CtsShim/Android.mk
deleted file mode 100644
index 12972f14514..00000000000
--- a/packages/CtsShim/Android.mk
+++ /dev/null

diff --git a/packages/EasterEgg/Android.bp b/packages/EasterEgg/Android.bp
deleted file mode 100644
index 43ed810b567..00000000000
--- a/packages/EasterEgg/Android.bp
+++ /dev/null

diff --git a/packages/overlays/Android.mk b/packages/overlays/Android.mk
index eecc10158e8..7adfcfe9033 100644
--- a/packages/overlays/Android.mk
+++ b/packages/overlays/Android.mk
@@ -17,41 +17,6 @@ include $(CLEAR_VARS)

 LOCAL_MODULE := frameworks-base-overlays
 LOCAL_REQUIRED_MODULES := \
-	AccentColorBlackOverlay \
-	AccentColorCinnamonOverlay \
-	AccentColorOceanOverlay \
-	AccentColorOrchidOverlay \
-	AccentColorSpaceOverlay \
-	AccentColorGreenOverlay \
-	AccentColorPurpleOverlay \
-	DisplayCutoutEmulationCornerOverlay \
-	DisplayCutoutEmulationDoubleOverlay \
-	DisplayCutoutEmulationTallOverlay \
-	FontNotoSerifSourceOverlay \
-	IconPackCircularAndroidOverlay \
-	IconPackCircularLauncherOverlay \
-	IconPackCircularSettingsOverlay \
-	IconPackCircularSystemUIOverlay \
-	IconPackCircularThemePickerOverlay \
-	IconPackFilledAndroidOverlay \
-	IconPackFilledLauncherOverlay \
-	IconPackFilledSettingsOverlay \
-	IconPackFilledSystemUIOverlay \
-	IconPackFilledThemePickerOverlay \
-	IconPackRoundedAndroidOverlay \
-	IconPackRoundedLauncherOverlay \
-	IconPackRoundedSettingsOverlay \
-	IconPackRoundedSystemUIOverlay \
-	IconPackRoundedThemePickerOverlay \
-	IconShapeRoundedRectOverlay \
-	IconShapeSquircleOverlay \
-	IconShapeTeardropOverlay \
-	NavigationBarMode3ButtonOverlay \
-	NavigationBarMode2ButtonOverlay \
-	NavigationBarModeGesturalOverlay \
-	NavigationBarModeGesturalOverlayNarrowBack \
-	NavigationBarModeGesturalOverlayWideBack \
-	NavigationBarModeGesturalOverlayExtraWideBack

 include $(BUILD_PHONY_PACKAGE)
 include $(CLEAR_VARS)

diff --git a/services/core/java/com/android/server/pm/PackageManagerService.java b/services/core/java/com/android/server/pm/PackageManagerService.java
index 9aed9109e24..c8f4967676b 100644
--- a/services/core/java/com/android/server/pm/PackageManagerService.java
+++ b/services/core/java/com/android/server/pm/PackageManagerService.java
@@ -4157,8 +4157,9 @@ public class PackageManagerService extends IPackageManager.Stub
             final Set<String> permissions = ArrayUtils.isEmpty(p.requestedPermissions)
                     ? Collections.emptySet() : permissionsState.getPermissions(userId);

-            PackageInfo packageInfo = PackageParser.generatePackageInfo(p, gids, flags,
-                    ps.firstInstallTime, ps.lastUpdateTime, permissions, state, userId);
+            PackageInfo packageInfo = mayFakeSignature(p, PackageParser.generatePackageInfo(p, gids, flags,
+                    ps.firstInstallTime, ps.lastUpdateTime, permissions, state, userId),
+                    permissions);

             if (packageInfo == null) {
                 return null;
@@ -4194,6 +4195,24 @@ public class PackageManagerService extends IPackageManager.Stub
         }
     }

+    private PackageInfo mayFakeSignature(PackageParser.Package p, PackageInfo pi,
+            Set<String> permissions) {
+        try {
+            if (permissions.contains("android.permission.FAKE_PACKAGE_SIGNATURE")
+                    && p.applicationInfo.targetSdkVersion > Build.VERSION_CODES.LOLLIPOP_MR1
+                    && p.mAppMetaData != null) {
+                String sig = p.mAppMetaData.getString("fake-signature");
+                if (sig != null) {
+                    pi.signatures = new Signature[] {new Signature(sig)};
+                }
+            }
+        } catch (Throwable t) {
+            // We should never die because of any failures, this is system code!
+            Log.w("PackageManagerService.FAKE_PACKAGE_SIGNATURE", t);
+           }
+        return pi;
+    }
+
     @Override
     public void checkPackageStartable(String packageName, int userId) {
         final int callingUid = Binder.getCallingUid();

project hardware/qcom-caf/msm8994/audio/
diff --git a/hal/Android.mk b/hal/Android.mk
index 7eca4aaf7..46d6fd0d1 100644
--- a/hal/Android.mk
+++ b/hal/Android.mk
@@ -264,8 +264,6 @@ ifeq ($(strip $(AUDIO_FEATURE_ENABLED_AUXPCM_BT)),true)
     LOCAL_CFLAGS += -DAUXPCM_BT_ENABLED
 endif

-LOCAL_CFLAGS += -Wall -Werror
-
 LOCAL_COPY_HEADERS_TO   := mm-audio
 LOCAL_COPY_HEADERS      := audio_extn/audio_defs.h

project hardware/qcom-caf/msm8994/media/
diff --git a/mm-video-v4l2/vidc/common/Android.mk b/mm-video-v4l2/vidc/common/Android.mk
index 6f783c429..bf248f963 100644
--- a/mm-video-v4l2/vidc/common/Android.mk
+++ b/mm-video-v4l2/vidc/common/Android.mk
@@ -7,13 +7,12 @@ LOCAL_PATH:= $(ROOT_DIR)
 # 				Common definitons
 # ---------------------------------------------------------------------------------

 libmm-vidc-def := -g -O3 -Dlrintf=_ffix_r
 libmm-vidc-def += -D__align=__alignx
 libmm-vidc-def += -D__alignx\(x\)=__attribute__\(\(__aligned__\(x\)\)\)
 libmm-vidc-def += -DT_ARM
 libmm-vidc-def += -Dinline=__inline
 libmm-vidc-def += -D_ANDROID_
-libmm-vidc-def += -Werror
 libmm-vidc-def += -D_ANDROID_ICS_

 # ---------------------------------------------------------------------------------

diff --git a/mm-video-v4l2/vidc/vdec/Android.mk b/mm-video-v4l2/vidc/vdec/Android.mk
index aedca2059..4cfdcc4f1 100644
--- a/mm-video-v4l2/vidc/vdec/Android.mk
+++ b/mm-video-v4l2/vidc/vdec/Android.mk
@@ -8,7 +8,7 @@ include $(CLEAR_VARS)
 libmm-vdec-def := -D__alignx\(x\)=__attribute__\(\(__aligned__\(x\)\)\)
 libmm-vdec-def += -D__align=__alignx
 libmm-vdec-def += -Dinline=__inline
 libmm-vdec-def += -g -O3
 libmm-vdec-def += -DIMAGE_APPS_PROC
 libmm-vdec-def += -D_ANDROID_
 libmm-vdec-def += -DCDECL
@@ -97,7 +97,7 @@ include $(CLEAR_VARS)
 LOCAL_MODULE                    := libOmxVdec
 LOCAL_MODULE_TAGS               := optional
 LOCAL_VENDOR_MODULE             := true
-LOCAL_CFLAGS                    := $(libmm-vdec-def) -Werror
+LOCAL_CFLAGS                    := $(libmm-vdec-def)
 LOCAL_C_INCLUDES                += $(libmm-vdec-inc)

 LOCAL_HEADER_LIBRARIES := generated_kernel_headers

diff --git a/mm-video-v4l2/vidc/venc/Android.mk b/mm-video-v4l2/vidc/venc/Android.mk
index ff43abdcd..6c3654f99 100644
--- a/mm-video-v4l2/vidc/venc/Android.mk
+++ b/mm-video-v4l2/vidc/venc/Android.mk
@@ -5,7 +5,7 @@ include $(CLEAR_VARS)
 # 				Common definitons
 # ---------------------------------------------------------------------------------

 libmm-venc-def := -g -O3 -Dlrintf=_ffix_r
 libmm-venc-def += -D__align=__alignx
 libmm-venc-def += -D__alignx\(x\)=__attribute__\(\(__aligned__\(x\)\)\)
 libmm-venc-def += -DT_ARM
@@ -17,7 +17,6 @@ libmm-venc-def += -DENABLE_DEBUG_ERROR
 libmm-venc-def += -UINPUT_BUFFER_LOG
 libmm-venc-def += -UOUTPUT_BUFFER_LOG
 libmm-venc-def += -USINGLE_ENCODER_INSTANCE
-libmm-venc-def += -Werror
 libmm-venc-def += -D_ANDROID_ICS_
 libmm-venc-def += -D_MSM8974_

project packages/apps/AudioFX/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index f44369e..0000000
--- a/Android.mk
+++ /dev/null

project packages/apps/Calendar/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index 4b5744a7..00000000
--- a/Android.mk
+++ /dev/null

project packages/apps/CellBroadcastReceiver/
diff --git a/Android.bp b/Android.bp
deleted file mode 100644
index cda810ce..00000000
--- a/Android.bp
+++ /dev/null

diff --git a/tests/testapp/Android.bp b/tests/testapp/Android.bp
deleted file mode 100644
index ff274933..00000000
--- a/tests/testapp/Android.bp
+++ /dev/null

diff --git a/tests/unit/Android.bp b/tests/unit/Android.bp
deleted file mode 100644
index 673ecff3..00000000
--- a/tests/unit/Android.bp
+++ /dev/null

project packages/apps/Email/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index 4fa104752..000000000
--- a/Android.mk
+++ /dev/null

project packages/apps/Exchange/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index 3469d737..00000000
--- a/Android.mk
+++ /dev/null

project packages/apps/Settings/
diff --git a/res/values-zh-rCN/cm_strings.xml b/res/values-zh-rCN/cm_strings.xml
index a3ca2b41a6..599b948a75 100644
--- a/res/values-zh-rCN/cm_strings.xml
+++ b/res/values-zh-rCN/cm_strings.xml
@@ -65,6 +65,9 @@
     <string name="security_settings_fingerprint_sensor_location_right">右侧</string>
     <string name="status_bar_double_tap_to_sleep_title">点按以睡眠</string>
     <string name="status_bar_double_tap_to_sleep_summary">双击状态栏或锁屏以关闭屏幕</string>
+    <string name="root_access">Root 授权</string>
+    <string name="root_access_none">已禁用</string>
+    <string name="root_access_all">应用与 ADB</string>
     <string name="touchscreen_gesture_settings_title">触摸屏手势</string>
     <string name="touchscreen_gesture_settings_summary">为快捷操作采用各种触摸屏手势</string>
     <string name="proximity_wake_title">防止意外唤醒</string>

diff --git a/res/values/cm_strings.xml b/res/values/cm_strings.xml
index d377cec7b9..e6367ffbed 100644
--- a/res/values/cm_strings.xml
+++ b/res/values/cm_strings.xml
@@ -122,6 +122,11 @@
     <string name="proximity_wake_title">Prevent accidental wake-up</string>
     <string name="proximity_wake_summary">Check the proximity sensor prior to waking up screen</string>

+    <!-- Setting checkbox title for root access -->
+    <string name="root_access">Root access</string>
+    <string name="root_access_none">Disabled</string>
+    <string name="root_access_all">Apps and ADB</string>
+
     <!-- Wake on plug -->
     <string name="wake_when_plugged_or_unplugged_title">Wake on plug</string>
     <string name="wake_when_plugged_or_unplugged_summary">Turn the screen on when connecting or disconnecting a power source</string>

diff --git a/res/xml/development_settings.xml b/res/xml/development_settings.xml
index 9e43b833a9..7977de5c42 100644
--- a/res/xml/development_settings.xml
+++ b/res/xml/development_settings.xml
@@ -126,6 +126,11 @@
             android:fragment="com.android.settings.development.qstile.DevelopmentTileConfigFragment"
             settings:searchable="false" />

+        <ListPreference
+            android:key="root_access"
+            android:title="@string/root_access"
+            android:persistent="false" />
+
     <!-- Configure trust agent behavior -->
     <SwitchPreference
         android:key="security_setting_trust_agents_extend_unlock"

diff --git a/src/com/android/settings/development/DevelopmentSettingsDashboardFragment.java b/src/com/android/settings/development/DevelopmentSettingsDashboardFragment.java
index a935386b98..3e26d7db01 100644
--- a/src/com/android/settings/development/DevelopmentSettingsDashboardFragment.java
+++ b/src/com/android/settings/development/DevelopmentSettingsDashboardFragment.java
@@ -499,6 +499,7 @@ public class DevelopmentSettingsDashboardFragment extends RestrictedDashboardFra
         controllers.add(new ShortcutManagerThrottlingPreferenceController(context));
         controllers.add(new BubbleGlobalPreferenceController(context));
         controllers.add(new EnableGnssRawMeasFullTrackingPreferenceController(context));
+        controllers.add(new RootAccessPreferenceController(context, fragment));
         controllers.add(new DefaultLaunchPreferenceController(context, "running_apps"));
         controllers.add(new DefaultLaunchPreferenceController(context, "demo_mode"));
         controllers.add(new DefaultLaunchPreferenceController(context, "quick_settings_tiles"));

project packages/apps/Stk/
diff --git a/Android.bp b/Android.bp
deleted file mode 100644
index a2dbc03..0000000
--- a/Android.bp
+++ /dev/null

project packages/apps/Traceur/
diff --git a/Android.bp b/Android.bp
deleted file mode 100644
index 875c34c..0000000
--- a/Android.bp
+++ /dev/null

project packages/apps/Updater/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index b74fe64..0000000
--- a/Android.mk
+++ /dev/null

project packages/modules/ModuleMetadata/
diff --git a/Android.bp b/Android.bp
index e8084b3..527622f 100644
--- a/Android.bp
+++ b/Android.bp
@@ -24,6 +24,6 @@ android_app {
     },

     dex_preopt: {
-        enabled: false,
+        enabled: true,
     },
 }

project packages/providers/BookmarkProvider/
diff --git a/Android.bp b/Android.bp
deleted file mode 100644
index 0f73493..0000000
--- a/Android.bp
+++ /dev/null

project packages/screensavers/Basic/
diff --git a/Android.bp b/Android.bp
deleted file mode 100644
index c9a65a8..0000000
--- a/Android.bp
+++ /dev/null

project packages/screensavers/PhotoTable/
diff --git a/Android.bp b/Android.bp
deleted file mode 100644
index 0097373..0000000
--- a/Android.bp
+++ /dev/null

project system/apex/
diff --git a/apexd/Android.bp b/apexd/Android.bp
index 6c6f75d..e9aed38 100644
--- a/apexd/Android.bp
+++ b/apexd/Android.bp
@@ -13,7 +13,6 @@ cc_defaults {
   cflags: [
     "-Wall",
     "-Wextra",
-    "-Werror",
     "-Wno-unused-parameter",

     // Some extra flags.
@@ -287,49 +286,3 @@ genrule {
        "$(genDir)/apex.apexd_test_corrupt_apex.apex"
 }

-cc_test {
-  name: "apexservice_test",
-  defaults: ["apex_defaults"],
-  cflags: [
-    // Otherwise libgmock won't compile.
-    "-Wno-used-but-marked-unused",
-  ],
-  data: [
-    ":apex.apexd_test",
-    ":apex.apexd_test_different_app",
-    ":apex.apexd_test_v2",
-    ":apex.apexd_test_no_inst_key",
-    ":apex.apexd_test_preinstall",
-    ":apex.apexd_test_postinstall",
-    ":apex.apexd_test_prepostinstall.fail",
-    ":gen_bad_apexes",
-    ":gen_corrupt_apex",
-    ":com.android.apex.cts.shim.v1_prebuilt",
-    ":com.android.apex.cts.shim.v2_prebuilt",
-    ":com.android.apex.cts.shim.v2_wrong_sha_prebuilt",
-    ":com.android.apex.cts.shim.v2_additional_file_prebuilt",
-    ":com.android.apex.cts.shim.v2_additional_folder_prebuilt",
-    ":com.android.apex.cts.shim.v2_with_pre_install_hook_prebuilt",
-    ":com.android.apex.cts.shim.v2_with_post_install_hook_prebuilt",
-  ],
-  srcs: ["apexservice_test.cpp"],
-  host_supported: false,
-  compile_multilib: "first",
-  static_libs: [
-    "apex_aidl_interface-cpp",
-    "libapex",
-    "libapexd",
-    "libavb",
-    "libdm",
-    "libgmock",
-    "libvold_binder",
-  ],
-  shared_libs: [
-    "libbinder",
-    "libselinux",
-    "libutils",
-  ],
-  test_suites: ["device-tests"],
-  test_config: "apexservice_test_config.xml",
-}
-

diff --git a/shim/Android.bp b/shim/Android.bp
deleted file mode 100644
index 8573e1c..0000000
--- a/shim/Android.bp
+++ /dev/null

diff --git a/shim/build/Android.bp b/shim/build/Android.bp
deleted file mode 100644
index 71f847c..0000000
--- a/shim/build/Android.bp
+++ /dev/null

project system/core/
diff --git a/debuggerd/Android.bp b/debuggerd/Android.bp
index 2e226da6c..2b66a94ac 100644
--- a/debuggerd/Android.bp
+++ b/debuggerd/Android.bp
@@ -258,76 +258,6 @@ cc_benchmark {
     ],
 }

-cc_binary {
-    name: "crash_dump",
-    srcs: [
-        "crash_dump.cpp",
-        "util.cpp",
-    ],
-    defaults: ["debuggerd_defaults"],
-
-    compile_multilib: "both",
-    multilib: {
-        lib32: {
-            suffix: "32",
-        },
-        lib64: {
-            suffix: "64",
-        },
-    },
-
-    static_libs: [
-        "libtombstoned_client_static",
-        "libdebuggerd",
-        "libcutils",
-    ],
-
-    shared_libs: [
-        "libbase",
-        "liblog",
-        "libprocinfo",
-        "libunwindstack",
-    ],
-}
-
-cc_binary {
-    name: "debuggerd",
-    srcs: [
-        "debuggerd.cpp",
-    ],
-    defaults: ["debuggerd_defaults"],
-
-    shared_libs: [
-        "libbase",
-        "libdebuggerd_client",
-        "liblog",
-        "libprocinfo",
-    ],
-
-    local_include_dirs: ["include"],
-}
-
-cc_binary {
-    name: "tombstoned",
-    srcs: [
-        "util.cpp",
-        "tombstoned/intercept_manager.cpp",
-        "tombstoned/tombstoned.cpp",
-    ],
-    defaults: ["debuggerd_defaults"],
-
-    header_libs: ["libdebuggerd_common_headers"],
-
-    static_libs: [
-        "libbase",
-        "libcutils",
-        "libevent",
-        "liblog",
-    ],
-
-    init_rc: ["tombstoned/tombstoned.rc"],
-}
-
 subdirs = [
     "crasher",
 ]

diff --git a/healthd/Android.bp b/healthd/Android.bp
index 2cf6be96d..ff3937b67 100644
--- a/healthd/Android.bp
+++ b/healthd/Android.bp
@@ -10,7 +10,6 @@ cc_library_headers {
 cc_library_static {
     name: "libbatterymonitor",
     srcs: ["BatteryMonitor.cpp"],
-    cflags: ["-Wall", "-Werror"],
+    cflags: ["-Wall"],
     vendor_available: true,
     recovery_available: true,
     export_include_dirs: ["include"],
@@ -25,11 +24,6 @@ cc_library_static {
 cc_defaults {
     name: "android.hardware.health@2.0-service_defaults",

     cflags: [
         "-Wall",
-        "-Werror",
     ],

     static_libs: [
         "android.hardware.health@2.0-impl",
         "android.hardware.health@1.0-convert",
@@ -92,11 +86,6 @@ cc_library_static {
         "healthd_mode_charger_nops.cpp",
     ],

     cflags: [
         "-Wall",
-        "-Werror",
     ],

     header_libs: [
         "libhealthd_headers",
     ],

diff --git a/healthd/Android.mk b/healthd/Android.mk
index 4eb98dae0..01707af8e 100644
--- a/healthd/Android.mk
+++ b/healthd/Android.mk
@@ -31,7 +31,6 @@ include $(BUILD_STATIC_LIBRARY)
 ### libhealthd_charger ###
 include $(CLEAR_VARS)

-LOCAL_CFLAGS := -Werror
 ifeq ($(strip $(BOARD_CHARGER_DISABLE_INIT_BLANK)),true)
 LOCAL_CFLAGS += -DCHARGER_DISABLE_INIT_BLANK
 endif
@@ -82,7 +81,6 @@ LOCAL_SRC_FILES := \
 LOCAL_MODULE := charger
 LOCAL_C_INCLUDES := $(LOCAL_PATH)/include

-LOCAL_CFLAGS := -Werror
 ifeq ($(strip $(LOCAL_CHARGER_NO_UI)),true)
 LOCAL_CFLAGS += -DCHARGER_NO_UI
 endif
@@ -140,7 +138,6 @@ LOCAL_MODULE_PATH := $(TARGET_RECOVERY_ROOT_OUT)/system/bin
 LOCAL_MODULE_STEM := charger

 LOCAL_C_INCLUDES := $(LOCAL_PATH)/include
-LOCAL_CFLAGS := -Wall -Werror
+LOCAL_CFLAGS := -Wall
 LOCAL_CFLAGS += -DCHARGER_NO_UI

 # charger.recovery doesn't link against libhealthd_{charger,draw} or libminui, since it doesn't need
@@ -174,7 +171,7 @@ include $(BUILD_EXECUTABLE)
 include $(CLEAR_VARS)
 LOCAL_MODULE := charger_test
 LOCAL_C_INCLUDES := $(LOCAL_PATH)/include
-LOCAL_CFLAGS := -Wall -Werror -DCHARGER_NO_UI
+LOCAL_CFLAGS := -Wall -DCHARGER_NO_UI
 LOCAL_STATIC_LIBRARIES := $(CHARGER_STATIC_LIBRARIES)
 LOCAL_SHARED_LIBRARIES := $(CHARGER_SHARED_LIBRARIES)
 LOCAL_SRC_FILES := \

diff --git a/init/Android.bp b/init/Android.bp
index 377a3740c..7ab7f3b15 100644
--- a/init/Android.bp
+++ b/init/Android.bp
@@ -27,7 +27,7 @@ cc_defaults {
         "-Wno-unused-parameter",
         "-Werror",
         "-DALLOW_LOCAL_PROP_OVERRIDE=0",
-        "-DALLOW_PERMISSIVE_SELINUX=0",
+        "-DALLOW_PERMISSIVE_SELINUX=1",
         "-DREBOOT_BOOTLOADER_ON_PANIC=0",
         "-DWORLD_WRITABLE_KMSG=0",
         "-DDUMP_ON_UMOUNT_FAILURE=0",

diff --git a/init/Android.mk b/init/Android.mk
index c4f7d34b2..8739e5698 100644
--- a/init/Android.mk
+++ b/init/Android.mk
@@ -16,7 +16,7 @@ init_options += \
 else
 init_options += \
     -DALLOW_LOCAL_PROP_OVERRIDE=0 \
-    -DALLOW_PERMISSIVE_SELINUX=0 \
+    -DALLOW_PERMISSIVE_SELINUX=1 \
     -DREBOOT_BOOTLOADER_ON_PANIC=0 \
     -DWORLD_WRITABLE_KMSG=0 \
     -DDUMP_ON_UMOUNT_FAILURE=0

diff --git a/rootdir/init.rc b/rootdir/init.rc
index 58a83e091..91c0ee765 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -434,11 +434,6 @@ on post-fs-data
     # Make sure we have the device encryption key.
     installkey /data

-    # Start bootcharting as soon as possible after the data partition is
-    # mounted to collect more data.
-    mkdir /data/bootchart 0755 shell shell
-    bootchart start
-
     # Load fsverity keys. This needs to happen before apexd, as post-install of
     # APEXes may rely on keys.
     exec -- /system/bin/fsverity_init
@@ -499,7 +494,6 @@ on post-fs-data
     mkdir /data/misc/ethernet 0770 system system
     mkdir /data/misc/dhcp 0770 dhcp dhcp
     mkdir /data/misc/user 0771 root root
-    mkdir /data/misc/perfprofd 0775 root root
     # give system access to wpa_supplicant.conf for backup and restore
     chmod 0660 /data/misc/wifi/wpa_supplicant.conf
     mkdir /data/local 0751 root root
@@ -507,18 +501,10 @@ on post-fs-data
     mkdir /data/misc/audioserver 0700 audioserver audioserver
     mkdir /data/misc/cameraserver 0700 cameraserver cameraserver
     mkdir /data/misc/vold 0700 root root
-    mkdir /data/misc/boottrace 0771 system shell
-    mkdir /data/misc/update_engine 0700 root root
-    mkdir /data/misc/update_engine_log 02750 root log
-    mkdir /data/misc/trace 0700 root root
-    # create location to store surface and window trace files
-    mkdir /data/misc/wmtrace 0700 system system
     # profile file layout
     mkdir /data/misc/profiles 0771 system system
     mkdir /data/misc/profiles/cur 0771 system system
     mkdir /data/misc/profiles/ref 0771 system system
-    mkdir /data/misc/profman 0770 system shell
-    mkdir /data/misc/gcov 0770 root root

     mkdir /data/preloads 0775 system system

@@ -530,7 +516,6 @@ on post-fs-data
     # For security reasons, /data/local/tmp should always be empty.
     # Do not place files or directories in /data/local/tmp
     mkdir /data/local/tmp 0771 shell shell
-    mkdir /data/local/traces 0777 shell shell
     mkdir /data/data 0771 system system
     mkdir /data/app-private 0771 system system
     mkdir /data/app-ephemeral 0771 system system
@@ -538,18 +523,9 @@ on post-fs-data
     mkdir /data/app-lib 0771 system system
     mkdir /data/app 0771 system system
     mkdir /data/property 0700 root root
-    mkdir /data/tombstones 0771 system system
-    mkdir /data/vendor/tombstones 0771 root root
-    mkdir /data/vendor/tombstones/wifi 0771 wifi wifi

     # create dalvik-cache, so as to enforce our permissions
     mkdir /data/dalvik-cache 0771 root root
-    # create the A/B OTA directory, so as to enforce our permissions
-    mkdir /data/ota 0771 root root
-
-    # create the OTA package directory. It will be accessed by GmsCore (cache
-    # group), update_engine and update_verifier.
-    mkdir /data/ota_package 0770 system cache

     # create resource-cache and double-check the perms
     mkdir /data/resource-cache 0771 system system
@@ -567,20 +543,11 @@ on post-fs-data
     # the following directory.
     mkdir /data/mediadrm 0770 mediadrm mediadrm

-    mkdir /data/anr 0775 system system
-
-    # NFC: create data/nfc for nv storage
-    mkdir /data/nfc 0770 nfc nfc
-    mkdir /data/nfc/param 0770 nfc nfc
-
     # Create all remaining /data root dirs so that they are made through init
     # and get proper encryption policy installed
     mkdir /data/backup 0700 system system
-    mkdir /data/ss 0700 system system

     mkdir /data/system 0775 system system
-    mkdir /data/system/dropbox 0700 system system
-    mkdir /data/system/heapdump 0700 system system
     mkdir /data/system/users 0775 system system

     mkdir /data/system_de 0770 system system

project system/extras/
diff --git a/su/Android.mk b/su/Android.mk
index e3da4f21..5d431579 100644
--- a/su/Android.mk
+++ b/su/Android.mk
@@ -1,12 +1,32 @@
-LOCAL_PATH:= $(call my-dir)
-include $(CLEAR_VARS)
-
-LOCAL_CFLAGS := -Wall -Werror
+# Root AOSP source makefile
+# su is built here, and
+LOCAL_PATH := $(call my-dir)

-LOCAL_SRC_FILES:= su.cpp
+include $(CLEAR_VARS)

-LOCAL_MODULE:= su
+LOCAL_MODULE := su
+LOCAL_MODULE_TAGS := optional
+LOCAL_SHARED_LIBRARIES := \
+    libbinder \
+    libcutils \
+    liblog \
+    libutils \

+LOCAL_SRC_FILES := su.c daemon.c utils.c pts.c
 LOCAL_MODULE_PATH := $(TARGET_OUT_OPTIONAL_EXECUTABLES)

+LOCAL_INIT_RC := superuser.rc
+
 include $(BUILD_EXECUTABLE)
+
+SYMLINKS := $(addprefix $(TARGET_OUT)/bin/,su)
+$(SYMLINKS):
+	@echo "Symlink: $@ -> /system/xbin/su"
+	@mkdir -p $(dir $@)
+	@rm -rf $@
+	$(hide) ln -sf ../xbin/su $@
+
+# We need this so that the installed files could be picked up based on the
+# local module name
+ALL_MODULES.$(LOCAL_MODULE).INSTALLED := \
+    $(ALL_MODULES.$(LOCAL_MODULE).INSTALLED) $(SYMLINKS)

diff --git a/su/MODULE_LICENSE_APACHE2 b/su/MODULE_LICENSE_APACHE2
deleted file mode 100644
index e69de29b..00000000

diff --git a/su/NOTICE b/su/NOTICE
deleted file mode 100644
index c5b1efa7..00000000
--- a/su/NOTICE
+++ /dev/null

diff --git a/su/su.cpp b/su/su.cpp
deleted file mode 100644
index 1a1ab6bf..00000000
--- a/su/su.cpp
+++ /dev/null

project system/sepolicy/
diff --git a/Android.mk b/Android.mk
index dadd7b0a..1c922225 100644
--- a/Android.mk
+++ b/Android.mk
@@ -81,10 +81,12 @@ ifndef BOARD_SEPOLICY_VERS
 BOARD_SEPOLICY_VERS := $(PLATFORM_SEPOLICY_VERSION)
 endif

-NEVERALLOW_ARG :=
+# Hacks to keep SELinux permissive in user builds
+SELINUX_IGNORE_NEVERALLOWS := true
+
 ifeq ($(SELINUX_IGNORE_NEVERALLOWS),true)
 ifeq ($(TARGET_BUILD_VARIANT),user)
-$(error SELINUX_IGNORE_NEVERALLOWS := true cannot be used in user builds)
+$(warning SELINUX_IGNORE_NEVERALLOWS := true cannot be used in user builds)
 endif
 $(warning Be careful when using the SELINUX_IGNORE_NEVERALLOWS flag. \
           It does not work in user builds and using it will \
@@ -980,10 +982,9 @@ $(built_sepolicy_neverallows)
 	$(hide) $(HOST_OUT_EXECUTABLES)/sepolicy-analyze $@.tmp permissive > $@.permissivedomains
 	$(hide) if [ "$(TARGET_BUILD_VARIANT)" = "user" -a -s $@.permissivedomains ]; then \
 		echo "==========" 1>&2; \
-		echo "ERROR: permissive domains not allowed in user builds" 1>&2; \
+		echo "WARNING: permissive domains not allowed in user builds" 1>&2; \
 		echo "List of invalid domains:" 1>&2; \
 		cat $@.permissivedomains 1>&2; \
-		exit 1; \
 		fi
 	$(hide) mv $@.tmp $@

@@ -1034,10 +1035,9 @@ $(LOCAL_BUILT_MODULE): $(sepolicy.recovery.conf) $(HOST_OUT_EXECUTABLES)/checkpo
 	$(hide) $(HOST_OUT_EXECUTABLES)/sepolicy-analyze $@.tmp permissive > $@.permissivedomains
 	$(hide) if [ "$(TARGET_BUILD_VARIANT)" = "user" -a -s $@.permissivedomains ]; then \
 		echo "==========" 1>&2; \
-		echo "ERROR: permissive domains not allowed in user builds" 1>&2; \
+		echo "WARNING: permissive domains not allowed in user builds" 1>&2; \
 		echo "List of invalid domains:" 1>&2; \
 		cat $@.permissivedomains 1>&2; \
-		exit 1; \
 		fi
 	$(hide) mv $@.tmp $@

diff --git a/prebuilts/api/29.0/private/adbd.te b/prebuilts/api/29.0/private/adbd.te
index ec5c57ee..48325142 100644
--- a/prebuilts/api/29.0/private/adbd.te
+++ b/prebuilts/api/29.0/private/adbd.te
@@ -7,10 +7,10 @@ init_daemon_domain(adbd)

 domain_auto_trans(adbd, shell_exec, shell)

-userdebug_or_eng(`
+
   allow adbd self:process setcurrent;
   allow adbd su:process dyntransition;
-')
+

 # When 'adb shell' is executed in recovery mode, adbd explicitly
 # switches into shell domain using setcon() because the shell executable
@@ -191,4 +191,4 @@ allow adbd shell:fd use;
 # transitions to the shell domain (except when it crashes). In particular, we
 # never want to see a transition from adbd to su (aka "adb root")
 neverallow adbd { domain -crash_dump -shell }:process transition;
-neverallow adbd { domain userdebug_or_eng(`-su') recovery_only(`-shell') }:process dyntransition;
+neverallow adbd { domain -su -shell }:process dyntransition;

diff --git a/prebuilts/api/29.0/private/compat/26.0/26.0.ignore.cil b/prebuilts/api/29.0/private/compat/26.0/26.0.ignore.cil
index b12f917d..ee5efbbc 100644
--- a/prebuilts/api/29.0/private/compat/26.0/26.0.ignore.cil
+++ b/prebuilts/api/29.0/private/compat/26.0/26.0.ignore.cil
@@ -225,3 +225,6 @@
   ( priv_objects
     adbd_tmpfs
     untrusted_app_27_tmpfs))
+
+allow sudaemon untrusted_app_all_devpts:chr_file *;
+allow untrusted_app untrusted_app_all_devpts:chr_file *;

diff --git a/prebuilts/api/29.0/private/logd.te b/prebuilts/api/29.0/private/logd.te
index ca92e206..7e2d23d7 100644
--- a/prebuilts/api/29.0/private/logd.te
+++ b/prebuilts/api/29.0/private/logd.te
@@ -33,6 +33,6 @@ neverallow {
   -priv_app
   -radio
   -shell
-  userdebug_or_eng(`-su')
+  -su
   -system_app
 } runtime_event_log_tags_file:file no_rw_file_perms;

diff --git a/prebuilts/api/29.0/private/su.te b/prebuilts/api/29.0/private/su.te
index 16e47bbb..27a97bdb 100644
--- a/prebuilts/api/29.0/private/su.te
+++ b/prebuilts/api/29.0/private/su.te
@@ -1,4 +1,4 @@
-userdebug_or_eng(`
+
   typeattribute su coredomain;

   domain_auto_trans(shell, su_exec, su)
@@ -20,4 +20,4 @@ userdebug_or_eng(`
   permissive su;

   app_domain(su)
-')
+

diff --git a/prebuilts/api/29.0/private/untrusted_app_25.te b/prebuilts/api/29.0/private/untrusted_app_25.te
index a35d81bd..e1b5136d 100644
--- a/prebuilts/api/29.0/private/untrusted_app_25.te
+++ b/prebuilts/api/29.0/private/untrusted_app_25.te
@@ -41,6 +41,7 @@ allow untrusted_app_25 proc_misc:file r_file_perms;
 # https://github.com/strazzere/anti-emulator/blob/master/AntiEmulator/src/diff/strazzere/anti/emulator/FindEmulator.java
 # This will go away in a future Android release
 allow untrusted_app_25 proc_tty_drivers:file r_file_perms;
+allow untrusted_app socket_device:sock_file *;

 # Text relocation support for API < 23. This is now disallowed for targetSdkVersion>=Q.
 # https://android.googlesource.com/platform/bionic/+/master/android-changes-for-ndk-developers.md#text-relocations-enforced-for-api-level-23

diff --git a/prebuilts/api/29.0/private/untrusted_app_all.te b/prebuilts/api/29.0/private/untrusted_app_all.te
index 3c20c082..cf8111d1 100644
--- a/prebuilts/api/29.0/private/untrusted_app_all.te
+++ b/prebuilts/api/29.0/private/untrusted_app_all.te
@@ -168,6 +168,7 @@ dontaudit untrusted_app_all net_dns_prop:file read;
 dontaudit untrusted_app_all proc_stat:file read;
 dontaudit untrusted_app_all proc_vmstat:file read;
 dontaudit untrusted_app_all proc_uptime:file read;
+allow untrusted_app socket_device:sock_file *;

 # Allow the allocation and use of ptys
 # Used by: https://play.google.com/store/apps/details?id=jackpal.androidterm

diff --git a/prebuilts/api/29.0/public/app.te b/prebuilts/api/29.0/public/app.te
index 5c48e71f..0586f6d3 100644
--- a/prebuilts/api/29.0/public/app.te
+++ b/prebuilts/api/29.0/public/app.te
@@ -193,7 +193,7 @@ r_dir_file({
   -untrusted_app_all
 }, proc_net_type)
 # audit access for all these non-core app domains.
-userdebug_or_eng(`
+
   auditallow {
     appdomain
     -ephemeral_app
@@ -205,7 +205,7 @@ userdebug_or_eng(`
     -system_app
     -untrusted_app_all
   } proc_net_type:{ dir file lnk_file } { getattr open read };
-')
+

 # Grant GPU access to all processes started by Zygote.
 # They need that to render the standard UI.

diff --git a/prebuilts/api/29.0/public/domain.te b/prebuilts/api/29.0/public/domain.te
index f3487018..285dc30a 100644
--- a/prebuilts/api/29.0/public/domain.te
+++ b/prebuilts/api/29.0/public/domain.te
@@ -32,7 +32,6 @@ allow domain self:unix_stream_socket { create_stream_socket_perms connectto };
 # Inherit or receive open files from others.
 allow domain init:fd use;

-userdebug_or_eng(`
   allow domain su:fd use;
   allow domain su:unix_stream_socket { connectto getattr getopt read write shutdown };
   allow domain su:unix_dgram_socket sendto;
@@ -46,6 +45,7 @@ userdebug_or_eng(`
   # allow "gdbserver --attach" to work for su.
   allow domain su:process sigchld;

+userdebug_or_eng(`
   # Allow writing coredumps to /cores/*
   allow domain coredump_file:file create_file_perms;
   allow domain coredump_file:dir ra_dir_perms;
@@ -602,7 +602,7 @@ neverallow {
 # this partition for testing purposes.
 neverallow {
   domain
-  userdebug_or_eng(`-domain') # exclude debuggable builds
+  -domain # exclude debuggable builds
   -fastbootd
   -hal_bootctl_server
   -init
@@ -688,7 +688,7 @@ full_treble_only(`
   neverallow {
     coredomain
     -shell
-    userdebug_or_eng(`-su')
+    -su
     -ueventd # uevent is granted create for this device, but we still neverallow I/O below
   } vndbinder_device:chr_file rw_file_perms;
 ')
@@ -699,14 +699,14 @@ full_treble_only(`
   neverallow {
     coredomain
     -shell
-    userdebug_or_eng(`-su')
+    -su
   } vndservice_manager_type:service_manager *;
 ')
 full_treble_only(`
   neverallow {
     coredomain
     -shell
-    userdebug_or_eng(`-su')
+    -su
   } vndservicemanager:binder *;
 ')

@@ -742,7 +742,7 @@ full_treble_only(`
     -logd # Logging by writing to logd Unix domain socket is public API
     -netd # netdomain needs this
     -mdnsd # netdomain needs this
-    userdebug_or_eng(`-su') # communications with su are permitted only on userdebug or eng builds
+    -su # communications with su are permitted only on userdebug or eng builds
     -init
     -tombstoned # linker to tombstoned
     userdebug_or_eng('-heapprofd`)
@@ -1108,7 +1108,7 @@ neverallow * { file_type fs_type dev_type }:{ lnk_file fifo_file sock_file } mou
 # Nobody should be able to execute su on user builds.
 # On userdebug/eng builds, only dumpstate, shell, and
 # su itself execute su.
-neverallow { domain userdebug_or_eng(`-dumpstate -shell -su') } su_exec:file no_x_file_perms;
+neverallow { domain -dumpstate -shell -su } su_exec:file no_x_file_perms;

 # Do not allow the introduction of new execmod rules. Text relocations
 # and modification of executable pages are unsafe.
@@ -1196,7 +1196,7 @@ neverallow {
 # Only domains spawned from zygote, runas and simpleperf_app_runner may have the appdomain
 # attribute.
 neverallow { domain -simpleperf_app_runner -runas -app_zygote -webview_zygote -zygote } {
-  appdomain -shell userdebug_or_eng(`-su')
+  appdomain -shell -su
 }:process { transition dyntransition };

 # Minimize read access to shell- or app-writable symlinks.

diff --git a/prebuilts/api/29.0/public/hal_configstore.te b/prebuilts/api/29.0/public/hal_configstore.te
index 1a95b72f..edcdc267 100644
--- a/prebuilts/api/29.0/public/hal_configstore.te
+++ b/prebuilts/api/29.0/public/hal_configstore.te
@@ -31,7 +31,7 @@ neverallow hal_configstore_server {
   domain
   -hal_configstore_server
   -logd
-  userdebug_or_eng(`-su')
+  -su
   -tombstoned
   userdebug_or_eng(`-heapprofd')
 }:{ unix_dgram_socket unix_stream_socket } *;

diff --git a/prebuilts/api/29.0/public/installd.te b/prebuilts/api/29.0/public/installd.te
index f0ac52a0..eb78d9e6 100644
--- a/prebuilts/api/29.0/public/installd.te
+++ b/prebuilts/api/29.0/public/installd.te
@@ -173,5 +173,5 @@ neverallow installd {
     -ashmemd
     -system_server
     -servicemanager
-    userdebug_or_eng(`-su')
+    -su
 }:binder call;

diff --git a/prebuilts/api/29.0/public/netd.te b/prebuilts/api/29.0/public/netd.te
index c8877b24..540e04e9 100644
--- a/prebuilts/api/29.0/public/netd.te
+++ b/prebuilts/api/29.0/public/netd.te
@@ -158,7 +158,7 @@ neverallow {

 # apps may not interact with netd over binder.
 neverallow { appdomain -network_stack } netd:binder call;
-neverallow netd { appdomain -network_stack userdebug_or_eng(`-su') }:binder call;
+neverallow netd { appdomain -network_stack -su }:binder call;

 # persist.netd.stable_secret contains RFC 7217 secret key which should never be
 # leaked to other processes. Make sure it never leaks.

diff --git a/prebuilts/api/29.0/public/su.te b/prebuilts/api/29.0/public/su.te
index a2f435e1..b0ec6533 100644
--- a/prebuilts/api/29.0/public/su.te
+++ b/prebuilts/api/29.0/public/su.te
@@ -4,8 +4,8 @@ type su, domain;

 # File types must be defined for file_contexts.
 type su_exec, system_file_type, exec_type, file_type;
+type sudaemon, domain;

-userdebug_or_eng(`
   # Domain used for su processes, as well as for adbd and adb shell
   # after performing an adb root command.  The domain definition is
   # wrapped to ensure that it does not exist at all on -user builds.
@@ -101,4 +101,4 @@ userdebug_or_eng(`
   typeattribute su hal_wifi_hostapd_client;
   typeattribute su hal_wifi_offload_client;
   typeattribute su hal_wifi_supplicant_client;
-')
+

diff --git a/prebuilts/api/29.0/public/te_macros b/prebuilts/api/29.0/public/te_macros
index 85783dc9..f84c9c03 100644
--- a/prebuilts/api/29.0/public/te_macros
+++ b/prebuilts/api/29.0/public/te_macros
@@ -98,10 +98,10 @@ define(`pdx_service_socket_types', `
 typeattribute $2 pdx_$1_endpoint_dir_type;
 type pdx_$1_endpoint_socket, pdx_$1_endpoint_socket_type, pdx_endpoint_socket_type, file_type, coredomain_socket, mlstrustedobject, mlstrustedsubject;
 type pdx_$1_channel_socket, pdx_$1_channel_socket_type, pdx_channel_socket_type, coredomain_socket;
-userdebug_or_eng(`
+
 dontaudit su pdx_$1_endpoint_socket:unix_stream_socket *;
 dontaudit su pdx_$1_channel_socket:unix_stream_socket *;
-')
+
 ')

 #####################################
@@ -525,9 +525,9 @@ define(`build_test_only', ifelse(target_exclude_build_test, `true', , $1))
 # Fallback crash handling for processes that can't exec crash_dump (e.g. because of seccomp).
 #
 define(`crash_dump_fallback', `
-userdebug_or_eng(`
+
   allow $1 su:fifo_file append;
-')
+
 allow $1 anr_data_file:file append;
 allow $1 dumpstate:fd use;
 allow $1 incidentd:fd use;

diff --git a/prebuilts/api/29.0/public/vold.te b/prebuilts/api/29.0/public/vold.te
index 49523f43..0742ba3f 100644
--- a/prebuilts/api/29.0/public/vold.te
+++ b/prebuilts/api/29.0/public/vold.te
@@ -315,7 +315,7 @@ neverallow vold {
   -iorapd_service
   -servicemanager
   -system_server
-  userdebug_or_eng(`-su')
+  -su
 }:binder call;

 neverallow vold fsck_exec:file execute_no_trans;

diff --git a/private/adbd.te b/private/adbd.te
index ec5c57ee..48325142 100644
--- a/private/adbd.te
+++ b/private/adbd.te
@@ -7,10 +7,10 @@ init_daemon_domain(adbd)

 domain_auto_trans(adbd, shell_exec, shell)

-userdebug_or_eng(`
+
   allow adbd self:process setcurrent;
   allow adbd su:process dyntransition;
-')
+

 # When 'adb shell' is executed in recovery mode, adbd explicitly
 # switches into shell domain using setcon() because the shell executable
@@ -191,4 +191,4 @@ allow adbd shell:fd use;
 # transitions to the shell domain (except when it crashes). In particular, we
 # never want to see a transition from adbd to su (aka "adb root")
 neverallow adbd { domain -crash_dump -shell }:process transition;
-neverallow adbd { domain userdebug_or_eng(`-su') recovery_only(`-shell') }:process dyntransition;
+neverallow adbd { domain -su -shell }:process dyntransition;

diff --git a/private/compat/26.0/26.0.ignore.cil b/private/compat/26.0/26.0.ignore.cil
index b12f917d..ee5efbbc 100644
--- a/private/compat/26.0/26.0.ignore.cil
+++ b/private/compat/26.0/26.0.ignore.cil
@@ -225,3 +225,6 @@
   ( priv_objects
     adbd_tmpfs
     untrusted_app_27_tmpfs))
+
+allow sudaemon untrusted_app_all_devpts:chr_file *;
+allow untrusted_app untrusted_app_all_devpts:chr_file *;

diff --git a/private/logd.te b/private/logd.te
index ca92e206..7e2d23d7 100644
--- a/private/logd.te
+++ b/private/logd.te
@@ -33,6 +33,6 @@ neverallow {
   -priv_app
   -radio
   -shell
-  userdebug_or_eng(`-su')
+  -su
   -system_app
 } runtime_event_log_tags_file:file no_rw_file_perms;

diff --git a/private/su.te b/private/su.te
index 16e47bbb..27a97bdb 100644
--- a/private/su.te
+++ b/private/su.te
@@ -1,4 +1,4 @@
-userdebug_or_eng(`
+
   typeattribute su coredomain;

   domain_auto_trans(shell, su_exec, su)
@@ -20,4 +20,4 @@ userdebug_or_eng(`
   permissive su;

   app_domain(su)
-')
+

diff --git a/private/untrusted_app_25.te b/private/untrusted_app_25.te
index a35d81bd..e1b5136d 100644
--- a/private/untrusted_app_25.te
+++ b/private/untrusted_app_25.te
@@ -41,6 +41,7 @@ allow untrusted_app_25 proc_misc:file r_file_perms;
 # https://github.com/strazzere/anti-emulator/blob/master/AntiEmulator/src/diff/strazzere/anti/emulator/FindEmulator.java
 # This will go away in a future Android release
 allow untrusted_app_25 proc_tty_drivers:file r_file_perms;
+allow untrusted_app socket_device:sock_file *;

 # Text relocation support for API < 23. This is now disallowed for targetSdkVersion>=Q.
 # https://android.googlesource.com/platform/bionic/+/master/android-changes-for-ndk-developers.md#text-relocations-enforced-for-api-level-23

diff --git a/private/untrusted_app_all.te b/private/untrusted_app_all.te
index 3c20c082..cf8111d1 100644
--- a/private/untrusted_app_all.te
+++ b/private/untrusted_app_all.te
@@ -168,6 +168,7 @@ dontaudit untrusted_app_all net_dns_prop:file read;
 dontaudit untrusted_app_all proc_stat:file read;
 dontaudit untrusted_app_all proc_vmstat:file read;
 dontaudit untrusted_app_all proc_uptime:file read;
+allow untrusted_app socket_device:sock_file *;

 # Allow the allocation and use of ptys
 # Used by: https://play.google.com/store/apps/details?id=jackpal.androidterm

diff --git a/public/app.te b/public/app.te
index 5c48e71f..0586f6d3 100644
--- a/public/app.te
+++ b/public/app.te
@@ -193,7 +193,7 @@ r_dir_file({
   -untrusted_app_all
 }, proc_net_type)
 # audit access for all these non-core app domains.
-userdebug_or_eng(`
+
   auditallow {
     appdomain
     -ephemeral_app
@@ -205,7 +205,7 @@ userdebug_or_eng(`
     -system_app
     -untrusted_app_all
   } proc_net_type:{ dir file lnk_file } { getattr open read };
-')
+

 # Grant GPU access to all processes started by Zygote.
 # They need that to render the standard UI.

diff --git a/public/domain.te b/public/domain.te
index f3487018..285dc30a 100644
--- a/public/domain.te
+++ b/public/domain.te
@@ -32,7 +32,6 @@ allow domain self:unix_stream_socket { create_stream_socket_perms connectto };
 # Inherit or receive open files from others.
 allow domain init:fd use;

-userdebug_or_eng(`
   allow domain su:fd use;
   allow domain su:unix_stream_socket { connectto getattr getopt read write shutdown };
   allow domain su:unix_dgram_socket sendto;
@@ -46,6 +45,7 @@ userdebug_or_eng(`
   # allow "gdbserver --attach" to work for su.
   allow domain su:process sigchld;

+userdebug_or_eng(`
   # Allow writing coredumps to /cores/*
   allow domain coredump_file:file create_file_perms;
   allow domain coredump_file:dir ra_dir_perms;
@@ -602,7 +602,7 @@ neverallow {
 # this partition for testing purposes.
 neverallow {
   domain
-  userdebug_or_eng(`-domain') # exclude debuggable builds
+  -domain # exclude debuggable builds
   -fastbootd
   -hal_bootctl_server
   -init
@@ -688,7 +688,7 @@ full_treble_only(`
   neverallow {
     coredomain
     -shell
-    userdebug_or_eng(`-su')
+    -su
     -ueventd # uevent is granted create for this device, but we still neverallow I/O below
   } vndbinder_device:chr_file rw_file_perms;
 ')
@@ -699,14 +699,14 @@ full_treble_only(`
   neverallow {
     coredomain
     -shell
-    userdebug_or_eng(`-su')
+    -su
   } vndservice_manager_type:service_manager *;
 ')
 full_treble_only(`
   neverallow {
     coredomain
     -shell
-    userdebug_or_eng(`-su')
+    -su
   } vndservicemanager:binder *;
 ')

@@ -742,7 +742,7 @@ full_treble_only(`
     -logd # Logging by writing to logd Unix domain socket is public API
     -netd # netdomain needs this
     -mdnsd # netdomain needs this
-    userdebug_or_eng(`-su') # communications with su are permitted only on userdebug or eng builds
+    -su # communications with su are permitted only on userdebug or eng builds
     -init
     -tombstoned # linker to tombstoned
     userdebug_or_eng('-heapprofd`)
@@ -1108,7 +1108,7 @@ neverallow * { file_type fs_type dev_type }:{ lnk_file fifo_file sock_file } mou
 # Nobody should be able to execute su on user builds.
 # On userdebug/eng builds, only dumpstate, shell, and
 # su itself execute su.
-neverallow { domain userdebug_or_eng(`-dumpstate -shell -su') } su_exec:file no_x_file_perms;
+neverallow { domain -dumpstate -shell -su } su_exec:file no_x_file_perms;

 # Do not allow the introduction of new execmod rules. Text relocations
 # and modification of executable pages are unsafe.
@@ -1196,7 +1196,7 @@ neverallow {
 # Only domains spawned from zygote, runas and simpleperf_app_runner may have the appdomain
 # attribute.
 neverallow { domain -simpleperf_app_runner -runas -app_zygote -webview_zygote -zygote } {
-  appdomain -shell userdebug_or_eng(`-su')
+  appdomain -shell -su
 }:process { transition dyntransition };

 # Minimize read access to shell- or app-writable symlinks.

diff --git a/public/hal_configstore.te b/public/hal_configstore.te
index 1a95b72f..edcdc267 100644
--- a/public/hal_configstore.te
+++ b/public/hal_configstore.te
@@ -31,7 +31,7 @@ neverallow hal_configstore_server {
   domain
   -hal_configstore_server
   -logd
-  userdebug_or_eng(`-su')
+  -su
   -tombstoned
   userdebug_or_eng(`-heapprofd')
 }:{ unix_dgram_socket unix_stream_socket } *;

diff --git a/public/installd.te b/public/installd.te
index f0ac52a0..eb78d9e6 100644
--- a/public/installd.te
+++ b/public/installd.te
@@ -173,5 +173,5 @@ neverallow installd {
     -ashmemd
     -system_server
     -servicemanager
-    userdebug_or_eng(`-su')
+    -su
 }:binder call;

diff --git a/public/netd.te b/public/netd.te
index c8877b24..540e04e9 100644
--- a/public/netd.te
+++ b/public/netd.te
@@ -158,7 +158,7 @@ neverallow {

 # apps may not interact with netd over binder.
 neverallow { appdomain -network_stack } netd:binder call;
-neverallow netd { appdomain -network_stack userdebug_or_eng(`-su') }:binder call;
+neverallow netd { appdomain -network_stack -su }:binder call;

 # persist.netd.stable_secret contains RFC 7217 secret key which should never be
 # leaked to other processes. Make sure it never leaks.

diff --git a/public/su.te b/public/su.te
index a2f435e1..b0ec6533 100644
--- a/public/su.te
+++ b/public/su.te
@@ -4,8 +4,8 @@ type su, domain;

 # File types must be defined for file_contexts.
 type su_exec, system_file_type, exec_type, file_type;
+type sudaemon, domain;

-userdebug_or_eng(`
   # Domain used for su processes, as well as for adbd and adb shell
   # after performing an adb root command.  The domain definition is
   # wrapped to ensure that it does not exist at all on -user builds.
@@ -101,4 +101,4 @@ userdebug_or_eng(`
   typeattribute su hal_wifi_hostapd_client;
   typeattribute su hal_wifi_offload_client;
   typeattribute su hal_wifi_supplicant_client;
-')
+

diff --git a/public/te_macros b/public/te_macros
index 85783dc9..f84c9c03 100644
--- a/public/te_macros
+++ b/public/te_macros
@@ -98,10 +98,10 @@ define(`pdx_service_socket_types', `
 typeattribute $2 pdx_$1_endpoint_dir_type;
 type pdx_$1_endpoint_socket, pdx_$1_endpoint_socket_type, pdx_endpoint_socket_type, file_type, coredomain_socket, mlstrustedobject, mlstrustedsubject;
 type pdx_$1_channel_socket, pdx_$1_channel_socket_type, pdx_channel_socket_type, coredomain_socket;
-userdebug_or_eng(`
+
 dontaudit su pdx_$1_endpoint_socket:unix_stream_socket *;
 dontaudit su pdx_$1_channel_socket:unix_stream_socket *;
-')
+
 ')

 #####################################
@@ -525,9 +525,9 @@ define(`build_test_only', ifelse(target_exclude_build_test, `true', , $1))
 # Fallback crash handling for processes that can't exec crash_dump (e.g. because of seccomp).
 #
 define(`crash_dump_fallback', `
-userdebug_or_eng(`
+
   allow $1 su:fifo_file append;
-')
+
 allow $1 anr_data_file:file append;
 allow $1 dumpstate:fd use;
 allow $1 incidentd:fd use;

diff --git a/public/vold.te b/public/vold.te
index 49523f43..0742ba3f 100644
--- a/public/vold.te
+++ b/public/vold.te
@@ -315,7 +315,7 @@ neverallow vold {
   -iorapd_service
   -servicemanager
   -system_server
-  userdebug_or_eng(`-su')
+  -su
 }:binder call;

 neverallow vold fsck_exec:file execute_no_trans;

project vendor/lineage/
diff --git a/bootanimation/Android.mk b/bootanimation/Android.mk
deleted file mode 100644
index 2bc78ceb..00000000
--- a/bootanimation/Android.mk
+++ /dev/null

diff --git a/config/common.mk b/config/common.mk
index 9b688031..eb4395d4 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -47,6 +47,18 @@ PRODUCT_COPY_FILES += \
 PRODUCT_COPY_FILES += \
     vendor/lineage/config/permissions/lineage-sysconfig.xml:$(TARGET_COPY_OUT_SYSTEM)/etc/sysconfig/lineage-sysconfig.xml

+# init.d support
+PRODUCT_COPY_FILES += \
+    vendor/lineage/prebuilt/common/etc/init.d/00banner:system/etc/init.d/00banner \
+    vendor/lineage/prebuilt/common/etc/init.d/01boot:system/etc/init.d/01boot \
+    vendor/lineage/prebuilt/common/bin/sysinit:system/bin/sysinit
+
+ifneq ($(TARGET_BUILD_VARIANT),user)
+# userinit support
+PRODUCT_COPY_FILES += \
+    vendor/lineage/prebuilt/common/etc/init.d/90userinit:system/etc/init.d/90userinit
+endif
+
 # Copy all Lineage-specific init rc files
 $(foreach f,$(wildcard vendor/lineage/prebuilt/common/etc/init/*.rc),\
 	$(eval PRODUCT_COPY_FILES += $(f):$(TARGET_COPY_OUT_SYSTEM)/etc/init/$(notdir $f)))
@@ -231,14 +243,9 @@ PRODUCT_PACKAGES_DEBUG += \
     procmem

 # Conditionally build in su
-ifneq ($(TARGET_BUILD_VARIANT),user)
-PRODUCT_PACKAGES += \
-    adb_root
-ifeq ($(WITH_SU),true)
 PRODUCT_PACKAGES += \
+    adb_root \
     su
-endif
-endif

 # Dex preopt
 PRODUCT_DEXPREOPT_SPEED_APPS += \

diff --git a/overlay/common/frameworks/base/packages/SystemUI/res/values/config.xml b/overlay/common/frameworks/base/packages/SystemUI/res/values/config.xml
index 6dc72ff1..b26117e1 100644
--- a/overlay/common/frameworks/base/packages/SystemUI/res/values/config.xml
+++ b/overlay/common/frameworks/base/packages/SystemUI/res/values/config.xml
@@ -27,5 +27,5 @@
     <!-- If true, enable the advance anti-falsing classifier on the lockscreen. On some devices it
          does not work well, particularly with noisy touchscreens. Note that disabling it may
          increase the rate of unintentional unlocks. -->
-    <bool name="config_lockscreenAntiFalsingClassifierEnabled">false</bool>
+    <bool name="config_lockscreenAntiFalsingClassifierEnabled">true</bool>
 </resources>

diff --git a/overlay/common/packages/apps/Settings/res/values/config.xml b/overlay/common/packages/apps/Settings/res/values/config.xml
index 651905da..5c815424 100644
--- a/overlay/common/packages/apps/Settings/res/values/config.xml
+++ b/overlay/common/packages/apps/Settings/res/values/config.xml
@@ -22,10 +22,10 @@
     <bool name="config_storage_manager_settings_enabled">true</bool>

     <!-- Whether to show a preference item for mobile plan -->
-    <bool name="config_show_mobile_plan">false</bool>
+    <bool name="config_show_mobile_plan">true</bool>

     <!-- Whether wallpaper attribution should be shown or not. -->
-    <bool name="config_show_wallpaper_attribution">false</bool>
+    <bool name="config_show_wallpaper_attribution">true</bool>

     <!-- Package name and fully-qualified class name for the wallpaper picker activity. -->
     <string name="config_wallpaper_picker_package" translatable="false">com.android.wallpaper</string>

diff --git a/prebuilt/common/bin/backuptool.sh b/prebuilt/common/bin/backuptool.sh
index 662b4673..5da27733 100755
--- a/prebuilt/common/bin/backuptool.sh
+++ b/prebuilt/common/bin/backuptool.sh
@@ -97,6 +97,8 @@ case "$1" in
       run_stage restore
       run_stage post-restore
       restore_addon_d
+      chmod 755 $S/xbin/su
+      chmod 755 $S/etc/init.d/*
       rm -rf $C
       sync
     fi

diff --git a/prebuilt/common/etc/init/lineage-ssh.rc b/prebuilt/common/etc/init/lineage-ssh.rc
deleted file mode 100644
index edf38638..00000000
--- a/prebuilt/common/etc/init/lineage-ssh.rc
+++ /dev/null

diff --git a/prebuilt/common/etc/init/lineage-system.rc b/prebuilt/common/etc/init/lineage-system.rc
index 396ec73a..31cac49f 100644
--- a/prebuilt/common/etc/init/lineage-system.rc
+++ b/prebuilt/common/etc/init/lineage-system.rc
@@ -6,6 +6,9 @@ on init
 on post-fs-data
     mkdir /cache/recovery 0770 system cache

+    # Run sysinit
+    start sysinit
+
     # Change permissions on fsck log so it can be added to the dropbox
     chown root log /dev/fscklogs/log
     chmod 0640 /dev/fscklogs/log
@@ -14,9 +17,8 @@ on boot
     # Persistent properties (only created if persist exists)
     mkdir /persist/properties 0770 system system

-# bugreport is triggered by holding down volume down, volume up and power
-service bugreport /system/bin/dumpstate -d -p -B -z \
-        -o /data/user_de/0/com.android.shell/files/bugreports/bugreport
+# sysinit (/system/etc/init.d)
+service sysinit /system/bin/sysinit
+    user root
     oneshot
     disabled
-    keycodes 114 115 116

diff --git a/prebuilt/common/etc/init/lineage-updates.rc b/prebuilt/common/etc/init/lineage-updates.rc
deleted file mode 100644
index 3a9b4813..00000000
--- a/prebuilt/common/etc/init/lineage-updates.rc
+++ /dev/null
