
project build/make/
diff --git a/core/binary.mk b/core/binary.mk
index 712f2df56..2fc96b536 100644
--- a/core/binary.mk
+++ b/core/binary.mk
@@ -613,7 +613,7 @@ endif
 
 # Turn on all warnings and warnings as errors for RS compiles.
 # This can be disabled with LOCAL_RENDERSCRIPT_FLAGS := -Wno-error
-renderscript_flags := -Wall -Werror
+renderscript_flags := -Wall
 renderscript_flags += $(LOCAL_RENDERSCRIPT_FLAGS)
 # -m32 or -m64
 renderscript_flags += -m$(my_32_64_bit_suffix)
@@ -1450,7 +1450,7 @@ my_asflags := $(call convert-to-clang-flags,$(my_asflags))
 my_ldflags := $(call convert-to-clang-flags,$(my_ldflags))
 
 ifeq ($(my_fdo_build), true)
-  my_cflags := $(patsubst -Os,-O2,$(my_cflags))
+  my_cflags := $(patsubst -O3,-g0,$(my_cflags))
   fdo_incompatible_flags := -fno-early-inlining -finline-limit=%
   my_cflags := $(filter-out $(fdo_incompatible_flags),$(my_cflags))
 endif
@@ -1490,10 +1490,10 @@ ifneq (HEADER_LIBRARIES,$(LOCAL_MODULE_CLASS))
       $(eval MODULES_USING_WNO_ERROR := $(MODULES_USING_WNO_ERROR) $(LOCAL_MODULE_MAKEFILE):$(LOCAL_MODULE))
     else
       # Issue warning if -Werror is not used. Add it.
-      ifeq (,$(filter -Werror,$(my_all_cflags)))
+      ifeq (,$(filter -Wall,$(my_all_cflags)))
         # Add -Wall -Werror unless the project is in the WARNING_ALLOWED project list.
         ifeq (,$(strip $(call find_warning_allowed_projects,$(LOCAL_PATH))))
-          my_cflags := -Wall -Werror $(my_cflags)
+          my_cflags := -Wall $(my_cflags)
         else
           $(eval MODULES_ADDED_WALL := $(MODULES_ADDED_WALL) $(LOCAL_MODULE_MAKEFILE):$(LOCAL_MODULE))
           my_cflags := -Wall $(my_cflags)
@@ -1672,7 +1672,7 @@ ifeq ($(NATIVE_COVERAGE),true)
         # with `--gcov-tool /usr/bin/gcov-4.6`.
         #
         # http://stackoverflow.com/questions/17758126/clang-code-coverage-invalid-output
-        my_cflags += --coverage -O0
+        my_cflags += --coverage -O3 -g0
         my_ldflags += --coverage
     endif
 
diff --git a/core/host_test_internal.mk b/core/host_test_internal.mk
index dfe8cf150..35bb766c6 100644
--- a/core/host_test_internal.mk
+++ b/core/host_test_internal.mk
@@ -6,7 +6,7 @@ ifeq ($(LOCAL_GTEST),true)
   LOCAL_CFLAGS_linux += -DGTEST_OS_LINUX
   LOCAL_CFLAGS_darwin += -DGTEST_OS_MAC
 
-  LOCAL_CFLAGS += -DGTEST_HAS_STD_STRING -O0 -g
+  LOCAL_CFLAGS += -DGTEST_HAS_STD_STRING -O3 -g0
 
   LOCAL_STATIC_LIBRARIES += libgtest_main_host libgtest_host
 endif
diff --git a/core/java_renderscript.mk b/core/java_renderscript.mk
index 13a6f8e66..72801a92d 100644
--- a/core/java_renderscript.mk
+++ b/core/java_renderscript.mk
@@ -46,7 +46,7 @@ endif
 
 # Turn on all warnings and warnings as errors for RS compiles.
 # This can be disabled with LOCAL_RENDERSCRIPT_FLAGS := -Wno-error
-renderscript_flags := -Wall -Werror
+renderscript_flags := -Wall
 renderscript_flags += $(LOCAL_RENDERSCRIPT_FLAGS)
 
 # prepend the RenderScript system include path
diff --git a/core/main.mk b/core/main.mk
index d1bd43bba..901f8e978 100644
--- a/core/main.mk
+++ b/core/main.mk
@@ -290,8 +290,6 @@ ifneq (,$(user_variant))
   ADDITIONAL_DEFAULT_PROPERTIES += ro.allow.mock.location=0
 
 else # !user_variant
-  # Turn on checkjni for non-user builds.
-  ADDITIONAL_BUILD_PROPERTIES += ro.kernel.android.checkjni=1
   # Set device insecure for non-user builds.
   ADDITIONAL_DEFAULT_PROPERTIES += ro.secure=0
   # Allow mock locations by default for non user builds
@@ -301,8 +299,6 @@ endif # !user_variant
 ifeq (true,$(strip $(enable_target_debugging)))
   # Target is more debuggable and adbd is on by default
   ADDITIONAL_DEFAULT_PROPERTIES += ro.debuggable=1
-  # Enable Dalvik lock contention logging.
-  ADDITIONAL_BUILD_PROPERTIES += dalvik.vm.lockprof.threshold=500
 else # !enable_target_debugging
   # Target is less debuggable and adbd is off by default
   ADDITIONAL_DEFAULT_PROPERTIES += ro.debuggable=0
@@ -320,7 +316,7 @@ ifneq ($(filter ro.setupwizard.mode=ENABLED, $(call collapse-pairs, $(ADDITIONAL
 endif
 ifndef is_sdk_build
   # To speedup startup of non-preopted builds, don't verify or compile the boot image.
-  ADDITIONAL_BUILD_PROPERTIES += dalvik.vm.image-dex2oat-filter=verify-at-runtime
+  ADDITIONAL_BUILD_PROPERTIES += dalvik.vm.image-dex2oat-filter=speed
 endif
 endif
 
diff --git a/core/product.mk b/core/product.mk
index b45d802d0..5ba7730f5 100644
--- a/core/product.mk
+++ b/core/product.mk
@@ -491,10 +491,16 @@ _readonly_late_variables := \
 
 # Modified internally in the build system
 _readonly_late_variables += \
+  PRODUCT_ART_TARGET_INCLUDE_DEBUG_BUILD \
   PRODUCT_CFI_INCLUDE_PATHS \
   PRODUCT_COPY_FILES \
   PRODUCT_DEX_PREOPT_NEVER_ALLOW_STRIPPING \
   PRODUCT_DEX_PREOPT_BOOT_FLAGS \
+  PRODUCT_DEX_PREOPT_DEFAULT_FLAGS \
+  PRODUCT_DEX_PREOPT_DEFAULT_COMPILER_FILTER \
+  PRODUCT_SYSTEM_SERVER_COMPILER_FILTER \
+  PRODUCT_SYSTEM_SERVER_DEBUG_INFO \
+  PRODUCT_OTHER_JAVA_DEBUG_INFO \
   PRODUCT_SOONG_NAMESPACES
 
 _readonly_early_variables := $(filter-out $(_readonly_late_variables),$(_product_var_list))
diff --git a/target/product/media_system.mk b/target/product/media_system.mk
index 0489b28d9..d29350b96 100644
--- a/target/product/media_system.mk
+++ b/target/product/media_system.mk
@@ -68,16 +68,6 @@ PRODUCT_COPY_FILES += $(call add-to-product-copy-files-if-exists,\
 PRODUCT_COPY_FILES += $(call add-to-product-copy-files-if-exists,\
     frameworks/base/dirty-image-objects-phone:system/etc/dirty-image-objects)
 
-# On userdebug builds, collect more tombstones by default.
-ifneq (,$(filter userdebug eng,$(TARGET_BUILD_VARIANT)))
-PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
-    tombstoned.max_tombstone_count=50
-endif
-
-PRODUCT_DEFAULT_PROPERTY_OVERRIDES += \
-    ro.logd.size.stats=64K \
-    log.tag.stats_log=I
-
 # Enable CFI for security-sensitive components
 $(call inherit-product, $(SRC_TARGET_DIR)/product/cfi-common.mk)
 $(call inherit-product-if-exists, vendor/google/products/cfi-vendor.mk)
diff --git a/target/product/runtime_libart.mk b/target/product/runtime_libart.mk
index a88ba3c8d..ed19be911 100644
--- a/target/product/runtime_libart.mk
+++ b/target/product/runtime_libart.mk
@@ -61,22 +61,22 @@ PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
 # On eng builds, make "boot" reasons only extract for faster turnaround.
 ifeq (eng,$(TARGET_BUILD_VARIANT))
     PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
-        pm.dexopt.first-boot=extract \
-        pm.dexopt.boot=extract
+        pm.dexopt.first-boot=speed \
+        pm.dexopt.boot=speed
 else
     PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
-        pm.dexopt.first-boot=quicken \
-        pm.dexopt.boot=verify
+        pm.dexopt.first-boot=speed \
+        pm.dexopt.boot=speed
 endif
 
 # The install filter is speed-profile in order to enable the use of
 # profiles from the dex metadata files. Note that if a profile is not provided
 # or if it is empty speed-profile is equivalent to (quicken + empty app image).
 PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
-    pm.dexopt.install=speed-profile \
-    pm.dexopt.bg-dexopt=speed-profile \
-    pm.dexopt.ab-ota=speed-profile \
-    pm.dexopt.inactive=verify \
+    pm.dexopt.install=speed \
+    pm.dexopt.bg-dexopt=speed \
+    pm.dexopt.ab-ota=speed \
+    pm.dexopt.inactive=speed \
     pm.dexopt.shared=speed
 
 # Enable resolution of startup const strings.
@@ -87,11 +87,6 @@ PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
 PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
     dalvik.vm.dex2oat-max-image-block-size=524288
 
-# Enable minidebuginfo generation unless overridden.
-PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
-    dalvik.vm.minidebuginfo=true \
-    dalvik.vm.dex2oat-minidebuginfo=true
-
 # Disable iorapd by default
 PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
     ro.iorapd.enable=false
diff --git a/tools/acp/Android.bp b/tools/acp/Android.bp
index 64f5a1013..8b84418b3 100644
--- a/tools/acp/Android.bp
+++ b/tools/acp/Android.bp
@@ -5,7 +5,7 @@
 cc_binary_host {
 
     srcs: ["acp.c"],
-    cflags: ["-Wall", "-Werror"],
+    cflags: ["-Wall"],
 
     static_libs: ["libhost"],
     name: "acp",
diff --git a/tools/atree/Android.bp b/tools/atree/Android.bp
index 5fbe042ea..fc8a8aada 100644
--- a/tools/atree/Android.bp
+++ b/tools/atree/Android.bp
@@ -9,6 +9,6 @@ cc_binary_host {
         "files.cpp",
         "fs.cpp",
     ],
-    cflags: ["-Wall", "-Werror"],
+    cflags: ["-Wall"],
     static_libs: ["libhost"],
 }
diff --git a/tools/fs_config/Android.bp b/tools/fs_config/Android.bp
index d9a48d704..60bfe673f 100644
--- a/tools/fs_config/Android.bp
+++ b/tools/fs_config/Android.bp
@@ -32,7 +32,6 @@ cc_binary_host {
         "libcutils",
         "libselinux",
     ],
-    cflags: ["-Werror"],
 }
 
 target_fs_config_gen_filegroup {
diff --git a/tools/fs_get_stats/Android.bp b/tools/fs_get_stats/Android.bp
index 67742b836..f710d506e 100644
--- a/tools/fs_get_stats/Android.bp
+++ b/tools/fs_get_stats/Android.bp
@@ -1,7 +1,7 @@
 cc_binary_host {
     name: "fs_get_stats",
     srcs: ["fs_get_stats.c"],
-    cflags: ["-Wall", "-Werror"],
+    cflags: ["-Wall"],
     shared_libs: [
         "libcutils",
         "liblog",
diff --git a/tools/libhost/Android.bp b/tools/libhost/Android.bp
index 4c9100f15..82268553b 100644
--- a/tools/libhost/Android.bp
+++ b/tools/libhost/Android.bp
@@ -3,7 +3,6 @@ cc_library_host_static {
     srcs: ["CopyFile.c"],
 
     cflags: [
-        "-Werror",
         "-Wall",
     ],
 
diff --git a/tools/makeparallel/Android.bp b/tools/makeparallel/Android.bp
index 898db6861..14be89ac2 100644
--- a/tools/makeparallel/Android.bp
+++ b/tools/makeparallel/Android.bp
@@ -17,5 +17,5 @@ cc_binary_host {
     srcs: [
         "makeparallel.cpp",
     ],
-    cflags: ["-Wall", "-Werror"],
+    cflags: ["-Wall"],
 }
diff --git a/tools/makeparallel/Makefile b/tools/makeparallel/Makefile
index 82a4abfac..b2595d45e 100644
--- a/tools/makeparallel/Makefile
+++ b/tools/makeparallel/Makefile
@@ -27,7 +27,7 @@ MAKEPARALLEL_BIN_PATH ?= .
 MAKEPARALLEL_CXX_SRCS := \
 	makeparallel.cpp
 
-MAKEPARALLEL_CXXFLAGS := -Wall -Werror -MMD -MP
+MAKEPARALLEL_CXXFLAGS := -Wall -MMD -MP
 
 MAKEPARALLEL_CXX_SRCS := $(addprefix $(MAKEPARALLEL_SRC_PATH)/,\
 	$(MAKEPARALLEL_CXX_SRCS))
diff --git a/tools/releasetools/common.py b/tools/releasetools/common.py
index d79f68f90..c17d007da 100644
--- a/tools/releasetools/common.py
+++ b/tools/releasetools/common.py
@@ -2129,7 +2129,7 @@ class BlockDifference(object):
     #   decompression_time: 15s  | 25s                | 25s
 
     if not self.src:
-      brotli_cmd = ['brotli', '--quality=6',
+      brotli_cmd = ['brotli', '--quality=7',
                     '--output={}.new.dat.br'.format(self.path),
                     '{}.new.dat'.format(self.path)]
       print("Compressing {}.new.dat with brotli".format(self.partition))
diff --git a/tools/zipalign/Android.bp b/tools/zipalign/Android.bp
index 8e6196d29..9e65a5cf9 100644
--- a/tools/zipalign/Android.bp
+++ b/tools/zipalign/Android.bp
@@ -13,7 +13,7 @@ cc_binary_host {
         "ZipFile.cpp",
     ],
 
-    cflags: ["-Wall", "-Werror"],
+    cflags: ["-Wall"],
 
     // NOTE: Do not add any shared_libs dependencies because they will break the
     // static_sdk_tools target.
diff --git a/tools/ziptime/Android.bp b/tools/ziptime/Android.bp
index 5ef45ed40..7d882ae72 100644
--- a/tools/ziptime/Android.bp
+++ b/tools/ziptime/Android.bp
@@ -27,7 +27,7 @@ cc_binary_host {
     ],
 
     name: "ziptime",
-    cflags: ["-Wall", "-Werror"],
+    cflags: ["-Wall"],
     target: {
         windows: {
             enabled: true,

project device/qcom/sepolicy-legacy/
diff --git a/common/file.te b/common/file.te
index ce308010..12fd5d5d 100644
--- a/common/file.te
+++ b/common/file.te
@@ -341,3 +341,6 @@ type kgsl_debugfs, debugfs_type, fs_type;
 
 #sysfs emmc dload type
 type sysfs_emmc_dload, sysfs_type, fs_type;
+
+# Custom policies
+type sysfs_disk_stat, fs_type, sysfs_type;

project external/bash/
diff --git a/Android.mk b/Android.mk
index 1a23613..4013be2 100644
--- a/Android.mk
+++ b/Android.mk
@@ -1,82 +1,6 @@
-LOCAL_PATH:= $(call my-dir)
+#LOCAL_PATH:= $(call my-dir)
 
 # ========================================================
 # bash
 # ========================================================
-include $(CLEAR_VARS)
 
-LOCAL_SRC_FILES:= \
-	shell.c eval.c y.tab.c general.c make_cmd.c print_cmd.c \
-	dispose_cmd.c execute_cmd.c variables.c copy_cmd.c error.c \
-	expr.c flags.c jobs.c subst.c hashcmd.c hashlib.c mailcheck.c \
-	trap.c input.c unwind_prot.c pathexp.c sig.c test.c version.c \
-	alias.c array.c arrayfunc.c assoc.c braces.c bracecomp.c \
-	bashhist.c bashline.c list.c stringlib.c locale.c findcmd.c \
-	redir.c pcomplete.c pcomplib.c syntax.c xmalloc.c
-LOCAL_C_INCLUDES += \
-	$(LOCAL_PATH) \
-	$(LOCAL_PATH)/include \
-	$(LOCAL_PATH)/lib
-LOCAL_CFLAGS += \
-	-DHAVE_CONFIG_H \
-	-DPROGRAM=\"bash\" \
-	-DPACKAGE=\"bash\" \
-	-DCONF_HOSTTYPE=\"arm\" \
-	-DCONF_OSTYPE=\"android-eabi\" \
-	-DCONF_MACHTYPE=\"arm-android-eabi\" \
-	-DCONF_VENDOR=\"unknown\" \
-	-DLOCALEDIR=\"/data/locale\" \
-	-DSHELL
-LOCAL_SHARED_LIBRARIES += \
-	libncurses
-LOCAL_STATIC_LIBRARIES += \
-	libbuiltins \
-	libglob \
-	libsh \
-	libtilde \
-	libhistory \
-	libreadline
-LOCAL_CFLAGS += \
-	-Wno-char-subscripts \
-	-Wno-format \
-	-Wno-implicit-function-declaration \
-	-Wno-incompatible-pointer-types \
-	-Wno-incompatible-pointer-types-discards-qualifiers \
-	-Wno-missing-field-initializers \
-	-Wno-non-literal-null-conversion \
-	-Wno-parentheses \
-	-Wno-sign-compare \
-	-Wno-sometimes-uninitialized \
-	-Wno-switch \
-	-Wno-unused-function \
-	-Wno-unused-label \
-	-Wno-unused-parameter \
-	-Wno-unused-variable \
-	-Wno-user-defined-warnings
-
-LOCAL_MODULE := bash
-LOCAL_MODULE_PATH := $(TARGET_OUT_OPTIONAL_EXECUTABLES)
-
-LOCAL_MODULE_TAGS := optional
-
-include $(BUILD_EXECUTABLE)
-
-# ========================================================
-# bash configs
-# ========================================================
-etc_files := $(wildcard $(LOCAL_PATH)/etc/*)
-
-BASH_ETC := $(TARGET_OUT)/etc/$(LOCAL_MODULE)
-BASH_CONFIGS := $(addprefix $(BASH_ETC)/,$(notdir $(etc_files)))
-$(BASH_CONFIGS): $(BASH_ETC)/%: $(LOCAL_PATH)/etc/% | $(LOCAL_BUILT_MODULE)
-	@echo "Install: $@ -> $(BASH_ETC)"
-	@mkdir -p $(dir $@)
-	$(hide) cp $< $@
-
-ALL_DEFAULT_INSTALLED_MODULES += $(BASH_CONFIGS)
-
-ALL_MODULES.$(LOCAL_MODULE).INSTALLED := \
-    $(ALL_MODULES.$(LOCAL_MODULE).INSTALLED) $(BASH_CONFIGS)
-
-# ========================================================
-include $(call all-makefiles-under,$(LOCAL_PATH))

project external/mksh/
diff --git a/mkshrc b/mkshrc
index 256f20e..ae4857b 100644
--- a/mkshrc
+++ b/mkshrc
@@ -1,20 +1,25 @@
-# Copyright (c) 2010, 2012, 2013, 2014
-#	Thorsten Glaser <tg@mirbsd.org>
+# Copyright (c) 2010
+#	Thorsten Glaser <t.glaser@tarent.de>
 # This file is provided under the same terms as mksh.
 #-
 # Minimal /system/etc/mkshrc for Android
-#
-# Support: https://launchpad.net/mksh
-
-set +o nohup
 
+: ${TERM:=xterm-256color} ${HOME:=/} ${MKSH:=/system/bin/sh} ${HOSTNAME:=android}
+: ${SHELL:=$MKSH} ${USER:=$(getprop ro.product.device)}
 if (( USER_ID )); then PS1='$'; else PS1='#'; fi
-PS4='[$EPOCHREALTIME] '; PS1='${|
-	local e=$?
 
-	(( e )) && REPLY+="$e|"
+PS1='$USER@$HOSTNAME:${PWD:-?} '"$PS1 "
+if [ -d "/sbin/.magisk/busybox" ]; then
+  BBDIR="/sbin/.magisk/busybox"
+elif [ -d "/sbin/.core/busybox" ]; then
+  BBDIR="/sbin/.core/busybox"
+fi
+PATH=$PATH:/sbin:$BBDIR:.
+export TERM HOME MKSH HOSTNAME SHELL USER PATH
 
-	return $e
-}$HOSTNAME:${PWD:-?} '"$PS1 "
+for p in ~/.bin; do
+	[[ -d $p/. ]] || continue
+	[[ :$PATH: = *:$p:* ]] || PATH=$p:$PATH
+done
 
-export TERM=xterm-256color
+unset p

project external/nano/
diff --git a/Android.mk b/Android.mk
index 4fcd4369..df6ba085 100644
--- a/Android.mk
+++ b/Android.mk
@@ -1,73 +1,6 @@
-LOCAL_PATH:= $(call my-dir)
+#LOCAL_PATH:= $(call my-dir)
 
 # ========================================================
 # nano
 # ========================================================
-include $(CLEAR_VARS)
 
-LOCAL_SRC_FILES:= \
-	src/browser.c \
-	src/chars.c \
-	src/color.c \
-	src/cut.c \
-	src/files.c \
-	src/global.c \
-	src/help.c \
-	src/history.c \
-	src/move.c \
-	src/nano.c \
-	src/prompt.c \
-	src/rcfile.c \
-	src/search.c \
-	src/text.c \
-	src/utils.c \
-	src/winio.c
-
-LOCAL_C_INCLUDES += \
-	$(LOCAL_PATH) \
-	external/libncurses/include \
-	external/openssh/openbsd-compat
-
-LOCAL_CFLAGS += \
-	-DHAVE_CONFIG_H \
-	-DLOCALEDIR=\"/data/locale\" \
-	-DSYSCONFDIR=\"/system/etc/nano\"
-
-LOCAL_CFLAGS += -Wno-sign-compare -Wno-unused-parameter
-
-LOCAL_SHARED_LIBRARIES += \
-	libncurses \
-	libssh
-
-LOCAL_MODULE := nano
-LOCAL_MODULE_PATH := $(TARGET_OUT_OPTIONAL_EXECUTABLES)
-LOCAL_MODULE_TAGS := optional
-include $(BUILD_EXECUTABLE)
-
-# ========================================================
-# nano configs
-# ========================================================
-NANO_ETC := $(TARGET_OUT_ETC)/$(LOCAL_MODULE)
-
-syntax_files := $(wildcard $(LOCAL_PATH)/syntax/*.nanorc)
-NANO_SYNTAX := $(addprefix $(NANO_ETC)/,$(notdir $(syntax_files)))
-$(NANO_SYNTAX): $(NANO_ETC)/%: $(LOCAL_PATH)/syntax/% | $(LOCAL_BUILT_MODULE)
-	@echo "Install: $@ -> $(NANO_ETC)"
-	@mkdir -p $(dir $@)
-	cp $< $@
-
-nanorc_file := $(LOCAL_PATH)/etc/nanorc
-NANO_NANORC := $(addprefix $(NANO_ETC)/,$(notdir $(nanorc_file)))
-$(NANO_NANORC): $(nanorc_file)
-	@echo "Install: $@ -> $(NANO_ETC)"
-	@mkdir -p $(dir $@)
-	$(hide) cp $< $@
-
-ALL_DEFAULT_INSTALLED_MODULES += $(NANO_SYNTAX) $(NANO_NANORC)
-
-ALL_MODULES.$(LOCAL_MODULE).INSTALLED := \
-	$(ALL_MODULES.$(LOCAL_MODULE).INSTALLED) $(NANO_SYNTAX) \
-	$(ALL_MODULES.$(LOCAL_MODULE).INSTALLED) $(NANO_NANORC)
-
-# ========================================================
-include $(call all-makefiles-under,$(LOCAL_PATH))

project external/noto-fonts/
diff --git a/Android.mk b/Android.mk
index bc5aacd..4141bc2 100644
--- a/Android.mk
+++ b/Android.mk
@@ -41,28 +41,13 @@ ifneq ($(MINIMAL_FONT_FOOTPRINT),true)
 LOCAL_PATH := $(NOTO_DIR)/cjk
 
 font_src_files := \
-    NotoSansCJK-Regular.ttc
+    NotoSansCJK.ttc
 
 $(foreach f, $(font_src_files), $(call build-one-font-module, $(f)))
 font_src_files :=
 
 endif # !MINIMAL_FONT_FOOTPRINT
 
-#############################################################################
-# Similary "build" the Noto CJK fonts for serif family.
-# These are not included in SMALLER_FONT_FOOTPRINT builds.
-#############################################################################
-ifeq ($(filter true,$(EXCLUDE_SERIF_FONTS) $(SMALLER_FONT_FOOTPRINT)),)
-LOCAL_PATH := $(NOTO_DIR)/cjk
-
-font_src_files := \
-    NotoSerifCJK-Regular.ttc
-
-$(foreach f, $(font_src_files), $(call build-one-font-module, $(f)))
-font_src_files :=
-
-endif # !EXCLUDE_SERIF_FONTS && !SMALLER_FONT_FOOTPRINT
-
 #############################################################################
 # Now "build" the Noto Color Emoji font, which is in its own directory. It is
 # not included in the MINIMAL_FONT_FOOTPRINT builds.
diff --git a/fonts.mk b/fonts.mk
index 1cccd97..58cb5a1 100644
--- a/fonts.mk
+++ b/fonts.mk
@@ -49,7 +49,7 @@ PRODUCT_PACKAGES := \
     NotoSansCham-Bold.ttf \
     NotoSansCham-Regular.ttf \
     NotoSansCherokee-Regular.ttf \
-    NotoSansCJK-Regular.ttc \
+    NotoSansCJK.ttc \
     NotoSansCoptic-Regular.ttf \
     NotoSansCuneiform-Regular.ttf \
     NotoSansCypriot-Regular.ttf \
@@ -203,7 +203,6 @@ PRODUCT_PACKAGES := \
     NotoSerifArmenian-Regular.otf \
     NotoSerifBengali-Bold.ttf \
     NotoSerifBengali-Regular.ttf \
-    NotoSerifCJK-Regular.ttc \
     NotoSerifDevanagari-Bold.ttf \
     NotoSerifDevanagari-Regular.ttf \
     NotoSerifEthiopic-Bold.otf \
diff --git a/other/NotoSerif-Bold.ttf b/other/NotoSerif-Bold.ttf
index 2ffdf8c..128288b 100644
Binary files a/other/NotoSerif-Bold.ttf and b/other/NotoSerif-Bold.ttf differ
diff --git a/other/NotoSerif-Regular.ttf b/other/NotoSerif-Regular.ttf
index 9d9a347..78136ac 100644
Binary files a/other/NotoSerif-Regular.ttf and b/other/NotoSerif-Regular.ttf differ

project external/openssh/
diff --git a/Android.mk b/Android.mk
index 6cefcf94..8f877908 100644
--- a/Android.mk
+++ b/Android.mk
@@ -1,338 +1,6 @@
-LOCAL_PATH:= $(call my-dir)
-
-openssh_common_cflags := \
-    -Wall \
-    -Werror \
-    -Wno-error=implicit-function-declaration \
-    -Wno-pointer-sign \
-    -Wno-sign-compare \
-    -Wno-type-limits \
-    -Wno-unused-parameter \
-    -Wno-unused-variable \
-    -Wno-error \
-
 # Use -Wno-error to allow at least the following warnings:
 # (1) bsd-openpty.c calls to 'ptsname' declared with attribute warning:
 #     ptsname is not thread-safe; use ptsname_r instead [-Werror]
 # (2) external/boringssl/src/include/openssl/opensslfeatures.h:
 #     error: "OPENSSL_NO_BF" redefined [-Werror]
 
-openssh_common_clang_cflags := \
-    -Wno-incompatible-pointer-types \
-    -Wno-macro-redefined \
-
-###################### libssh ######################
-include $(CLEAR_VARS)
-
-LOCAL_MODULE_TAGS := optional
-
-LOCAL_SRC_FILES := \
-    addrmatch.c \
-    atomicio.c \
-    authfd.c \
-    authfile.c \
-    bitmap.c \
-    canohost.c \
-    chacha.c \
-    channels.c \
-    cipher-aes.c \
-    cipher-aesctr.c \
-    cipher.c \
-    cipher-chachapoly.c \
-    cipher-ctr.c \
-    cleanup.c \
-    compat.c \
-    crc32.c \
-    dh.c \
-    digest-openssl.c \
-    dispatch.c \
-    dns.c \
-    ed25519.c \
-    entropy.c \
-    fatal.c \
-    fe25519.c \
-    ge25519.c \
-    gss-genr.c \
-    hash.c \
-    hmac.c \
-    hostfile.c \
-    kex.c \
-    kexc25519.c \
-    kexdh.c \
-    kexecdh.c \
-    kexgen.c \
-    kexgex.c \
-    kexgexc.c \
-    kexsntrup4591761x25519.c \
-    krl.c \
-    log.c \
-    mac.c \
-    match.c \
-    misc.c \
-    moduli.c \
-    monitor_fdpass.c \
-    msg.c \
-    nchan.c \
-    openbsd-compat/bcrypt_pbkdf.c \
-    openbsd-compat/bindresvport.c \
-    openbsd-compat/blowfish.c \
-    openbsd-compat/bsd-closefrom.c \
-    openbsd-compat/bsd-getpeereid.c \
-    openbsd-compat/bsd-misc.c \
-    openbsd-compat/bsd-openpty.c \
-    openbsd-compat/bsd-signal.c \
-    openbsd-compat/bsd-statvfs.c \
-    openbsd-compat/explicit_bzero.c \
-    openbsd-compat/freezero.c \
-    openbsd-compat/fmt_scaled.c \
-    openbsd-compat/getopt_long.c \
-    openbsd-compat/glob.c \
-    openbsd-compat/boringssl-api-compat.c \
-    openbsd-compat/openssl-compat.c \
-    openbsd-compat/port-linux.c \
-    openbsd-compat/port-net.c \
-    openbsd-compat/pwcache.c \
-    openbsd-compat/readpassphrase.c \
-    openbsd-compat/reallocarray.c \
-    openbsd-compat/recallocarray.c \
-    openbsd-compat/rresvport.c \
-    openbsd-compat/setproctitle.c \
-    openbsd-compat/strmode.c \
-    openbsd-compat/strtonum.c \
-    openbsd-compat/timingsafe_bcmp.c \
-    openbsd-compat/vis.c \
-    packet.c \
-    platform-misc.c \
-    platform-pledge.c \
-    platform-tracing.c \
-    poly1305.c \
-    readpass.c \
-    rijndael.c \
-    sc25519.c \
-    smult_curve25519_ref.c \
-    sshbuf.c \
-    sshbuf-getput-basic.c \
-    sshbuf-getput-crypto.c \
-    sshbuf-misc.c \
-    ssh-dss.c \
-    ssh-ecdsa.c \
-    ssh-ed25519.c \
-    ssherr.c \
-    sshkey.c \
-    sshkey-xmss.c \
-    ssh-rsa.c \
-    ssh-xmss.c \
-    sntrup4591761.c \
-    ttymodes.c \
-    uidswap.c \
-    umac128.c \
-    umac.c \
-    utf8.c \
-    uuencode.c \
-    verify.c \
-    xmalloc.c \
-    xmss_commons.c \
-    xmss_fast.c \
-    xmss_hash_address.c \
-    xmss_hash.c \
-    xmss_wots.c
-
-LOCAL_C_INCLUDES := \
-    external/zlib \
-    external/openssl/include \
-    external/openssh/openbsd-compat
-
-LOCAL_SHARED_LIBRARIES += libssl libcrypto libdl libz
-
-LOCAL_MODULE := libssh
-
-LOCAL_CFLAGS += -O3 $(openssh_common_cflags)
-LOCAL_CLANG_CFLAGS += $(openssh_common_clang_cflags)
-
-LOCAL_CFLAGS += -DGCE_PLATFORM_SDK_VERSION=$(PLATFORM_SDK_VERSION)
-
-ifneq (,$(SSHDIR))
-LOCAL_CFLAGS += -DSSHDIR=\"$(SSHDIR)\"
-endif
-
-include $(BUILD_SHARED_LIBRARY)
-
-###################### ssh ######################
-
-include $(CLEAR_VARS)
-
-LOCAL_MODULE_TAGS := optional
-
-LOCAL_SRC_FILES := \
-    clientloop.c \
-    readconf.c \
-    mux.c \
-    ssh.c \
-    sshconnect2.c \
-    sshconnect.c \
-    sshtty.c \
-    uidswap.c
-
-LOCAL_MODULE := ssh
-
-LOCAL_CFLAGS += $(openssh_common_cflags)
-LOCAL_CLANG_CFLAGS += $(openssh_common_clang_cflags)
-
-LOCAL_C_INCLUDES := \
-    external/zlib \
-    external/openssl/include \
-    external/openssh/openbsd-compat
-
-LOCAL_SHARED_LIBRARIES += libssh libssl libcrypto libdl libz
-
-include $(BUILD_EXECUTABLE)
-
-###################### sftp ######################
-
-include $(CLEAR_VARS)
-
-LOCAL_MODULE_TAGS := optional
-
-LOCAL_SRC_FILES := \
-    sftp.c sftp-client.c sftp-common.c sftp-glob.c progressmeter.c
-
-LOCAL_MODULE := sftp
-
-LOCAL_CFLAGS += $(openssh_common_cflags)
-LOCAL_CLANG_CFLAGS += $(openssh_common_clang_cflags)
-
-LOCAL_C_INCLUDES := \
-    external/zlib \
-    external/openssl/include \
-    external/openssh/openbsd-compat
-
-LOCAL_SHARED_LIBRARIES += libssh libssl libcrypto libdl libz
-
-include $(BUILD_EXECUTABLE)
-
-###################### scp ######################
-
-include $(CLEAR_VARS)
-
-LOCAL_MODULE_TAGS := optional
-
-LOCAL_SRC_FILES := \
-    scp.c progressmeter.c
-
-LOCAL_MODULE := scp
-
-LOCAL_CFLAGS += $(openssh_common_cflags)
-LOCAL_CLANG_CFLAGS += $(openssh_common_clang_cflags)
-
-LOCAL_C_INCLUDES := \
-    external/zlib \
-    external/openssl/include \
-    external/openssh/openbsd-compat
-
-LOCAL_SHARED_LIBRARIES += libssh libssl libcrypto libdl libz
-
-include $(BUILD_EXECUTABLE)
-
-###################### sshd ######################
-
-include $(CLEAR_VARS)
-
-LOCAL_MODULE_TAGS := optional
-
-LOCAL_SRC_FILES := \
-    audit-bsm.c \
-    audit-linux.c \
-    audit.c \
-    auth-bsdauth.c \
-    auth-krb5.c \
-    auth-options.c \
-    auth-pam.c \
-    auth-rhosts.c \
-    auth-shadow.c \
-    auth-sia.c \
-    auth.c \
-    auth2-chall.c \
-    auth2-gss.c \
-    auth2-hostbased.c \
-    auth2-kbdint.c \
-    auth2-none.c \
-    auth2-passwd.c \
-    auth2-pubkey.c \
-    auth2.c \
-    groupaccess.c \
-    gss-serv-krb5.c \
-    gss-serv.c \
-    kexgexs.c \
-    loginrec.c \
-    md5crypt.c \
-    monitor.c \
-    monitor_wrap.c \
-    platform.c \
-    sandbox-null.c \
-    sandbox-rlimit.c \
-    sandbox-systrace.c \
-    servconf.c \
-    serverloop.c \
-    session.c \
-    sftp-common.c \
-    sftp-server.c \
-    sshd.c \
-    sshlogin.c \
-    sshpty.c
-
-LOCAL_MODULE := sshd
-
-LOCAL_CFLAGS += $(openssh_common_cflags)
-LOCAL_CLANG_CFLAGS += $(openssh_common_clang_cflags)
-
-LOCAL_C_INCLUDES := \
-    external/zlib \
-    external/openssl/include \
-    external/openssh/openbsd-compat
-
-LOCAL_SHARED_LIBRARIES += libssh libssl libcrypto libdl libz libcutils
-LOCAL_SHARED_LIBRARIES += libc.bootstrap
-
-include $(BUILD_EXECUTABLE)
-
-###################### ssh-keygen ######################
-
-include $(CLEAR_VARS)
-
-LOCAL_MODULE_TAGS := optional
-
-LOCAL_SRC_FILES := \
-    ssh-keygen.c
-
-LOCAL_MODULE := ssh-keygen
-
-LOCAL_CFLAGS += $(openssh_common_cflags)
-LOCAL_CLANG_CFLAGS += $(openssh_common_clang_cflags)
-
-LOCAL_C_INCLUDES := \
-    external/zlib \
-    external/openssl/include \
-    external/openssh/openbsd-compat
-
-LOCAL_SHARED_LIBRARIES += libssh libssl libcrypto libdl libz
-
-include $(BUILD_EXECUTABLE)
-
-###################### sshd_config ######################
-
-include $(CLEAR_VARS)
-LOCAL_MODULE_TAGS := optional
-LOCAL_MODULE := sshd_config
-LOCAL_MODULE_CLASS := ETC
-LOCAL_MODULE_PATH := $(TARGET_OUT_ETC)/ssh
-LOCAL_SRC_FILES := sshd_config.android
-include $(BUILD_PREBUILT)
-
-###################### start-ssh ######################
-
-include $(CLEAR_VARS)
-LOCAL_MODULE_TAGS := optional
-LOCAL_MODULE := start-ssh
-LOCAL_MODULE_CLASS := EXECUTABLES
-LOCAL_SRC_FILES := start-ssh
-include $(BUILD_PREBUILT)

project external/perfetto/
diff --git a/Android.bp b/Android.bp
index b7f12d29..e9ffde1d 100644
--- a/Android.bp
+++ b/Android.bp
@@ -3352,49 +3352,6 @@ cc_binary_host {
   ],
 }
 
-// GN target: //:traced
-cc_binary {
-  name: "traced",
-  srcs: [
-    "src/traced/service/main.cc",
-  ],
-  shared_libs: [
-    "liblog",
-    "libperfetto",
-  ],
-  init_rc: [
-    "perfetto.rc",
-  ],
-  defaults: [
-    "perfetto_defaults",
-  ],
-  cflags: [
-    "-DPERFETTO_BUILD_WITH_ANDROID",
-  ],
-}
-
-// GN target: //:traced_probes
-cc_binary {
-  name: "traced_probes",
-  srcs: [
-    "src/traced/probes/main.cc",
-  ],
-  shared_libs: [
-    "liblog",
-    "libperfetto",
-  ],
-  defaults: [
-    "perfetto_defaults",
-  ],
-  cflags: [
-    "-DPERFETTO_BUILD_WITH_ANDROID",
-  ],
-  required: [
-    "libperfetto_android_internal",
-    "trigger_perfetto",
-  ],
-}
-
 // GN target: //:trigger_perfetto
 cc_binary {
   name: "trigger_perfetto",
@@ -3585,4 +3542,4 @@ java_library_host {
   srcs: [
     "protos/perfetto/config/perfetto_config.proto",
   ],
-}
\ No newline at end of file
+}

project external/vim/
diff --git a/Android.mk b/Android.mk
index 858a4e8a3..daa840d7d 100644
--- a/Android.mk
+++ b/Android.mk
@@ -1,218 +1,6 @@
-vim_src := $(call my-dir)
+#vim_src := $(call my-dir)
 
 # ========================================================
 # etc/vimrc
 # ========================================================
 
-LOCAL_PATH := $(vim_src)
-include $(CLEAR_VARS)
-
-LOCAL_MODULE := vimrc
-LOCAL_MODULE_TAGS := optional
-LOCAL_MODULE_CLASS := ETC
-
-LOCAL_SRC_FILES := vimrc.android
-
-include $(BUILD_PREBUILT)
-
-# ========================================================
-# vim
-# ========================================================
-
-LOCAL_PATH := $(vim_src)/src
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := \
-	auto/pathdef.c \
-	autocmd.c \
-	blob.c \
-	blowfish.c \
-	buffer.c \
-	change.c \
-	channel.c \
-	charset.c \
-	debugger.c \
-	dict.c \
-	diff.c \
-	digraph.c \
-	edit.c \
-	eval.c \
-	evalfunc.c \
-	ex_cmds.c \
-	ex_cmds2.c \
-	ex_docmd.c \
-	ex_eval.c \
-	ex_getln.c \
-	fileio.c \
-	findfile.c \
-	fold.c \
-	getchar.c \
-	hardcopy.c \
-	hashtab.c \
-	if_cscope.c \
-	if_xcmdsrv.c \
-	indent.c \
-	json.c \
-	list.c \
-	main.c \
-	mark.c \
-	mbyte.c \
-	memfile.c \
-	memline.c \
-	menu.c \
-	message.c \
-	misc1.c \
-	misc2.c \
-	move.c \
-	normal.c \
-	ops.c \
-	option.c \
-	os_unix.c \
-	popupmnu.c \
-	popupwin.c \
-	pty.c \
-	quickfix.c \
-	regexp.c \
-	screen.c \
-	search.c \
-	sha256.c \
-	spell.c \
-	syntax.c \
-	tag.c \
-	term.c \
-	textprop.c \
-	ui.c \
-	undo.c \
-	usercmd.c \
-	userfunc.c \
-	version.c \
-	window.c
-
-LOCAL_C_INCLUDES += \
-	external/libselinux/include \
-	external/libncurses/include \
-	$(LOCAL_PATH)/proto \
-	$(LOCAL_PATH)/auto
-
-LOCAL_SHARED_LIBRARIES += \
-	libselinux \
-	libncurses \
-	libm \
-	libdl
-
-LOCAL_CFLAGS += \
-	-DHAVE_CONFIG_H \
-	-DSYS_VIMRC_FILE=\"/system/etc/vimrc\"
-
-# vim variants: TINY SMALL CM NORMAL BIG HUGE
-#
-# NORMAL, BIG and HUGE are almost the same (1.1M)
-# TINY and SMALL are similar to busybox vi (460K)
-#
-# our profile is between SMALL and NORMAL (780K)
-# with syntax and utf8 (mbyte) support
-#
-# to reduce vim size, manually define wanted features
-LOCAL_CFLAGS += \
-	-DFEAT_SMALL=1 \
-	-DFEAT_MBYTE=1 \
-	-DFEAT_SYN_HL=1 \
-	-DFEAT_CINDENT=1 \
-	-DFEAT_COMMENTS=1 \
-	-DFEAT_EVAL=1 \
-	-DFEAT_AUTOCMD=1 \
-	-DFEAT_USR_CMDS=1 \
-	-DFEAT_EX_EXTRA=1 \
-	-DFEAT_CMDL_COMPL=1 \
-	-DFEAT_LISTCMDS=1 \
-	-DFEAT_CMDL_INFO=1 \
-	-DFEAT_SEARCH_EXTRA=1
-
-LOCAL_CFLAGS += -Wno-unused-variable -Wno-unused-parameter
-
-LOCAL_MODULE := vim
-LOCAL_MODULE_TAGS := optional
-LOCAL_MODULE_PATH := $(TARGET_OUT_OPTIONAL_EXECUTABLES)
-LOCAL_REQUIRED_MODULES := vimrc
-include $(BUILD_EXECUTABLE)
-
-# ========================================================
-# vim runtime files
-# ========================================================
-ifeq (vim,$(filter vim, $(ALL_MODULES)))
-
-vim_runtime_path := $(vim_src)/runtime
-
-vim_runtime_files := \
-	scripts.vim \
-	indent.vim \
-	indoff.vim \
-	filetype.vim \
-	ftoff.vim
-
-vim_doc_files := \
-	help.txt intro.txt tags \
-	motion.txt editing.txt scroll.txt \
-	options.txt term.txt
-
-vim_colors_files := \
-	default.vim \
-	desert.vim
-
-vim_syntax_files := \
-	logcat.vim \
-	awk.vim \
-	config.vim \
-	conf.vim \
-	cpp.vim \
-	c.vim \
-	css.vim \
-	diff.vim \
-	doxygen.vim \
-	html.vim vb.vim \
-	xml.vim dtd.vim \
-	context.vim \
-	gitcommit.vim \
-	help.vim \
-	javascript.vim \
-	java.vim \
-	lua.vim \
-	manual.vim \
-	markdown.vim \
-	pod.vim \
-	sh.vim \
-	syncolor.vim \
-	synload.vim \
-	syntax.vim \
-	vim.vim
-
-vim_plugin_files := \
-	matchparen.vim \
-
-vim_autoload_files := \
-	dist/ft.vim \
-	spacehi.vim
-
-VIM_SHARED := $(TARGET_OUT)/usr/share/vim
-
-vim_runtime_files := \
-  $(vim_runtime_files) \
-  $(addprefix doc/, $(vim_doc_files)) \
-  $(addprefix colors/, $(vim_colors_files)) \
-  $(addprefix syntax/, $(vim_syntax_files)) \
-  $(addprefix plugin/, $(vim_plugin_files)) \
-  $(addprefix autoload/, $(vim_autoload_files)) \
-
-vim_runtime_modules := $(addprefix $(VIM_SHARED)/, $(vim_runtime_files))
-$(vim_runtime_modules): $(VIM_SHARED)/%: $(vim_runtime_path)/% | $(LOCAL_BUILT_MODULE)
-	@echo "Install: $@"
-	@mkdir -p $(dir $@)
-	$(hide) cp $< $@
-
-ALL_DEFAULT_INSTALLED_MODULES += $(vim_runtime_modules)
-
-ALL_MODULES.$(LOCAL_MODULE).INSTALLED := \
-  $(ALL_MODULES.$(LOCAL_MODULE).INSTALLED) \
-  $(vim_runtime_modules)
-
-endif

project frameworks/base/
diff --git a/core/res/AndroidManifest.xml b/core/res/AndroidManifest.xml
index 0c3b7a137b9..fc6f9d9a1ec 100644
--- a/core/res/AndroidManifest.xml
+++ b/core/res/AndroidManifest.xml
@@ -2637,6 +2637,13 @@
         android:description="@string/permdesc_getPackageSize"
         android:protectionLevel="normal" />
 
+    <!-- @hide Allows an application to change the package signature as
+         seen by applications -->
+    <permission android:name="android.permission.FAKE_PACKAGE_SIGNATURE"
+        android:protectionLevel="dangerous"
+        android:label="@string/permlab_fakePackageSignature"
+        android:description="@string/permdesc_fakePackageSignature" />
+
     <!-- @deprecated No longer useful, see
          {@link android.content.pm.PackageManager#addPackageToPreferred}
          for details. -->
diff --git a/core/res/res/values-zh-rCN/strings.xml b/core/res/res/values-zh-rCN/strings.xml
index ccbd5d29324..40556d6109a 100644
--- a/core/res/res/values-zh-rCN/strings.xml
+++ b/core/res/res/values-zh-rCN/strings.xml
@@ -2034,4 +2034,6 @@
     </plurals>
     <string name="chooser_no_direct_share_targets" msgid="997970693708458895">"无法使用直接分享功能"</string>
     <string name="chooser_all_apps_button_label" msgid="3631524352936289457">"应用列表"</string>
+    <string name="permlab_fakePackageSignature">模拟软件包签名</string>
+    <string name="permdesc_fakePackageSignature">允许应用伪装成为其他应用</string>
 </resources>
diff --git a/core/res/res/values/config.xml b/core/res/res/values/config.xml
index 91f4cf3093e..91fb69f8066 100644
--- a/core/res/res/values/config.xml
+++ b/core/res/res/values/config.xml
@@ -1856,6 +1856,8 @@
     <string-array name="config_locationProviderPackageNames" translatable="false">
         <!-- The standard AOSP fused location provider -->
         <item>com.android.location.fused</item>
+        <!-- The (faked) microg fused location provider (a free reimplementation) -->
+        <item>com.google.android.gms</item>
     </string-array>
 
     <!-- This string array can be overriden to enable test location providers initially. -->
diff --git a/core/res/res/values/strings.xml b/core/res/res/values/strings.xml
index 152b131af8f..f8d87a37064 100644
--- a/core/res/res/values/strings.xml
+++ b/core/res/res/values/strings.xml
@@ -827,6 +827,11 @@
 
     <!--  Permissions -->
 
+    <!-- Title of an application permission, listed so the user can choose whether they want to allow the application to do this. -->
+    <string name="permlab_fakePackageSignature">Spoof package signature</string>
+    <!-- Description of an application permission, listed so the user can choose whether they want to allow the application to do this. -->
+    <string name="permdesc_fakePackageSignature">Allows the app to pretend to be a different app. Malicious applications might be able to use this to access private application data. Legitimate uses include an emulator pretending to be what it emulates. Grant this permission with caution only!</string>
+
     <!-- Title of an application permission, listed so the user can choose whether they want to allow the application to do this. -->
     <string name="permlab_statusBar">disable or modify status bar</string>
     <!-- Description of an application permission, listed so the user can choose whether they want to allow the application to do this. -->
diff --git a/data/fonts/fonts.xml b/data/fonts/fonts.xml
index 072beae8baf..5cb3f11dcf0 100644
--- a/data/fonts/fonts.xml
+++ b/data/fonts/fonts.xml
@@ -544,20 +544,56 @@
         <font weight="400" style="normal">NotoSansSymbols-Regular-Subsetted.ttf</font>
     </family>
     <family lang="zh-Hans">
-        <font weight="400" style="normal" index="2">NotoSansCJK-Regular.ttc</font>
-        <font weight="400" style="normal" index="2" fallbackFor="serif">NotoSerifCJK-Regular.ttc</font>
-    </family>
-    <family lang="zh-Hant,zh-Bopo">
-        <font weight="400" style="normal" index="3">NotoSansCJK-Regular.ttc</font>
-        <font weight="400" style="normal" index="3" fallbackFor="serif">NotoSerifCJK-Regular.ttc</font>
+        <font weight="100" style="normal" index="2">NotoSansCJK.ttc</font>
+        <font weight="300" style="normal" index="7">NotoSansCJK.ttc</font>
+        <font weight="350" style="normal" index="12">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="17">NotoSansCJK.ttc</font>
+        <font weight="500" style="normal" index="27">NotoSansCJK.ttc</font>
+        <font weight="700" style="normal" index="32">NotoSansCJK.ttc</font>
+        <font weight="900" style="normal" index="42">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" fallbackFor="serif">NotoSerif-Regular.ttf</font>
+        <font weight="700" style="normal" fallbackFor="serif">NotoSerif-Bold.ttf</font>
+        <font weight="400" style="italic" fallbackFor="serif">NotoSerif-Italic.ttf</font>
+        <font weight="700" style="italic" fallbackFor="serif">NotoSerif-BoldItalic.ttf</font>
+    </family>
+    <family lang="zh-Hant zh-Bopo">
+        <font weight="100" style="normal" index="3">NotoSansCJK.ttc</font>
+        <font weight="300" style="normal" index="8">NotoSansCJK.ttc</font>
+        <font weight="350" style="normal" index="13">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="18">NotoSansCJK.ttc</font>
+        <font weight="500" style="normal" index="28">NotoSansCJK.ttc</font>
+        <font weight="700" style="normal" index="33">NotoSansCJK.ttc</font>
+        <font weight="900" style="normal" index="43">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" fallbackFor="serif">NotoSerif-Regular.ttf</font>
+        <font weight="700" style="normal" fallbackFor="serif">NotoSerif-Bold.ttf</font>
+        <font weight="400" style="italic" fallbackFor="serif">NotoSerif-Italic.ttf</font>
+        <font weight="700" style="italic" fallbackFor="serif">NotoSerif-BoldItalic.ttf</font>
     </family>
     <family lang="ja">
-        <font weight="400" style="normal" index="0">NotoSansCJK-Regular.ttc</font>
-        <font weight="400" style="normal" index="0" fallbackFor="serif">NotoSerifCJK-Regular.ttc</font>
+        <font weight="100" style="normal" index="0">NotoSansCJK.ttc</font>
+        <font weight="300" style="normal" index="5">NotoSansCJK.ttc</font>
+        <font weight="350" style="normal" index="10">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="15">NotoSansCJK.ttc</font>
+        <font weight="500" style="normal" index="25">NotoSansCJK.ttc</font>
+        <font weight="700" style="normal" index="30">NotoSansCJK.ttc</font>
+        <font weight="900" style="normal" index="40">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" fallbackFor="serif">NotoSerif-Regular.ttf</font>
+        <font weight="700" style="normal" fallbackFor="serif">NotoSerif-Bold.ttf</font>
+        <font weight="400" style="italic" fallbackFor="serif">NotoSerif-Italic.ttf</font>
+        <font weight="700" style="italic" fallbackFor="serif">NotoSerif-BoldItalic.ttf</font>
     </family>
     <family lang="ko">
-        <font weight="400" style="normal" index="1">NotoSansCJK-Regular.ttc</font>
-        <font weight="400" style="normal" index="1" fallbackFor="serif">NotoSerifCJK-Regular.ttc</font>
+        <font weight="100" style="normal" index="1">NotoSansCJK.ttc</font>
+        <font weight="300" style="normal" index="6">NotoSansCJK.ttc</font>
+        <font weight="350" style="normal" index="11">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="16">NotoSansCJK.ttc</font>
+        <font weight="500" style="normal" index="26">NotoSansCJK.ttc</font>
+        <font weight="700" style="normal" index="31">NotoSansCJK.ttc</font>
+        <font weight="900" style="normal" index="41">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" fallbackFor="serif">NotoSerif-Regular.ttf</font>
+        <font weight="700" style="normal" fallbackFor="serif">NotoSerif-Bold.ttf</font>
+        <font weight="400" style="italic" fallbackFor="serif">NotoSerif-Italic.ttf</font>
+        <font weight="700" style="italic" fallbackFor="serif">NotoSerif-BoldItalic.ttf</font>
     </family>
     <family lang="und-Zsye">
         <font weight="400" style="normal">NotoColorEmoji.ttf</font>
diff --git a/packages/CtsShim/Android.mk b/packages/CtsShim/Android.mk
index 12972f14514..4bd31e5781a 100644
--- a/packages/CtsShim/Android.mk
+++ b/packages/CtsShim/Android.mk
@@ -14,51 +14,9 @@
 # limitations under the License.
 #
 
-LOCAL_PATH := $(call my-dir)
-
 ###########################################################
 # Variant: Privileged app
 
-include $(CLEAR_VARS)
-
-LOCAL_MODULE := CtsShimPrivPrebuilt
-LOCAL_MODULE_TAGS := optional
-# this needs to be a privileged application
-LOCAL_PRIVILEGED_MODULE := true
-LOCAL_MODULE_CLASS := APPS
-LOCAL_BUILT_MODULE_STEM := package.apk
-# Make sure the build system doesn't try to resign the APK
-LOCAL_CERTIFICATE := PRESIGNED
-LOCAL_DEX_PREOPT := false
-LOCAL_MODULE_TARGET_ARCH := arm arm64 x86 x86_64
-
-LOCAL_SRC_FILES_arm := apk/arm/CtsShimPriv.apk
-LOCAL_SRC_FILES_arm64 := apk/arm/CtsShimPriv.apk
-LOCAL_SRC_FILES_x86 := apk/x86/CtsShimPriv.apk
-LOCAL_SRC_FILES_x86_64 := apk/x86/CtsShimPriv.apk
-
-include $(BUILD_PREBUILT)
-
-
 ###########################################################
 # Variant: System app
 
-include $(CLEAR_VARS)
-
-LOCAL_MODULE := CtsShimPrebuilt
-LOCAL_MODULE_TAGS := optional
-LOCAL_MODULE_CLASS := APPS
-LOCAL_BUILT_MODULE_STEM := package.apk
-# Make sure the build system doesn't try to resign the APK
-LOCAL_CERTIFICATE := PRESIGNED
-LOCAL_DEX_PREOPT := false
-LOCAL_MODULE_TARGET_ARCH := arm arm64 x86 x86_64
-
-LOCAL_SRC_FILES_arm := apk/arm/CtsShim.apk
-LOCAL_SRC_FILES_arm64 := apk/arm/CtsShim.apk
-LOCAL_SRC_FILES_x86 := apk/x86/CtsShim.apk
-LOCAL_SRC_FILES_x86_64 := apk/x86/CtsShim.apk
-
-include $(BUILD_PREBUILT)
-
-include $(call all-makefiles-under,$(LOCAL_PATH))
diff --git a/packages/overlays/Android.mk b/packages/overlays/Android.mk
index 90d1c11e6c0..f4628d76188 100644
--- a/packages/overlays/Android.mk
+++ b/packages/overlays/Android.mk
@@ -24,9 +24,6 @@ LOCAL_REQUIRED_MODULES := \
 	AccentColorSpaceOverlay \
 	AccentColorGreenOverlay \
 	AccentColorPurpleOverlay \
-	DisplayCutoutEmulationCornerOverlay \
-	DisplayCutoutEmulationDoubleOverlay \
-	DisplayCutoutEmulationTallOverlay \
 	FontNotoSerifSourceOverlay \
 	IconPackCircularAndroidOverlay \
 	IconPackCircularLauncherOverlay \
diff --git a/services/core/java/com/android/server/pm/PackageManagerService.java b/services/core/java/com/android/server/pm/PackageManagerService.java
index 5c150de4308..4e9762a5ed4 100644
--- a/services/core/java/com/android/server/pm/PackageManagerService.java
+++ b/services/core/java/com/android/server/pm/PackageManagerService.java
@@ -4141,8 +4141,9 @@ public class PackageManagerService extends IPackageManager.Stub
             final Set<String> permissions = ArrayUtils.isEmpty(p.requestedPermissions)
                     ? Collections.emptySet() : permissionsState.getPermissions(userId);
 
-            PackageInfo packageInfo = PackageParser.generatePackageInfo(p, gids, flags,
-                    ps.firstInstallTime, ps.lastUpdateTime, permissions, state, userId);
+            PackageInfo packageInfo = mayFakeSignature(p, PackageParser.generatePackageInfo(p, gids, flags,
+                    ps.firstInstallTime, ps.lastUpdateTime, permissions, state, userId),
+                    permissions);
 
             if (packageInfo == null) {
                 return null;
@@ -4178,6 +4179,24 @@ public class PackageManagerService extends IPackageManager.Stub
         }
     }
 
+    private PackageInfo mayFakeSignature(PackageParser.Package p, PackageInfo pi,
+            Set<String> permissions) {
+        try {
+            if (permissions.contains("android.permission.FAKE_PACKAGE_SIGNATURE")
+                    && p.applicationInfo.targetSdkVersion > Build.VERSION_CODES.LOLLIPOP_MR1
+                    && p.mAppMetaData != null) {
+                String sig = p.mAppMetaData.getString("fake-signature");
+                if (sig != null) {
+                    pi.signatures = new Signature[] {new Signature(sig)};
+                }
+            }
+        } catch (Throwable t) {
+            // We should never die because of any failures, this is system code!
+            Log.w("PackageManagerService.FAKE_PACKAGE_SIGNATURE", t);
+	    }
+        return pi;
+    }
+
     @Override
     public void checkPackageStartable(String packageName, int userId) {
         final int callingUid = Binder.getCallingUid();

project hardware/qcom-caf/msm8994/audio/
diff --git a/hal/Android.mk b/hal/Android.mk
index 7eca4aaf7..5818b498f 100644
--- a/hal/Android.mk
+++ b/hal/Android.mk
@@ -264,7 +264,7 @@ ifeq ($(strip $(AUDIO_FEATURE_ENABLED_AUXPCM_BT)),true)
     LOCAL_CFLAGS += -DAUXPCM_BT_ENABLED
 endif
 
-LOCAL_CFLAGS += -Wall -Werror
+LOCAL_CFLAGS += -Wall -Wno-error=implicit-function-declaration
 
 LOCAL_COPY_HEADERS_TO   := mm-audio
 LOCAL_COPY_HEADERS      := audio_extn/audio_defs.h
diff --git a/hal/audio_hw.c b/hal/audio_hw.c
index 260f156b9..3dc11ea60 100644
--- a/hal/audio_hw.c
+++ b/hal/audio_hw.c
@@ -2381,12 +2381,14 @@ static ssize_t out_write(struct audio_stream_out *stream, const void *buffer,
             ALOGD("copl(%p):send new gapless metadata", out);
             compress_set_gapless_metadata(out->compr, &out->gapless_mdata);
             out->send_new_metadata = 0;
+#if defined(SNDRV_COMPRESS_SET_NEXT_TRACK_PARAM)
             if (out->send_next_track_params && out->is_compr_metadata_avail) {
                 ALOGD("copl(%p):send next track params in gapless", out);
                 compress_set_next_track_param(out->compr, &(out->compr_config.codec->options));
                 out->send_next_track_params = false;
                 out->is_compr_metadata_avail = false;
             }
+#endif
         }
 
         ret = compress_write(out->compr, buffer, bytes);

project hardware/qcom-caf/msm8994/display/
diff --git a/libhwcomposer/Android.mk b/libhwcomposer/Android.mk
index 3dc471459..47d7c4a4b 100644
--- a/libhwcomposer/Android.mk
+++ b/libhwcomposer/Android.mk
@@ -19,7 +19,8 @@ LOCAL_SHARED_LIBRARIES += libhwui
 endif #TARGET_USES_QCOM_BSP
 
 LOCAL_CFLAGS                  := $(common_flags) -DLOG_TAG=\"qdhwcomposer\" -Wno-absolute-value \
-                                 -Wno-float-conversion -Wno-unused-parameter
+                                 -Wno-float-conversion -Wno-unused-parameter \
+                                 -Wno-error=implicit-fallthrough
 #Enable Dynamic FPS if PHASE_OFFSET is not set
 ifeq ($(VSYNC_EVENT_PHASE_OFFSET_NS),)
     LOCAL_CFLAGS += -DDYNAMIC_FPS

project hardware/qcom-caf/msm8994/media/
diff --git a/mm-video-v4l2/vidc/common/Android.mk b/mm-video-v4l2/vidc/common/Android.mk
index 6f783c429..c14bdcffe 100644
--- a/mm-video-v4l2/vidc/common/Android.mk
+++ b/mm-video-v4l2/vidc/common/Android.mk
@@ -13,7 +13,6 @@ libmm-vidc-def += -D__alignx\(x\)=__attribute__\(\(__aligned__\(x\)\)\)
 libmm-vidc-def += -DT_ARM
 libmm-vidc-def += -Dinline=__inline
 libmm-vidc-def += -D_ANDROID_
-libmm-vidc-def += -Werror
 libmm-vidc-def += -D_ANDROID_ICS_
 
 # ---------------------------------------------------------------------------------
diff --git a/mm-video-v4l2/vidc/vdec/Android.mk b/mm-video-v4l2/vidc/vdec/Android.mk
index aedca2059..de931ab52 100644
--- a/mm-video-v4l2/vidc/vdec/Android.mk
+++ b/mm-video-v4l2/vidc/vdec/Android.mk
@@ -97,7 +97,7 @@ include $(CLEAR_VARS)
 LOCAL_MODULE                    := libOmxVdec
 LOCAL_MODULE_TAGS               := optional
 LOCAL_VENDOR_MODULE             := true
-LOCAL_CFLAGS                    := $(libmm-vdec-def) -Werror
+LOCAL_CFLAGS                    := $(libmm-vdec-def)
 LOCAL_C_INCLUDES                += $(libmm-vdec-inc)
 
 LOCAL_HEADER_LIBRARIES := generated_kernel_headers
diff --git a/mm-video-v4l2/vidc/venc/Android.mk b/mm-video-v4l2/vidc/venc/Android.mk
index ff43abdcd..f1b15dfc3 100644
--- a/mm-video-v4l2/vidc/venc/Android.mk
+++ b/mm-video-v4l2/vidc/venc/Android.mk
@@ -17,7 +17,6 @@ libmm-venc-def += -DENABLE_DEBUG_ERROR
 libmm-venc-def += -UINPUT_BUFFER_LOG
 libmm-venc-def += -UOUTPUT_BUFFER_LOG
 libmm-venc-def += -USINGLE_ENCODER_INSTANCE
-libmm-venc-def += -Werror
 libmm-venc-def += -D_ANDROID_ICS_
 libmm-venc-def += -D_MSM8974_
 

project packages/apps/AudioFX/
diff --git a/Android.mk b/Android.mk
index f44369e..792d600 100644
--- a/Android.mk
+++ b/Android.mk
@@ -1,36 +1 @@
-ifneq ($(TARGET_EXCLUDES_AUDIOFX),true)
-
-LOCAL_PATH := $(call my-dir)
-
-include $(CLEAR_VARS)
-LOCAL_PACKAGE_NAME	:= AudioFX
-LOCAL_MODULE_TAGS	:= optional
-
-LOCAL_SRC_FILES := $(call all-java-files-under, src)
-
-LOCAL_STATIC_ANDROID_LIBRARIES := \
-    androidx.core_core \
-    androidx.legacy_legacy-support-v4
-
-LOCAL_STATIC_JAVA_LIBRARIES := \
-    org.lineageos.platform.internal
-
-LOCAL_RESOURCE_DIR := $(LOCAL_PATH)/res
-LOCAL_USE_AAPT2 := true
-
-LOCAL_PROGUARD_ENABLED := disabled
-
-LOCAL_PRIVATE_PLATFORM_APIS := true
-
-LOCAL_PRIVILEGED_MODULE := true
-
-LOCAL_CERTIFICATE := platform
-
-include $(BUILD_PACKAGE)
-
-# Use the following include to make our test apk.
-ifeq (,$(ONE_SHOT_MAKEFILE))
-include $(call all-makefiles-under,$(LOCAL_PATH))
-endif
-
-endif # TARGET_EXCLUDES_AUDIOFX
+#

project packages/apps/Bluetooth/
diff --git a/Android.mk b/Android.mk
index a34d2a34f..07396867b 100644
--- a/Android.mk
+++ b/Android.mk
@@ -58,8 +58,8 @@ include $(call all-makefiles-under,$(LOCAL_PATH))
 
 include $(CLEAR_VARS)
 
-COMMON_LIBS_PATH := ../../../../../prebuilts/tools/common/m2/repository
-ROOM_LIBS_PATH := ../../lib/room
+COMMON_LIBS_PATH := ../../../prebuilts/tools/common/m2/repository
+ROOM_LIBS_PATH := lib/room
 
 LOCAL_PREBUILT_STATIC_JAVA_LIBRARIES := \
         bt-androidx-annotation-nodeps:$(ROOM_LIBS_PATH)/annotation-1.0.0-beta01.jar \
@@ -72,6 +72,6 @@ LOCAL_PREBUILT_STATIC_JAVA_LIBRARIES := \
         bt-javapoet-nodeps:$(COMMON_LIBS_PATH)/com/squareup/javapoet/1.8.0/javapoet-1.8.0.jar \
         bt-kotlin-metadata-nodeps:$(COMMON_LIBS_PATH)/me/eugeniomarletti/kotlin-metadata/1.2.1/kotlin-metadata-1.2.1.jar \
         bt-sqlite-jdbc-nodeps:$(COMMON_LIBS_PATH)/org/xerial/sqlite-jdbc/3.20.1/sqlite-jdbc-3.20.1.jar \
-        bt-jetbrain-nodeps:../../../../../prebuilts/tools/common/m2/repository/org/jetbrains/annotations/13.0/annotations-13.0.jar
+        bt-jetbrain-nodeps:../../../prebuilts/tools/common/m2/repository/org/jetbrains/annotations/13.0/annotations-13.0.jar
 
 include $(BUILD_HOST_PREBUILT)
diff --git a/tests/Android.mk b/tests/Android.mk
index fd297e353..792d60054 100644
--- a/tests/Android.mk
+++ b/tests/Android.mk
@@ -1,5 +1 @@
-LOCAL_PATH := $(call my-dir)
-include $(CLEAR_VARS)
-
-# Include all makefiles in subdirectories
-include $(call all-makefiles-under,$(LOCAL_PATH))
+#

project packages/apps/Calendar/
diff --git a/Android.mk b/Android.mk
index 4b5744a7..792d6005 100644
--- a/Android.mk
+++ b/Android.mk
@@ -1,54 +1 @@
-LOCAL_PATH:= $(call my-dir)
-include $(CLEAR_VARS)
-
-# Include res dir from chips
-chips_dir := ../../../frameworks/opt/chips/res
-color_picker_dir := ../../../frameworks/opt/colorpicker/res
-datetimepicker_dir := ../../../frameworks/opt/datetimepicker/res
-timezonepicker_dir := ../../../frameworks/opt/timezonepicker/res
-res_dirs := $(chips_dir) $(color_picker_dir) $(datetimepicker_dir) $(timezonepicker_dir) res
-src_dirs := src
-
-LOCAL_JACK_COVERAGE_INCLUDE_FILTER := com.android.calendar.*
-
-LOCAL_MODULE_TAGS := optional
-
-LOCAL_SRC_FILES := $(call all-java-files-under,$(src_dirs))
-
-# bundled
-#LOCAL_STATIC_JAVA_LIBRARIES += \
-#        android-common \
-#        libchips \
-#        calendar-common
-
-# unbundled
-LOCAL_STATIC_JAVA_LIBRARIES := \
-        android-common \
-        libchips \
-        colorpicker \
-        android-opt-datetimepicker \
-        android-opt-timezonepicker \
-        androidx.legacy_legacy-support-v4 \
-        calendar-common
-
-LOCAL_SDK_VERSION := current
-
-LOCAL_RESOURCE_DIR := $(addprefix $(LOCAL_PATH)/, $(res_dirs))
-
-LOCAL_PACKAGE_NAME := Calendar
-
-LOCAL_PROGUARD_FLAG_FILES := proguard.flags \
-                             ../../../frameworks/opt/datetimepicker/proguard.flags
-
-LOCAL_PRODUCT_MODULE := true
-
-LOCAL_AAPT_FLAGS := --auto-add-overlay
-LOCAL_AAPT_FLAGS += --extra-packages com.android.ex.chips
-LOCAL_AAPT_FLAGS += --extra-packages com.android.colorpicker
-LOCAL_AAPT_FLAGS += --extra-packages com.android.datetimepicker
-LOCAL_AAPT_FLAGS += --extra-packages com.android.timezonepicker
-
-include $(BUILD_PACKAGE)
-
-# Use the following include to make our test apk.
-include $(call all-makefiles-under,$(LOCAL_PATH))
+#

project packages/apps/Email/
diff --git a/Android.mk b/Android.mk
index 4fa104752..c54a7949e 100644
--- a/Android.mk
+++ b/Android.mk
@@ -12,72 +12,3 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-LOCAL_PATH := $(call my-dir)
-
-# Build the Email application itself, along with its tests and tests for the emailcommon
-# static library.  All tests can be run via runtest email
-
-include $(CLEAR_VARS)
-
-unified_email_dir := ../UnifiedEmail
-res_dir := res $(unified_email_dir)/res
-
-LOCAL_MODULE_TAGS := optional
-
-LOCAL_SRC_FILES := $(call all-java-files-under, $(unified_email_dir)/src)
-LOCAL_SRC_FILES += $(call all-java-files-under, src/com/android)
-LOCAL_SRC_FILES += $(call all-java-files-under, provider_src/com/android)
-LOCAL_SRC_FILES += $(call all-java-files-under, src/com/beetstra)
-
-LOCAL_RESOURCE_DIR := $(addprefix $(LOCAL_PATH)/, $(res_dir))
-LOCAL_USE_AAPT2 := true
-
-# Use assets dir from UnifiedEmail
-# (the default package target doesn't seem to deal with multiple asset dirs)
-LOCAL_ASSET_DIR := $(LOCAL_PATH)/$(unified_email_dir)/assets
-
-LOCAL_AAPT_FLAGS := --auto-add-overlay
-LOCAL_AAPT_FLAGS += --extra-packages com.android.mail:com.android.email:com.android.emailcommon
-
-LOCAL_STATIC_ANDROID_LIBRARIES := \
-    com.android.emailcommon \
-    libchips \
-    libphotoviewer_appcompat \
-    android-opt-bitmap \
-    androidx.core_core \
-    androidx.media_media \
-    androidx.legacy_legacy-support-core-utils \
-    androidx.legacy_legacy-support-core-ui \
-    androidx.fragment_fragment \
-    androidx.appcompat_appcompat \
-    androidx.gridlayout_gridlayout \
-    androidx.legacy_legacy-support-v13
-
-LOCAL_STATIC_JAVA_LIBRARIES := \
-    androidx.annotation_annotation \
-    android-common \
-    guava \
-    owasp-html-sanitizer
-
-LOCAL_JAVA_LIBRARIES := org.apache.http.legacy
-
-LOCAL_PACKAGE_NAME := Email
-
-LOCAL_PRODUCT_MODULE := true
-
-LOCAL_PROGUARD_FLAG_FILES := proguard.flags $(unified_email_dir)/proguard.flags
-LOCAL_PROGUARD_FLAG_FILES += proguard-test.flags
-
-LOCAL_SDK_VERSION := current
-
-LOCAL_MODULE_PATH := $(TARGET_OUT_PRODUCT_APPS)
-
-LOCAL_COMPATIBILITY_SUITE := general-tests
-
-include $(BUILD_PACKAGE)
-
-# only include rules to build other stuff for the original package, not the derived package.
-ifeq ($(strip $(LOCAL_PACKAGE_OVERRIDES)),)
-# additionally, build unit tests in a separate .apk
-include $(call all-makefiles-under,$(LOCAL_PATH))
-endif

project packages/apps/Exchange/
diff --git a/Android.mk b/Android.mk
index fb1e6d14..c54a7949 100644
--- a/Android.mk
+++ b/Android.mk
@@ -12,39 +12,3 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-LOCAL_PATH:= $(call my-dir)
-include $(CLEAR_VARS)
-
-#
-# Exchange2
-#
-LOCAL_MODULE_TAGS := optional
-
-# Include res dir from emailcommon
-emailcommon_dir := ../Email/emailcommon
-res_dir := res $(emailcommon_dir)/res
-
-LOCAL_RESOURCE_DIR := $(addprefix $(LOCAL_PATH)/, $(res_dir))
-
-LOCAL_AAPT_FLAGS := --auto-add-overlay
-LOCAL_AAPT_FLAGS += --extra-packages com.android.emailcommon
-
-LOCAL_SRC_FILES := $(call all-java-files-under, src)
-LOCAL_SRC_FILES += $(call all-java-files-under, build/src)
-
-LOCAL_STATIC_ANDROID_LIBRARIES := androidx.core_core
-LOCAL_STATIC_JAVA_LIBRARIES := android-common com.android.emailcommon
-LOCAL_STATIC_JAVA_LIBRARIES += calendar-common
-
-LOCAL_PACKAGE_NAME := Exchange2
-LOCAL_OVERRIDES_PACKAGES := Exchange
-
-LOCAL_PROGUARD_FLAG_FILES := proguard.flags
-LOCAL_SDK_VERSION := 19
-
-LOCAL_JACK_COVERAGE_INCLUDE_FILTER += com.android.exchange.*
-
-include $(BUILD_PACKAGE)
-
-# additionally, build unit tests in a separate .apk
-include $(call all-makefiles-under,$(LOCAL_PATH))

project packages/apps/Stk/
diff --git a/Android.bp b/Android.bp
index a2dbc03..89b963c 100644
--- a/Android.bp
+++ b/Android.bp
@@ -1,10 +1,2 @@
 // Copyright 2007-2008 The Android Open Source Project
 
-android_app {
-    name: "Stk",
-    libs: ["telephony-common"],
-    static_libs: ["android-support-v4"],
-    srcs: ["**/*.java"],
-    platform_apis: true,
-    certificate: "platform",
-}

project packages/apps/Traceur/
diff --git a/Android.bp b/Android.bp
index 875c34c..8337712 100644
--- a/Android.bp
+++ b/Android.bp
@@ -1,30 +1 @@
-android_app {
-    name: "Traceur",
-    platform_apis: true,
-    certificate: "platform",
-    required: ["notify_traceur.sh"],
-    optimize: {
-        proguard_flags_files: ["proguard.flags"],
-    },
-    static_libs: [
-        "androidx.leanback_leanback",
-        "androidx.leanback_leanback-preference",
-        "androidx.legacy_legacy-preference-v14",
-        "androidx.appcompat_appcompat",
-        "androidx.preference_preference",
-        "androidx.recyclerview_recyclerview",
-        "androidx.legacy_legacy-support-v4",
-    ],
-    resource_dirs: ["res"],
-    srcs: ["src/**/*.java"],
-    init_rc: ["traceur.rc"],
-    defaults: [
-        "SettingsLibDefaults",
-    ],
-}
-
-sh_binary {
-    name: "notify_traceur.sh",
-    src: "notify_traceur.sh",
-}
-
+//

project packages/apps/Updater/
diff --git a/Android.mk b/Android.mk
index b74fe64..792d600 100644
--- a/Android.mk
+++ b/Android.mk
@@ -1,45 +1 @@
-LOCAL_PATH:= $(call my-dir)
-
-include $(CLEAR_VARS)
-
-LOCAL_MODULE_TAGS := optional
-
-LOCAL_SRC_FILES := $(call all-java-files-under, src)
-
-LOCAL_USE_AAPT2 := true
-
-LOCAL_STATIC_ANDROID_LIBRARIES := \
-    com.google.android.material_material \
-    androidx.core_core \
-    androidx.appcompat_appcompat \
-    androidx.cardview_cardview \
-    androidx.preference_preference \
-    androidx.recyclerview_recyclerview \
-
-LOCAL_RESOURCE_DIR := \
-    $(LOCAL_PATH)/res
-
-LOCAL_PACKAGE_NAME := Updater
-LOCAL_PRIVATE_PLATFORM_APIS := true
-LOCAL_PRIVILEGED_MODULE := true
-LOCAL_PROGUARD_FLAG_FILES := proguard.flags
-
-include $(BUILD_PACKAGE)
-
-
-include $(CLEAR_VARS)
-LOCAL_MODULE := UpdaterStudio
-LOCAL_MODULE_CLASS := FAKE
-LOCAL_MODULE_SUFFIX := -timestamp
-updater_system_deps := $(call java-lib-deps,framework)
-updater_system_libs_path := $(abspath $(LOCAL_PATH))/system_libs
-
-include $(BUILD_SYSTEM)/base_rules.mk
-
-$(LOCAL_BUILT_MODULE): $(updater_system_deps)
-	$(hide) mkdir -p $(updater_system_libs_path)
-	$(hide) rm -rf $(updater_system_libs_path)/*.jar
-	$(hide) cp $(updater_system_deps) $(updater_system_libs_path)/framework.jar
-	$(hide) echo "Fake: $@"
-	$(hide) mkdir -p $(dir $@)
-	$(hide) touch $@
+#

project packages/providers/BookmarkProvider/
diff --git a/Android.bp b/Android.bp
index 0f73493..1a24490 100644
--- a/Android.bp
+++ b/Android.bp
@@ -12,8 +12,3 @@
 // License for the specific language governing permissions and limitations under
 // the License.
 
-android_app {
-    name: "BookmarkProvider",
-    srcs: ["src/**/*.java"],
-    platform_apis: true,
-}

project packages/screensavers/Basic/
diff --git a/Android.bp b/Android.bp
index c9a65a8..8337712 100644
--- a/Android.bp
+++ b/Android.bp
@@ -1,6 +1 @@
-android_app {
-    name: "BasicDreams",
-    // Only compile source java files in this apk.
-    srcs: ["src/**/*.java"],
-    platform_apis: true,
-}
+//

project packages/screensavers/PhotoTable/
diff --git a/Android.bp b/Android.bp
index 0097373..8337712 100644
--- a/Android.bp
+++ b/Android.bp
@@ -1,7 +1 @@
-android_app {
-    name: "PhotoTable",
-    // Only compile source java files in this apk.
-    srcs: ["src/**/*.java"],
-    product_specific: true,
-    sdk_version: "current",
-}
+//

project system/apex/
diff --git a/apexd/Android.bp b/apexd/Android.bp
index 6c6f75d..e9aed38 100644
--- a/apexd/Android.bp
+++ b/apexd/Android.bp
@@ -13,7 +13,6 @@ cc_defaults {
   cflags: [
     "-Wall",
     "-Wextra",
-    "-Werror",
     "-Wno-unused-parameter",
 
     // Some extra flags.
@@ -287,49 +286,3 @@ genrule {
        "$(genDir)/apex.apexd_test_corrupt_apex.apex"
 }
 
-cc_test {
-  name: "apexservice_test",
-  defaults: ["apex_defaults"],
-  cflags: [
-    // Otherwise libgmock won't compile.
-    "-Wno-used-but-marked-unused",
-  ],
-  data: [
-    ":apex.apexd_test",
-    ":apex.apexd_test_different_app",
-    ":apex.apexd_test_v2",
-    ":apex.apexd_test_no_inst_key",
-    ":apex.apexd_test_preinstall",
-    ":apex.apexd_test_postinstall",
-    ":apex.apexd_test_prepostinstall.fail",
-    ":gen_bad_apexes",
-    ":gen_corrupt_apex",
-    ":com.android.apex.cts.shim.v1_prebuilt",
-    ":com.android.apex.cts.shim.v2_prebuilt",
-    ":com.android.apex.cts.shim.v2_wrong_sha_prebuilt",
-    ":com.android.apex.cts.shim.v2_additional_file_prebuilt",
-    ":com.android.apex.cts.shim.v2_additional_folder_prebuilt",
-    ":com.android.apex.cts.shim.v2_with_pre_install_hook_prebuilt",
-    ":com.android.apex.cts.shim.v2_with_post_install_hook_prebuilt",
-  ],
-  srcs: ["apexservice_test.cpp"],
-  host_supported: false,
-  compile_multilib: "first",
-  static_libs: [
-    "apex_aidl_interface-cpp",
-    "libapex",
-    "libapexd",
-    "libavb",
-    "libdm",
-    "libgmock",
-    "libvold_binder",
-  ],
-  shared_libs: [
-    "libbinder",
-    "libselinux",
-    "libutils",
-  ],
-  test_suites: ["device-tests"],
-  test_config: "apexservice_test_config.xml",
-}
-
diff --git a/shim/Android.bp b/shim/Android.bp
index 8573e1c..d190578 100644
--- a/shim/Android.bp
+++ b/shim/Android.bp
@@ -15,52 +15,3 @@
 // TODO: consider removing _prebuilt suffix from module names and make use of
 // 'prefer: true'
 
-prebuilt_apex {
-  name: "com.android.apex.cts.shim.v1_prebuilt",
-  src: "com.android.apex.cts.shim.v1.apex",
-  filename: "com.android.apex.cts.shim.apex",
-  prefer: true,
-  installable: true,
-}
-
-prebuilt_apex {
-  name: "com.android.apex.cts.shim.v2_prebuilt",
-  src: "com.android.apex.cts.shim.v2.apex",
-  filename: "com.android.apex.cts.shim.v2.apex",
-  installable: false,
-}
-
-prebuilt_apex {
-  name: "com.android.apex.cts.shim.v2_wrong_sha_prebuilt",
-  src: "com.android.apex.cts.shim.v2_wrong_sha.apex",
-  filename: "com.android.apex.cts.shim.v2_wrong_sha.apex",
-  installable: false,
-}
-
-prebuilt_apex {
-  name: "com.android.apex.cts.shim.v2_additional_file_prebuilt",
-  src: "com.android.apex.cts.shim.v2_additional_file.apex",
-  filename: "com.android.apex.cts.shim.v2_additional_file.apex",
-  installable: false,
-}
-
-prebuilt_apex {
-  name: "com.android.apex.cts.shim.v2_additional_folder_prebuilt",
-  src: "com.android.apex.cts.shim.v2_additional_folder.apex",
-  filename: "com.android.apex.cts.shim.v2_additional_folder.apex",
-  installable: false,
-}
-
-prebuilt_apex {
-  name: "com.android.apex.cts.shim.v2_with_pre_install_hook_prebuilt",
-  src: "com.android.apex.cts.shim.v2_with_pre_install_hook.apex",
-  filename: "com.android.apex.cts.shim.v2_with_pre_install_hook.apex",
-  installable: false,
-}
-
-prebuilt_apex {
-  name: "com.android.apex.cts.shim.v2_with_post_install_hook_prebuilt",
-  src: "com.android.apex.cts.shim.v2_with_post_install_hook.apex",
-  filename: "com.android.apex.cts.shim.v2_with_post_install_hook.apex",
-  installable: false,
-}
diff --git a/shim/build/Android.bp b/shim/build/Android.bp
index 71f847c..af99926 100644
--- a/shim/build/Android.bp
+++ b/shim/build/Android.bp
@@ -14,189 +14,3 @@
 
 // Build rules to build shim apexes.
 
-genrule {
-  name: "com.android.apex.cts.shim.pem",
-  out: ["com.android.apex.cts.shim.pem"],
-  cmd: "openssl genrsa -out $(out) 4096",
-}
-
-genrule {
-  name: "com.android.apex.cts.shim.pubkey",
-  srcs: [":com.android.apex.cts.shim.pem"],
-  out: ["com.android.apex.cts.shim.pubkey"],
-  tools: ["avbtool"],
-  cmd: "$(location avbtool) extract_public_key --key $(in) --output $(out)",
-}
-
-apex_key {
-  name: "com.android.apex.cts.shim.key",
-  private_key: ":com.android.apex.cts.shim.pem",
-  public_key: ":com.android.apex.cts.shim.pubkey",
-  installable: false,
-}
-
-genrule {
-  name: "generate_hash_of_dev_null",
-  out: ["hash.txt"],
-  cmd: "sha512sum -b /dev/null | cut -d' ' -f1 | tee $(out)",
-}
-
-prebuilt_etc {
-  name: "hash_of_dev_null",
-  src: ":generate_hash_of_dev_null",
-  filename: "hash.txt",
-  installable: false,
-}
-
-apex {
-  name: "com.android.apex.cts.shim.v3",
-  manifest: "manifest_v3.json",
-  file_contexts: "apex.test",
-  key: "com.android.apex.cts.shim.key",
-  prebuilts: ["hash_of_dev_null"],
-  installable: false,
-}
-
-apex {
-  name: "com.android.apex.cts.shim.v2",
-  manifest: "manifest_v2.json",
-  file_contexts: "apex.test",
-  key: "com.android.apex.cts.shim.key",
-  prebuilts: ["hash_of_dev_null"],
-  installable: false,
-}
-
-genrule {
-  name: "generate_empty_hash",
-  out: ["hash.txt"],
-  cmd: "touch $(out)",
-}
-
-prebuilt_etc {
-  name: "empty_hash",
-  src: ":generate_empty_hash",
-  filename: "hash.txt",
-  installable: false,
-}
-
-// Use empty hash.txt to make sure that this apex has wrong SHA512, hence trying
-// to stage it should fail.
-apex {
-  name: "com.android.apex.cts.shim.v2_wrong_sha",
-  manifest: "manifest_v2.json",
-  file_contexts: "apex.test",
-  key: "com.android.apex.cts.shim.key",
-  prebuilts: ["empty_hash"],
-  installable: false,
-}
-
-prebuilt_etc {
-  name: "apex_shim_additional_file",
-  src: "additional_file",
-  filename: "additional_file",
-  installable: false,
-}
-
-apex {
-  name: "com.android.apex.cts.shim.v2_additional_file",
-  manifest: "manifest_v2.json",
-  file_contexts: "apex.test",
-  key: "com.android.apex.cts.shim.key",
-  prebuilts: ["hash_of_dev_null", "apex_shim_additional_file"],
-  installable: false,
-}
-
-prebuilt_etc {
-  name: "apex_shim_additional_folder",
-  src: "additional_file",
-  filename: "additional_file",
-  sub_dir: "additional_folder",
-  installable: false,
-}
-
-apex {
-  name: "com.android.apex.cts.shim.v2_additional_folder",
-  manifest: "manifest_v2.json",
-  file_contexts: "apex.test",
-  key: "com.android.apex.cts.shim.key",
-  prebuilts: ["hash_of_dev_null", "apex_shim_additional_folder"],
-  installable: false,
-}
-
-apex {
-  name: "com.android.apex.cts.shim.v2_with_pre_install_hook",
-  manifest: "manifest_v2_with_pre_install_hook.json",
-  file_contexts: "apex.test",
-  key: "com.android.apex.cts.shim.key",
-  prebuilts: ["hash_of_dev_null"],
-  installable: false,
-}
-
-apex {
-  name: "com.android.apex.cts.shim.v2_with_post_install_hook",
-  manifest: "manifest_v2_with_post_install_hook.json",
-  file_contexts: "apex.test",
-  key: "com.android.apex.cts.shim.key",
-  prebuilts: ["hash_of_dev_null"],
-  installable: false,
-}
-
-genrule {
-  name: "generate_hash_v1",
-  srcs: [
-    ":com.android.apex.cts.shim.v2",
-    ":com.android.apex.cts.shim.v2_additional_file",
-    ":com.android.apex.cts.shim.v2_additional_folder",
-    ":com.android.apex.cts.shim.v2_with_pre_install_hook",
-    ":com.android.apex.cts.shim.v2_with_post_install_hook",
-    ":com.android.apex.cts.shim.v3",
-  ],
-  out: ["hash.txt"],
-  cmd: "sha512sum -b $(in) | cut -d' ' -f1 | tee $(out)",
-}
-
-prebuilt_etc {
-  name: "hash_v1",
-  src: ":generate_hash_v1",
-  filename: "hash.txt",
-  installable: false,
-}
-
-apex {
-  name: "com.android.apex.cts.shim.v1",
-  manifest: "manifest.json",
-  file_contexts: "apex.test",
-  key: "com.android.apex.cts.shim.key",
-  prebuilts: ["hash_v1"],
-  installable: false,
-}
-
-genrule {
-  name: "com.android.apex.cts.shim_not_pre_installed.pem",
-  out: ["com.android.apex.cts.shim_not_pre_installed.pem"],
-  cmd: "openssl genrsa -out $(out) 4096",
-}
-
-genrule {
-  name: "com.android.apex.cts.shim_not_pre_installed.pubkey",
-  srcs: [":com.android.apex.cts.shim_not_pre_installed.pem"],
-  out: ["com.android.apex.cts.shim_not_pre_installed.pubkey"],
-  tools: ["avbtool"],
-  cmd: "$(location avbtool) extract_public_key --key $(in) --output $(out)",
-}
-
-apex_key {
-  name: "com.android.apex.cts.shim_not_pre_installed.key",
-  private_key: ":com.android.apex.cts.shim_not_pre_installed.pem",
-  public_key: ":com.android.apex.cts.shim_not_pre_installed.pubkey",
-  installable: false,
-}
-
-apex {
-  name: "com.android.apex.cts.shim_not_pre_installed",
-  manifest: "manifest_not_pre_installed.json",
-  file_contexts: "apex.test",
-  key: "com.android.apex.cts.shim_not_pre_installed.key",
-  prebuilts: ["hash_of_dev_null"],
-  installable: false,
-}

project system/core/
diff --git a/debuggerd/Android.bp b/debuggerd/Android.bp
index 2e226da6c..2b66a94ac 100644
--- a/debuggerd/Android.bp
+++ b/debuggerd/Android.bp
@@ -258,76 +258,6 @@ cc_benchmark {
     ],
 }
 
-cc_binary {
-    name: "crash_dump",
-    srcs: [
-        "crash_dump.cpp",
-        "util.cpp",
-    ],
-    defaults: ["debuggerd_defaults"],
-
-    compile_multilib: "both",
-    multilib: {
-        lib32: {
-            suffix: "32",
-        },
-        lib64: {
-            suffix: "64",
-        },
-    },
-
-    static_libs: [
-        "libtombstoned_client_static",
-        "libdebuggerd",
-        "libcutils",
-    ],
-
-    shared_libs: [
-        "libbase",
-        "liblog",
-        "libprocinfo",
-        "libunwindstack",
-    ],
-}
-
-cc_binary {
-    name: "debuggerd",
-    srcs: [
-        "debuggerd.cpp",
-    ],
-    defaults: ["debuggerd_defaults"],
-
-    shared_libs: [
-        "libbase",
-        "libdebuggerd_client",
-        "liblog",
-        "libprocinfo",
-    ],
-
-    local_include_dirs: ["include"],
-}
-
-cc_binary {
-    name: "tombstoned",
-    srcs: [
-        "util.cpp",
-        "tombstoned/intercept_manager.cpp",
-        "tombstoned/tombstoned.cpp",
-    ],
-    defaults: ["debuggerd_defaults"],
-
-    header_libs: ["libdebuggerd_common_headers"],
-
-    static_libs: [
-        "libbase",
-        "libcutils",
-        "libevent",
-        "liblog",
-    ],
-
-    init_rc: ["tombstoned/tombstoned.rc"],
-}
-
 subdirs = [
     "crasher",
 ]
diff --git a/healthd/Android.bp b/healthd/Android.bp
index 2cf6be96d..b744e96b2 100644
--- a/healthd/Android.bp
+++ b/healthd/Android.bp
@@ -10,7 +10,7 @@ cc_library_headers {
 cc_library_static {
     name: "libbatterymonitor",
     srcs: ["BatteryMonitor.cpp"],
-    cflags: ["-Wall", "-Werror"],
+    cflags: ["-Wall"],
     vendor_available: true,
     recovery_available: true,
     export_include_dirs: ["include"],
@@ -27,7 +27,6 @@ cc_defaults {
 
     cflags: [
         "-Wall",
-        "-Werror",
     ],
 
     static_libs: [
@@ -94,7 +93,6 @@ cc_library_static {
 
     cflags: [
         "-Wall",
-        "-Werror",
     ],
 
     header_libs: [
diff --git a/healthd/Android.mk b/healthd/Android.mk
index 4eb98dae0..fb50aad80 100644
--- a/healthd/Android.mk
+++ b/healthd/Android.mk
@@ -31,7 +31,6 @@ include $(BUILD_STATIC_LIBRARY)
 ### libhealthd_charger ###
 include $(CLEAR_VARS)
 
-LOCAL_CFLAGS := -Werror
 ifeq ($(strip $(BOARD_CHARGER_DISABLE_INIT_BLANK)),true)
 LOCAL_CFLAGS += -DCHARGER_DISABLE_INIT_BLANK
 endif
@@ -82,7 +81,6 @@ LOCAL_SRC_FILES := \
 LOCAL_MODULE := charger
 LOCAL_C_INCLUDES := $(LOCAL_PATH)/include
 
-LOCAL_CFLAGS := -Werror
 ifeq ($(strip $(LOCAL_CHARGER_NO_UI)),true)
 LOCAL_CFLAGS += -DCHARGER_NO_UI
 endif
@@ -140,7 +138,7 @@ LOCAL_MODULE_PATH := $(TARGET_RECOVERY_ROOT_OUT)/system/bin
 LOCAL_MODULE_STEM := charger
 
 LOCAL_C_INCLUDES := $(LOCAL_PATH)/include
-LOCAL_CFLAGS := -Wall -Werror
+LOCAL_CFLAGS := -Wall
 LOCAL_CFLAGS += -DCHARGER_NO_UI
 
 # charger.recovery doesn't link against libhealthd_{charger,draw} or libminui, since it doesn't need
@@ -174,7 +172,7 @@ include $(BUILD_EXECUTABLE)
 include $(CLEAR_VARS)
 LOCAL_MODULE := charger_test
 LOCAL_C_INCLUDES := $(LOCAL_PATH)/include
-LOCAL_CFLAGS := -Wall -Werror -DCHARGER_NO_UI
+LOCAL_CFLAGS := -Wall -DCHARGER_NO_UI
 LOCAL_STATIC_LIBRARIES := $(CHARGER_STATIC_LIBRARIES)
 LOCAL_SHARED_LIBRARIES := $(CHARGER_SHARED_LIBRARIES)
 LOCAL_SRC_FILES := \
diff --git a/rootdir/init.rc b/rootdir/init.rc
index 7559886d9..643c2059f 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -429,11 +429,6 @@ on post-fs-data
     # Make sure we have the device encryption key.
     installkey /data
 
-    # Start bootcharting as soon as possible after the data partition is
-    # mounted to collect more data.
-    mkdir /data/bootchart 0755 shell shell
-    bootchart start
-
     # Load fsverity keys. This needs to happen before apexd, as post-install of
     # APEXes may rely on keys.
     exec -- /system/bin/fsverity_init
@@ -494,7 +489,6 @@ on post-fs-data
     mkdir /data/misc/ethernet 0770 system system
     mkdir /data/misc/dhcp 0770 dhcp dhcp
     mkdir /data/misc/user 0771 root root
-    mkdir /data/misc/perfprofd 0775 root root
     # give system access to wpa_supplicant.conf for backup and restore
     chmod 0660 /data/misc/wifi/wpa_supplicant.conf
     mkdir /data/local 0751 root root
@@ -502,18 +496,10 @@ on post-fs-data
     mkdir /data/misc/audioserver 0700 audioserver audioserver
     mkdir /data/misc/cameraserver 0700 cameraserver cameraserver
     mkdir /data/misc/vold 0700 root root
-    mkdir /data/misc/boottrace 0771 system shell
-    mkdir /data/misc/update_engine 0700 root root
-    mkdir /data/misc/update_engine_log 02750 root log
-    mkdir /data/misc/trace 0700 root root
-    # create location to store surface and window trace files
-    mkdir /data/misc/wmtrace 0700 system system
     # profile file layout
     mkdir /data/misc/profiles 0771 system system
     mkdir /data/misc/profiles/cur 0771 system system
     mkdir /data/misc/profiles/ref 0771 system system
-    mkdir /data/misc/profman 0770 system shell
-    mkdir /data/misc/gcov 0770 root root
 
     mkdir /data/preloads 0775 system system
 
@@ -525,7 +511,6 @@ on post-fs-data
     # For security reasons, /data/local/tmp should always be empty.
     # Do not place files or directories in /data/local/tmp
     mkdir /data/local/tmp 0771 shell shell
-    mkdir /data/local/traces 0777 shell shell
     mkdir /data/data 0771 system system
     mkdir /data/app-private 0771 system system
     mkdir /data/app-ephemeral 0771 system system
@@ -533,18 +518,9 @@ on post-fs-data
     mkdir /data/app-lib 0771 system system
     mkdir /data/app 0771 system system
     mkdir /data/property 0700 root root
-    mkdir /data/tombstones 0771 system system
-    mkdir /data/vendor/tombstones 0771 root root
-    mkdir /data/vendor/tombstones/wifi 0771 wifi wifi
 
     # create dalvik-cache, so as to enforce our permissions
     mkdir /data/dalvik-cache 0771 root root
-    # create the A/B OTA directory, so as to enforce our permissions
-    mkdir /data/ota 0771 root root
-
-    # create the OTA package directory. It will be accessed by GmsCore (cache
-    # group), update_engine and update_verifier.
-    mkdir /data/ota_package 0770 system cache
 
     # create resource-cache and double-check the perms
     mkdir /data/resource-cache 0771 system system
@@ -562,20 +538,11 @@ on post-fs-data
     # the following directory.
     mkdir /data/mediadrm 0770 mediadrm mediadrm
 
-    mkdir /data/anr 0775 system system
-
-    # NFC: create data/nfc for nv storage
-    mkdir /data/nfc 0770 nfc nfc
-    mkdir /data/nfc/param 0770 nfc nfc
-
     # Create all remaining /data root dirs so that they are made through init
     # and get proper encryption policy installed
     mkdir /data/backup 0700 system system
-    mkdir /data/ss 0700 system system
 
     mkdir /data/system 0775 system system
-    mkdir /data/system/dropbox 0700 system system
-    mkdir /data/system/heapdump 0700 system system
     mkdir /data/system/users 0775 system system
 
     mkdir /data/system_de 0770 system system

project system/sepolicy/
diff --git a/Android.mk b/Android.mk
index dadd7b0a..b07223bf 100644
--- a/Android.mk
+++ b/Android.mk
@@ -7,6 +7,7 @@ include $(CLEAR_VARS)
 
 MLS_SENS=1
 MLS_CATS=1024
+SELINUX_IGNORE_NEVERALLOWS := true
 
 ifdef BOARD_SEPOLICY_UNION
 $(warning BOARD_SEPOLICY_UNION is no longer required - all files found in BOARD_SEPOLICY_DIRS are implicitly unioned; please remove from your BoardConfig.mk or other .mk file.)
@@ -81,10 +82,10 @@ ifndef BOARD_SEPOLICY_VERS
 BOARD_SEPOLICY_VERS := $(PLATFORM_SEPOLICY_VERSION)
 endif
 
-NEVERALLOW_ARG :=
+NEVERALLOW_ARG := -N
 ifeq ($(SELINUX_IGNORE_NEVERALLOWS),true)
 ifeq ($(TARGET_BUILD_VARIANT),user)
-$(error SELINUX_IGNORE_NEVERALLOWS := true cannot be used in user builds)
+$(warning SELINUX_IGNORE_NEVERALLOWS := true cannot be used in user builds)
 endif
 $(warning Be careful when using the SELINUX_IGNORE_NEVERALLOWS flag. \
           It does not work in user builds and using it will \
@@ -980,10 +981,9 @@ $(built_sepolicy_neverallows)
 	$(hide) $(HOST_OUT_EXECUTABLES)/sepolicy-analyze $@.tmp permissive > $@.permissivedomains
 	$(hide) if [ "$(TARGET_BUILD_VARIANT)" = "user" -a -s $@.permissivedomains ]; then \
 		echo "==========" 1>&2; \
-		echo "ERROR: permissive domains not allowed in user builds" 1>&2; \
+		echo "WARNING: permissive domains not allowed in user builds" 1>&2; \
 		echo "List of invalid domains:" 1>&2; \
 		cat $@.permissivedomains 1>&2; \
-		exit 1; \
 		fi
 	$(hide) mv $@.tmp $@
 
@@ -1034,10 +1034,9 @@ $(LOCAL_BUILT_MODULE): $(sepolicy.recovery.conf) $(HOST_OUT_EXECUTABLES)/checkpo
 	$(hide) $(HOST_OUT_EXECUTABLES)/sepolicy-analyze $@.tmp permissive > $@.permissivedomains
 	$(hide) if [ "$(TARGET_BUILD_VARIANT)" = "user" -a -s $@.permissivedomains ]; then \
 		echo "==========" 1>&2; \
-		echo "ERROR: permissive domains not allowed in user builds" 1>&2; \
+		echo "WARNING: permissive domains not allowed in user builds" 1>&2; \
 		echo "List of invalid domains:" 1>&2; \
 		cat $@.permissivedomains 1>&2; \
-		exit 1; \
 		fi
 	$(hide) mv $@.tmp $@
 

project vendor/lineage/
diff --git a/bootanimation/Android.mk b/bootanimation/Android.mk
index 2bc78ceb..34316ed5 100644
--- a/bootanimation/Android.mk
+++ b/bootanimation/Android.mk
@@ -15,52 +15,3 @@
 # limitations under the License.
 #
 
-ifeq ($(TARGET_SCREEN_WIDTH),)
-    $(warning TARGET_SCREEN_WIDTH is not set, using default value: 1080)
-    TARGET_SCREEN_WIDTH := 1080
-endif
-ifeq ($(TARGET_SCREEN_HEIGHT),)
-    $(warning TARGET_SCREEN_HEIGHT is not set, using default value: 1920)
-    TARGET_SCREEN_HEIGHT := 1920
-endif
-
-TARGET_GENERATED_BOOTANIMATION := $(TARGET_OUT_INTERMEDIATES)/BOOTANIMATION/bootanimation.zip
-$(TARGET_GENERATED_BOOTANIMATION): INTERMEDIATES := $(TARGET_OUT_INTERMEDIATES)/BOOTANIMATION
-$(TARGET_GENERATED_BOOTANIMATION): $(SOONG_ZIP)
-	@echo "Building bootanimation.zip"
-	@rm -rf $(dir $@)
-	@mkdir -p $(dir $@)
-	$(hide) tar xfp vendor/lineage/bootanimation/bootanimation.tar -C $(INTERMEDIATES)
-	$(hide) if [ $(TARGET_SCREEN_HEIGHT) -lt $(TARGET_SCREEN_WIDTH) ]; then \
-	    IMAGEWIDTH=$(TARGET_SCREEN_HEIGHT); \
-	else \
-	    IMAGEWIDTH=$(TARGET_SCREEN_WIDTH); \
-	fi; \
-	IMAGESCALEWIDTH=$$IMAGEWIDTH; \
-	IMAGESCALEHEIGHT=$$(expr $$IMAGESCALEWIDTH / 3); \
-	if [ "$(TARGET_BOOTANIMATION_HALF_RES)" = "true" ]; then \
-	    IMAGEWIDTH="$$(expr "$$IMAGEWIDTH" / 2)"; \
-	fi; \
-	IMAGEHEIGHT=$$(expr $$IMAGEWIDTH / 3); \
-	RESOLUTION="$$IMAGEWIDTH"x"$$IMAGEHEIGHT"; \
-	for part_cnt in 0 1 2 3 4; do \
-	    mkdir -p $(INTERMEDIATES)/part$$part_cnt; \
-	done; \
-	prebuilts/tools-lineage/${HOST_OS}-x86/bin/mogrify -resize $$RESOLUTION -colors 250 $(INTERMEDIATES)/*/*.png; \
-	echo "$$IMAGESCALEWIDTH $$IMAGESCALEHEIGHT 60" > $(INTERMEDIATES)/desc.txt; \
-	cat vendor/lineage/bootanimation/desc.txt >> $(INTERMEDIATES)/desc.txt
-	$(hide) $(SOONG_ZIP) -L 0 -o $(TARGET_GENERATED_BOOTANIMATION) -C $(INTERMEDIATES) -D $(INTERMEDIATES)
-
-ifeq ($(TARGET_BOOTANIMATION),)
-    TARGET_BOOTANIMATION := $(TARGET_GENERATED_BOOTANIMATION)
-endif
-
-include $(CLEAR_VARS)
-LOCAL_MODULE := bootanimation.zip
-LOCAL_MODULE_CLASS := ETC
-LOCAL_MODULE_PATH := $(TARGET_OUT)/media
-
-include $(BUILD_SYSTEM)/base_rules.mk
-
-$(LOCAL_BUILT_MODULE): $(TARGET_BOOTANIMATION)
-	@cp $(TARGET_BOOTANIMATION) $@
diff --git a/overlay/common/frameworks/base/packages/SystemUI/res/values/config.xml b/overlay/common/frameworks/base/packages/SystemUI/res/values/config.xml
index 6dc72ff1..b26117e1 100644
--- a/overlay/common/frameworks/base/packages/SystemUI/res/values/config.xml
+++ b/overlay/common/frameworks/base/packages/SystemUI/res/values/config.xml
@@ -27,5 +27,5 @@
     <!-- If true, enable the advance anti-falsing classifier on the lockscreen. On some devices it
          does not work well, particularly with noisy touchscreens. Note that disabling it may
          increase the rate of unintentional unlocks. -->
-    <bool name="config_lockscreenAntiFalsingClassifierEnabled">false</bool>
+    <bool name="config_lockscreenAntiFalsingClassifierEnabled">true</bool>
 </resources>
diff --git a/overlay/common/packages/apps/Settings/res/values/config.xml b/overlay/common/packages/apps/Settings/res/values/config.xml
index 651905da..5c815424 100644
--- a/overlay/common/packages/apps/Settings/res/values/config.xml
+++ b/overlay/common/packages/apps/Settings/res/values/config.xml
@@ -22,10 +22,10 @@
     <bool name="config_storage_manager_settings_enabled">true</bool>
 
     <!-- Whether to show a preference item for mobile plan -->
-    <bool name="config_show_mobile_plan">false</bool>
+    <bool name="config_show_mobile_plan">true</bool>
 
     <!-- Whether wallpaper attribution should be shown or not. -->
-    <bool name="config_show_wallpaper_attribution">false</bool>
+    <bool name="config_show_wallpaper_attribution">true</bool>
 
     <!-- Package name and fully-qualified class name for the wallpaper picker activity. -->
     <string name="config_wallpaper_picker_package" translatable="false">com.android.wallpaper</string>
diff --git a/prebuilt/common/etc/init/lineage-ssh.rc b/prebuilt/common/etc/init/lineage-ssh.rc
deleted file mode 100644
index edf38638..00000000
--- a/prebuilt/common/etc/init/lineage-ssh.rc
+++ /dev/null
@@ -1,3 +0,0 @@
-on post-fs-data
-    mkdir /data/ssh 0750 root shell
-    mkdir /data/ssh/empty 0600 root shell
diff --git a/prebuilt/common/etc/init/lineage-system.rc b/prebuilt/common/etc/init/lineage-system.rc
index 396ec73a..60f8f405 100644
--- a/prebuilt/common/etc/init/lineage-system.rc
+++ b/prebuilt/common/etc/init/lineage-system.rc
@@ -13,10 +13,3 @@ on post-fs-data
 on boot
     # Persistent properties (only created if persist exists)
     mkdir /persist/properties 0770 system system
-
-# bugreport is triggered by holding down volume down, volume up and power
-service bugreport /system/bin/dumpstate -d -p -B -z \
-        -o /data/user_de/0/com.android.shell/files/bugreports/bugreport
-    oneshot
-    disabled
-    keycodes 114 115 116
diff --git a/prebuilt/common/etc/init/lineage-updates.rc b/prebuilt/common/etc/init/lineage-updates.rc
deleted file mode 100644
index 3a9b4813..00000000
--- a/prebuilt/common/etc/init/lineage-updates.rc
+++ /dev/null
@@ -1,4 +0,0 @@
-on post-fs-data
-    # Create an additional OTA package directory that unlike /data/ota_package
-    # will not be touched by GmsCore.
-    mkdir /data/lineageos_updates 0770 system cache
