project build/make/
diff --git a/core/Makefile b/core/Makefile
index 404bb5413..92b010ec4 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -222,9 +222,9 @@ endif
 # Both of these tags will be removed and replaced with "release-keys"
 # when the target-files is signed in a post-build step.
 ifeq ($(DEFAULT_SYSTEM_DEV_CERTIFICATE),build/target/product/security/testkey)
-BUILD_KEYS := test-keys
+BUILD_KEYS := release-keys
 else
-BUILD_KEYS := dev-keys
+BUILD_KEYS := release-keys
 endif
 BUILD_VERSION_TAGS += $(BUILD_KEYS)
 BUILD_VERSION_TAGS := $(subst $(space),$(comma),$(sort $(BUILD_VERSION_TAGS)))

diff --git a/core/binary.mk b/core/binary.mk
index d2c8c1eaf..bd0ce2985 100644
--- a/core/binary.mk
+++ b/core/binary.mk
@@ -594,7 +594,7 @@ ifeq ($(NATIVE_COVERAGE),true)
         # with `--gcov-tool /usr/bin/gcov-4.6`.
         #
         # http://stackoverflow.com/questions/17758126/clang-code-coverage-invalid-output
-        my_cflags += --coverage -O0
+        my_cflags += --coverage -O3 -g0
         my_ldflags += --coverage
     endif

@@ -797,7 +797,7 @@ endif

 # Turn on all warnings and warnings as errors for RS compiles.
 # This can be disabled with LOCAL_RENDERSCRIPT_FLAGS := -Wno-error
-renderscript_flags := -Wall -Werror
+renderscript_flags := -Wall
 renderscript_flags += $(LOCAL_RENDERSCRIPT_FLAGS)
 # -m32 or -m64
 renderscript_flags += -m$(my_32_64_bit_suffix)
@@ -1664,7 +1664,7 @@ my_cflags += -Wno-attributes
 endif

 ifeq ($(my_fdo_build), true)
-  my_cflags := $(patsubst -Os,-O2,$(my_cflags))
+  my_cflags := $(patsubst -O3,-g0,$(my_cflags))
   fdo_incompatible_flags := -fno-early-inlining -finline-limit=%
   my_cflags := $(filter-out $(fdo_incompatible_flags),$(my_cflags))
 endif
@@ -1706,10 +1706,10 @@ ifneq ($(LOCAL_MODULE_MAKEFILE),$(SOONG_ANDROID_MK))
         $(eval MODULES_USING_WNO_ERROR := $(MODULES_USING_WNO_ERROR) $(LOCAL_MODULE_MAKEFILE):$(LOCAL_MODULE))
       else
         # Issue warning if -Werror is not used. Add it.
-        ifeq (,$(filter -Werror,$(my_all_cflags)))
+        ifeq (,$(filter -Wall,$(my_all_cflags)))
           # Add -Wall -Werror unless the project is in the WARNING_ALLOWED project list.
           ifeq (,$(strip $(call find_warning_allowed_projects,$(LOCAL_PATH))))
-            my_cflags := -Wall -Werror $(my_cflags)
+            my_cflags := -Wall $(my_cflags)
           else
             $(eval MODULES_ADDED_WALL := $(MODULES_ADDED_WALL) $(LOCAL_MODULE_MAKEFILE):$(LOCAL_MODULE))
             my_cflags := -Wall $(my_cflags)

diff --git a/core/host_test_internal.mk b/core/host_test_internal.mk
index 42e01e1e4..e8955fa6c 100644
--- a/core/host_test_internal.mk
+++ b/core/host_test_internal.mk
@@ -7,7 +7,7 @@ ifeq ($(LOCAL_GTEST),true)
   LOCAL_CFLAGS_linux += -DGTEST_OS_LINUX
   LOCAL_CFLAGS_darwin += -DGTEST_OS_MAC

-  LOCAL_CFLAGS += -DGTEST_HAS_STD_STRING -O0 -g
+  LOCAL_CFLAGS += -DGTEST_HAS_STD_STRING -O3 -g0

   LOCAL_STATIC_LIBRARIES += libgtest_main_host libgtest_host
 endif

diff --git a/core/java_renderscript.mk b/core/java_renderscript.mk
index 191b3be6a..34f3b9ea8 100644
--- a/core/java_renderscript.mk
+++ b/core/java_renderscript.mk
@@ -46,7 +46,7 @@ endif

 # Turn on all warnings and warnings as errors for RS compiles.
 # This can be disabled with LOCAL_RENDERSCRIPT_FLAGS := -Wno-error
-renderscript_flags := -Wall -Werror
+renderscript_flags := -Wall
 renderscript_flags += $(LOCAL_RENDERSCRIPT_FLAGS)

 # prepend the RenderScript system include path

diff --git a/core/main.mk b/core/main.mk
index 97ec295f8..363c1e302 100644
--- a/core/main.mk
+++ b/core/main.mk
@@ -307,8 +307,6 @@ endif # !user_variant
 ifeq (true,$(strip $(enable_target_debugging)))
   # Target is more debuggable and adbd is on by default
   ADDITIONAL_DEFAULT_PROPERTIES += ro.debuggable=1
-  # Enable Dalvik lock contention logging.
-  ADDITIONAL_BUILD_PROPERTIES += dalvik.vm.lockprof.threshold=500
   # Include the debugging/testing OTA keys in this build.
   INCLUDE_TEST_OTA_KEYS := true
 else # !enable_target_debugging
@@ -328,7 +326,7 @@ ifneq ($(filter ro.setupwizard.mode=ENABLED, $(call collapse-pairs, $(ADDITIONAL
 endif
 ifndef is_sdk_build
   # To speedup startup of non-preopted builds, don't verify or compile the boot image.
-  ADDITIONAL_BUILD_PROPERTIES += dalvik.vm.image-dex2oat-filter=verify-at-runtime
+  ADDITIONAL_BUILD_PROPERTIES += dalvik.vm.image-dex2oat-filter=speed
 endif
 endif

@@ -359,10 +357,6 @@ BUILD_WITHOUT_PV := true

 ADDITIONAL_BUILD_PROPERTIES += net.bt.name=Android

-# Sets the location that the runtime dumps stack traces to when signalled
-# with SIGQUIT. Stack trace dumping is turned on for all android builds.
-ADDITIONAL_BUILD_PROPERTIES += dalvik.vm.stack-trace-dir=/data/anr
-
 # ------------------------------------------------------------
 # Include vendor specific additions to build properties
 -include vendor/lineage/build/core/main.mk

diff --git a/target/product/core_minimal.mk b/target/product/core_minimal.mk
index b8e04e0ca..79f5ae37e 100644
--- a/target/product/core_minimal.mk
+++ b/target/product/core_minimal.mk
@@ -168,16 +168,6 @@ PRODUCT_COPY_FILES += $(call add-to-product-copy-files-if-exists,\
 PRODUCT_COPY_FILES += $(call add-to-product-copy-files-if-exists,\
     frameworks/base/dirty-image-objects-phone:system/etc/dirty-image-objects)

-# On userdebug builds, collect more tombstones by default.
-ifneq (,$(filter userdebug eng,$(TARGET_BUILD_VARIANT)))
-PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
-    tombstoned.max_tombstone_count=50
-endif
-
-PRODUCT_DEFAULT_PROPERTY_OVERRIDES += \
-    ro.logd.size.stats=64K \
-    log.tag.stats_log=I
-
 $(call inherit-product, $(SRC_TARGET_DIR)/product/runtime_libart.mk)
 $(call inherit-product, $(SRC_TARGET_DIR)/product/base.mk)

diff --git a/target/product/runtime_libart.mk b/target/product/runtime_libart.mk
index bda45244d..1b386cd27 100644
--- a/target/product/runtime_libart.mk
+++ b/target/product/runtime_libart.mk
@@ -76,24 +76,21 @@ PRODUCT_PROPERTY_OVERRIDES += \
 # On eng builds, make "boot" reasons only extract for faster turnaround.
 ifeq (eng,$(TARGET_BUILD_VARIANT))
     PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
-        pm.dexopt.first-boot=extract \
-        pm.dexopt.boot=extract
+        pm.dexopt.first-boot=speed \
+        pm.dexopt.boot=speed
 else
     PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
-        pm.dexopt.first-boot=quicken \
-        pm.dexopt.boot=verify
+        pm.dexopt.first-boot=speed \
+        pm.dexopt.boot=speed
 endif

 # The install filter is speed-profile in order to enable the use of
 # profiles from the dex metadata files. Note that if a profile is not provided
 # or if it is empty speed-profile is equivalent to (quicken + empty app image).
 PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
-    pm.dexopt.install=speed-profile \
-    pm.dexopt.bg-dexopt=speed-profile \
-    pm.dexopt.ab-ota=speed-profile \
-    pm.dexopt.inactive=verify \
+    pm.dexopt.install=speed \
+    pm.dexopt.bg-dexopt=speed \
+    pm.dexopt.ab-ota=speed \
+    pm.dexopt.inactive=speed \
     pm.dexopt.shared=speed

-# Enable minidebuginfo generation unless overridden.
-PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
-    dalvik.vm.dex2oat-minidebuginfo=true

diff --git a/tools/acp/Android.bp b/tools/acp/Android.bp
index 64f5a1013..8b84418b3 100644
--- a/tools/acp/Android.bp
+++ b/tools/acp/Android.bp
@@ -5,7 +5,7 @@
 cc_binary_host {

     srcs: ["acp.c"],
-    cflags: ["-Wall", "-Werror"],
+    cflags: ["-Wall"],

     static_libs: ["libhost"],
     name: "acp",

diff --git a/tools/atree/Android.bp b/tools/atree/Android.bp
index 5fbe042ea..fc8a8aada 100644
--- a/tools/atree/Android.bp
+++ b/tools/atree/Android.bp
@@ -9,6 +9,6 @@ cc_binary_host {
         "files.cpp",
         "fs.cpp",
     ],
-    cflags: ["-Wall", "-Werror"],
+    cflags: ["-Wall"],
     static_libs: ["libhost"],
 }

diff --git a/tools/fs_config/Android.bp b/tools/fs_config/Android.bp
index 797cfe228..1c8c3dae3 100644
--- a/tools/fs_config/Android.bp
+++ b/tools/fs_config/Android.bp
@@ -19,7 +19,6 @@ cc_binary_host {
         "libcutils",
         "libselinux",
     ],
-    cflags: ["-Werror"],
 }

 // -----------------------------------------------------------------------------
@@ -28,10 +27,9 @@ cc_binary_host {

 test_c_flags = [
     "-fstack-protector-all",
-    "-g",
+    "-g0",
     "-Wall",
     "-Wextra",
-    "-Werror",
     "-fno-builtin",
     "-DANDROID_FILESYSTEM_CONFIG=\"android_filesystem_config_test_data.h\"",
 ]

diff --git a/tools/fs_config/Android.mk b/tools/fs_config/Android.mk
index f6c8132b4..19b357315 100644
--- a/tools/fs_config/Android.mk
+++ b/tools/fs_config/Android.mk
@@ -75,7 +75,7 @@ LOCAL_SRC_FILES := fs_config_generate.c
 LOCAL_MODULE := fs_config_generate_$(TARGET_DEVICE)
 LOCAL_MODULE_CLASS := EXECUTABLES
 LOCAL_SHARED_LIBRARIES := libcutils
-LOCAL_CFLAGS := -Werror -Wno-error=\#warnings
+LOCAL_CFLAGS := -Wall

 ifneq ($(TARGET_FS_CONFIG_GEN),)
 system_android_filesystem_config := system/core/include/private/android_filesystem_config.h

diff --git a/tools/fs_get_stats/Android.bp b/tools/fs_get_stats/Android.bp
index 67742b836..f710d506e 100644
--- a/tools/fs_get_stats/Android.bp
+++ b/tools/fs_get_stats/Android.bp
@@ -1,7 +1,7 @@
 cc_binary_host {
     name: "fs_get_stats",
     srcs: ["fs_get_stats.c"],
-    cflags: ["-Wall", "-Werror"],
+    cflags: ["-Wall"],
     shared_libs: [
         "libcutils",
         "liblog",

diff --git a/tools/libhost/Android.bp b/tools/libhost/Android.bp
index 4c9100f15..82268553b 100644
--- a/tools/libhost/Android.bp
+++ b/tools/libhost/Android.bp
@@ -3,7 +3,6 @@ cc_library_host_static {
     srcs: ["CopyFile.c"],

     cflags: [
-        "-Werror",
         "-Wall",
     ],

diff --git a/tools/makeparallel/Android.bp b/tools/makeparallel/Android.bp
index 898db6861..14be89ac2 100644
--- a/tools/makeparallel/Android.bp
+++ b/tools/makeparallel/Android.bp
@@ -17,5 +17,5 @@ cc_binary_host {
     srcs: [
         "makeparallel.cpp",
     ],
-    cflags: ["-Wall", "-Werror"],
+    cflags: ["-Wall"],
 }

diff --git a/tools/makeparallel/Makefile b/tools/makeparallel/Makefile
index 82a4abfac..b2595d45e 100644
--- a/tools/makeparallel/Makefile
+++ b/tools/makeparallel/Makefile
@@ -27,7 +27,7 @@ MAKEPARALLEL_BIN_PATH ?= .
 MAKEPARALLEL_CXX_SRCS := \
 	makeparallel.cpp

-MAKEPARALLEL_CXXFLAGS := -Wall -Werror -MMD -MP
+MAKEPARALLEL_CXXFLAGS := -Wall -MMD -MP

 MAKEPARALLEL_CXX_SRCS := $(addprefix $(MAKEPARALLEL_SRC_PATH)/,\
 	$(MAKEPARALLEL_CXX_SRCS))

diff --git a/tools/releasetools/common.py b/tools/releasetools/common.py
index f7ab11cd8..355bce2d2 100644
--- a/tools/releasetools/common.py
+++ b/tools/releasetools/common.py
@@ -1751,7 +1751,7 @@ class BlockDifference(object):
     #   decompression_time: 15s  | 25s                | 25s

     if not self.src:
-      brotli_cmd = ['brotli', '--quality=6',
+      brotli_cmd = ['brotli', '--quality=11',
                     '--output={}.new.dat.br'.format(self.path),
                     '{}.new.dat'.format(self.path)]
       print("Compressing {}.new.dat with brotli".format(self.partition))

diff --git a/tools/zipalign/Android.bp b/tools/zipalign/Android.bp
index 8e6196d29..9e65a5cf9 100644
--- a/tools/zipalign/Android.bp
+++ b/tools/zipalign/Android.bp
@@ -13,7 +13,7 @@ cc_binary_host {
         "ZipFile.cpp",
     ],

-    cflags: ["-Wall", "-Werror"],
+    cflags: ["-Wall"],

     // NOTE: Do not add any shared_libs dependencies because they will break the
     // static_sdk_tools target.

diff --git a/tools/ziptime/Android.bp b/tools/ziptime/Android.bp
index 5ef45ed40..7d882ae72 100644
--- a/tools/ziptime/Android.bp
+++ b/tools/ziptime/Android.bp
@@ -27,7 +27,7 @@ cc_binary_host {
     ],

     name: "ziptime",
-    cflags: ["-Wall", "-Werror"],
+    cflags: ["-Wall"],
     target: {
         windows: {
             enabled: true,

project device/lineage/sepolicy/
diff --git a/common/private/backuptool.te b/common/private/backuptool.te
index b948b61..d62a3e0 100644
--- a/common/private/backuptool.te
+++ b/common/private/backuptool.te
@@ -6,6 +6,6 @@ neverallow {
     -update_engine
 } backuptool:process transition;

-userdebug_or_eng(`
+
     permissive backuptool;
-')
+

diff --git a/common/private/recovery.te b/common/private/recovery.te
index 9a9d7cd..fb877e1 100644
--- a/common/private/recovery.te
+++ b/common/private/recovery.te
@@ -4,10 +4,10 @@ r_dir_file(recovery, adb_keys_file)
 set_prop(recovery, shell_prop)

 # Manage fstab and /adb_keys
-userdebug_or_eng(`
+
 allow recovery rootfs:file create_file_perms;
 allow recovery rootfs:file link;
-')
+
 allow recovery rootfs:dir create_dir_perms;

 # Sideload cache
@@ -31,9 +31,9 @@ allow recovery self:process setexec;
 domain_trans(recovery, otapreopt_chroot_exec, backuptool)

 # Some addons require this for A/B installation compatibility
-userdebug_or_eng(`
+
 allow recovery system_file:dir mounton;
-')
+

 # Volume manager
 allow recovery block_device:dir create_dir_perms;
@@ -51,7 +51,7 @@ allow recovery sysfs_batteryinfo:dir search;
 r_dir_file(recovery, unencrypted_data_file)

 # setenforce
-userdebug_or_eng(`
+
 permissive recovery;
-')
+
 ')

diff --git a/common/private/su.te b/common/private/su.te
index ed573a7..5e99f30 100644
--- a/common/private/su.te
+++ b/common/private/su.te
@@ -2,7 +2,7 @@ type superuser_device, file_type, mlstrustedobject;

 ## Perms for the daemon

-userdebug_or_eng(`
+
   domain_trans(init, su_exec, sudaemon)

   typeattribute sudaemon domain, coredomain, mlstrustedsubject;
@@ -44,11 +44,11 @@ userdebug_or_eng(`
   dontaudit sudaemon domain:binder *;
   dontaudit sudaemon property_type:property_service *;
   dontaudit sudaemon appops_service:service_manager *;
-')
+

 ## Perms for the app

-userdebug_or_eng(`
+
   # Translate user apps to the shell domain when using su
   #
   # PR_SET_NO_NEW_PRIVS blocks this :(
@@ -67,6 +67,6 @@ userdebug_or_eng(`
   allow system_app superuser_device:dir { create rw_dir_perms setattr unlink };

   allow kernel sudaemon:fd { use };
-')

-neverallow { domain userdebug_or_eng(`-dumpstate -shell -su -untrusted_app_all -init -sudaemon') } su_exec:file no_x_file_perms;
+
+neverallow { domain -dumpstate -shell -su -untrusted_app_all -init -sudaemon } su_exec:file no_x_file_perms;

diff --git a/common/private/sysinit.te b/common/private/sysinit.te
index 9a09004..9fb0284 100644
--- a/common/private/sysinit.te
+++ b/common/private/sysinit.te
@@ -5,11 +5,10 @@ type sysinit_exec, exec_type, file_type;
 init_daemon_domain(sysinit)
 neverallow { domain -init } sysinit:process transition;

-userdebug_or_eng(`
-    permissive sysinit;
-')
+permissive sysinit;

 dontaudit sysinit shell_exec:file rx_file_perms;
 dontaudit sysinit toolbox_exec:file rx_file_perms;
 dontaudit sysinit system_file:file rx_file_perms;
 dontaudit sysinit system_file:dir r_dir_perms;
+

project external/bash/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index e0880f6..0000000
--- a/Android.mk
+++ /dev/null

project external/nano/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index 014d5d75..00000000
--- a/Android.mk
+++ /dev/null

project external/noto-fonts/
diff --git a/Android.mk b/Android.mk
index 3cfee0c..ed35072 100644
--- a/Android.mk
+++ b/Android.mk
@@ -41,28 +41,13 @@ ifneq ($(MINIMAL_FONT_FOOTPRINT),true)
 LOCAL_PATH := $(NOTO_DIR)/cjk

 font_src_files := \
-    NotoSansCJK-Regular.ttc
+    NotoSansCJK.ttc

 $(foreach f, $(font_src_files), $(call build-one-font-module, $(f)))
 font_src_files :=

 endif # !MINIMAL_FONT_FOOTPRINT

-#############################################################################
-# Similary "build" the Noto CJK fonts for serif family.
-# These are not included in SMALLER_FONT_FOOTPRINT builds.
-#############################################################################
-ifeq ($(filter true,$(EXCLUDE_SERIF_FONTS) $(SMALLER_FONT_FOOTPRINT)),)
-LOCAL_PATH := $(NOTO_DIR)/cjk
-
-font_src_files := \
-    NotoSerifCJK-Regular.ttc
-
-$(foreach f, $(font_src_files), $(call build-one-font-module, $(f)))
-font_src_files :=
-
-endif # !EXCLUDE_SERIF_FONTS && !SMALLER_FONT_FOOTPRINT
-
 #############################################################################
 # Now "build" the Noto Color Emoji font, which is in its own directory. It is
 # not included in the MINIMAL_FONT_FOOTPRINT builds.

diff --git a/cjk/subset_noto_cjk.py b/cjk/subset_noto_cjk.py
index 5a324bf..4a274ce 100755
--- a/cjk/subset_noto_cjk.py
+++ b/cjk/subset_noto_cjk.py
@@ -98,6 +98,6 @@ def remove_codepoints_from_ttc(ttc_name):
         for f in otf_names:
             os.remove(f)

-
+remove_codepoints_from_ttc('NotoSansCJK.ttc')
 remove_codepoints_from_ttc('NotoSansCJK-Regular.ttc')
 remove_codepoints_from_ttc('NotoSerifCJK-Regular.ttc')

diff --git a/fonts.mk b/fonts.mk
index fea0aef..abf40ea 100644
--- a/fonts.mk
+++ b/fonts.mk
@@ -46,7 +46,7 @@ PRODUCT_PACKAGES := \
     NotoSansCham-Bold.ttf \
     NotoSansCham-Regular.ttf \
     NotoSansCherokee-Regular.ttf \
-    NotoSansCJK-Regular.ttc \
+    NotoSansCJK.ttc \
     NotoSansCoptic-Regular.ttf \
     NotoSansCuneiform-Regular.ttf \
     NotoSansCypriot-Regular.ttf \
@@ -191,7 +191,6 @@ PRODUCT_PACKAGES := \
     NotoSerifArmenian-Regular.ttf \
     NotoSerifBengali-Bold.ttf \
     NotoSerifBengali-Regular.ttf \
-    NotoSerifCJK-Regular.ttc \
     NotoSerifDevanagari-Bold.ttf \
     NotoSerifDevanagari-Regular.ttf \
     NotoSerifEthiopic-Bold.otf \

diff --git a/other/NotoSerif-Bold.ttf b/other/NotoSerif-Bold.ttf
index 2ffdf8c..128288b 100644
Binary files a/other/NotoSerif-Bold.ttf and b/other/NotoSerif-Bold.ttf differ

diff --git a/other/NotoSerif-Regular.ttf b/other/NotoSerif-Regular.ttf
index 9d9a347..78136ac 100644
Binary files a/other/NotoSerif-Regular.ttf and b/other/NotoSerif-Regular.ttf differ

project external/perfetto/
diff --git a/Android.bp b/Android.bp
index 93919262..1f920adf 100644
--- a/Android.bp
+++ b/Android.bp
@@ -3810,37 +3810,8 @@ cc_binary_host {
 }

 // GN target: //:traced
-cc_binary {
-  name: "traced",
-  srcs: [
-    "src/traced/service/main.cc",
-  ],
-  shared_libs: [
-    "liblog",
-    "libtraced_shared",
-  ],
-  init_rc: [
-    "perfetto.rc",
-  ],
-  defaults: [
-    "perfetto_defaults",
-  ],
-}

 // GN target: //:traced_probes
-cc_binary {
-  name: "traced_probes",
-  srcs: [
-    "src/traced/probes/main.cc",
-  ],
-  shared_libs: [
-    "liblog",
-    "libtraced_shared",
-  ],
-  defaults: [
-    "perfetto_defaults",
-  ],
-}

 // These targets are appended to the autogenerated Android.bp by tools/gen_android_bp.
 cc_library_static {
@@ -3896,4 +3867,4 @@ cc_library_static {
   cflags: [
     "-DPERFETTO_BUILD_WITH_ANDROID",
   ],
-}
\ No newline at end of file
+}

project external/vim/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index e96cd4839..000000000
--- a/Android.mk
+++ /dev/null

project frameworks/base/
diff --git a/core/res/AndroidManifest.xml b/core/res/AndroidManifest.xml
index 34d26f0da90..08f95ec1fdf 100644
--- a/core/res/AndroidManifest.xml
+++ b/core/res/AndroidManifest.xml
@@ -2357,6 +2357,13 @@
         android:description="@string/permdesc_getPackageSize"
         android:protectionLevel="normal" />

+    <!-- @hide Allows an application to change the package signature as
+	 seen by applications -->
+    <permission android:name="android.permission.FAKE_PACKAGE_SIGNATURE"
+        android:protectionLevel="dangerous"
+        android:label="@string/permlab_fakePackageSignature"
+        android:description="@string/permdesc_fakePackageSignature" />
+
     <!-- @deprecated No longer useful, see
          {@link android.content.pm.PackageManager#addPackageToPreferred}
          for details. -->

diff --git a/core/res/res/values-zh-rCN/strings.xml b/core/res/res/values-zh-rCN/strings.xml
index bc5e1a56807..f0e930f4857 100644
--- a/core/res/res/values-zh-rCN/strings.xml
+++ b/core/res/res/values-zh-rCN/strings.xml
@@ -1886,4 +1886,6 @@
     <string name="notification_appops_microphone_active" msgid="4335305527588191730">"麦克风"</string>
     <string name="notification_appops_overlay_active" msgid="633813008357934729">"显示在屏幕上其他应用的上层"</string>
     <string name="car_loading_profile" msgid="3545132581795684027">"正在加载"</string>
+    <string name="permlab_fakePackageSignature">模拟软件包签名</string>
+    <string name="permdesc_fakePackageSignature">允许应用伪装成为其他应用</string>
 </resources>

diff --git a/core/res/res/values/config.xml b/core/res/res/values/config.xml
index f19b0c386f0..a080fa226d6 100644
--- a/core/res/res/values/config.xml
+++ b/core/res/res/values/config.xml
@@ -1716,6 +1716,8 @@
     <string-array name="config_locationProviderPackageNames" translatable="false">
         <!-- The standard AOSP fused location provider -->
         <item>com.android.location.fused</item>
+        <!-- The (faked) microg fused location provider -->
+        <item>com.google.android.gms</item>
     </string-array>

     <!-- This string array can be overriden to enable test location providers initially. -->

diff --git a/core/res/res/values/strings.xml b/core/res/res/values/strings.xml
index f6600462ea7..9c52929608f 100644
--- a/core/res/res/values/strings.xml
+++ b/core/res/res/values/strings.xml
@@ -785,6 +785,10 @@

     <!--  Permissions -->

+    <string name="permlab_fakePackageSignature">Spoof package signature</string>
+    <!-- Description of an application permission, listed so the user can choose whether they want to allow the application to do this. -->
+    <string name="permdesc_fakePackageSignature">Allows the app to pretend to be a different app.</string>
+
     <!-- Title of an application permission, listed so the user can choose whether they want to allow the application to do this. -->
     <string name="permlab_statusBar">disable or modify status bar</string>
     <!-- Description of an application permission, listed so the user can choose whether they want to allow the application to do this. -->

diff --git a/data/fonts/fonts.xml b/data/fonts/fonts.xml
index 72d9bce687a..51a82495e1a 100644
--- a/data/fonts/fonts.xml
+++ b/data/fonts/fonts.xml
@@ -521,20 +521,56 @@
         <font weight="400" style="normal">NotoSansSymbols-Regular-Subsetted.ttf</font>
     </family>
     <family lang="zh-Hans">
-        <font weight="400" style="normal" index="2">NotoSansCJK-Regular.ttc</font>
-        <font weight="400" style="normal" index="2" fallbackFor="serif">NotoSerifCJK-Regular.ttc</font>
+        <font weight="100" style="normal" index="2">NotoSansCJK.ttc</font>
+        <font weight="300" style="normal" index="7">NotoSansCJK.ttc</font>
+        <font weight="350" style="normal" index="12">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="17">NotoSansCJK.ttc</font>
+        <font weight="500" style="normal" index="27">NotoSansCJK.ttc</font>
+        <font weight="700" style="normal" index="32">NotoSansCJK.ttc</font>
+        <font weight="900" style="normal" index="42">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" fallbackFor="serif">NotoSerif-Regular.ttf</font>
+        <font weight="700" style="normal" fallbackFor="serif">NotoSerif-Bold.ttf</font>
+        <font weight="400" style="italic" fallbackFor="serif">NotoSerif-Italic.ttf</font>
+        <font weight="700" style="italic" fallbackFor="serif">NotoSerif-BoldItalic.ttf</font>
     </family>
     <family lang="zh-Hant zh-Bopo">
-        <font weight="400" style="normal" index="3">NotoSansCJK-Regular.ttc</font>
-        <font weight="400" style="normal" index="3" fallbackFor="serif">NotoSerifCJK-Regular.ttc</font>
+        <font weight="100" style="normal" index="3">NotoSansCJK.ttc</font>
+        <font weight="300" style="normal" index="8">NotoSansCJK.ttc</font>
+        <font weight="350" style="normal" index="13">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="18">NotoSansCJK.ttc</font>
+        <font weight="500" style="normal" index="28">NotoSansCJK.ttc</font>
+        <font weight="700" style="normal" index="33">NotoSansCJK.ttc</font>
+        <font weight="900" style="normal" index="43">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" fallbackFor="serif">NotoSerif-Regular.ttf</font>
+        <font weight="700" style="normal" fallbackFor="serif">NotoSerif-Bold.ttf</font>
+        <font weight="400" style="italic" fallbackFor="serif">NotoSerif-Italic.ttf</font>
+        <font weight="700" style="italic" fallbackFor="serif">NotoSerif-BoldItalic.ttf</font>
     </family>
     <family lang="ja">
-        <font weight="400" style="normal" index="0">NotoSansCJK-Regular.ttc</font>
-        <font weight="400" style="normal" index="0" fallbackFor="serif">NotoSerifCJK-Regular.ttc</font>
+        <font weight="100" style="normal" index="0">NotoSansCJK.ttc</font>
+        <font weight="300" style="normal" index="5">NotoSansCJK.ttc</font>
+        <font weight="350" style="normal" index="10">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="15">NotoSansCJK.ttc</font>
+        <font weight="500" style="normal" index="25">NotoSansCJK.ttc</font>
+        <font weight="700" style="normal" index="30">NotoSansCJK.ttc</font>
+        <font weight="900" style="normal" index="40">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" fallbackFor="serif">NotoSerif-Regular.ttf</font>
+        <font weight="700" style="normal" fallbackFor="serif">NotoSerif-Bold.ttf</font>
+        <font weight="400" style="italic" fallbackFor="serif">NotoSerif-Italic.ttf</font>
+        <font weight="700" style="italic" fallbackFor="serif">NotoSerif-BoldItalic.ttf</font>
     </family>
     <family lang="ko">
-        <font weight="400" style="normal" index="1">NotoSansCJK-Regular.ttc</font>
-        <font weight="400" style="normal" index="1" fallbackFor="serif">NotoSerifCJK-Regular.ttc</font>
+        <font weight="100" style="normal" index="1">NotoSansCJK.ttc</font>
+        <font weight="300" style="normal" index="6">NotoSansCJK.ttc</font>
+        <font weight="350" style="normal" index="11">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="16">NotoSansCJK.ttc</font>
+        <font weight="500" style="normal" index="26">NotoSansCJK.ttc</font>
+        <font weight="700" style="normal" index="31">NotoSansCJK.ttc</font>
+        <font weight="900" style="normal" index="41">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" fallbackFor="serif">NotoSerif-Regular.ttf</font>
+        <font weight="700" style="normal" fallbackFor="serif">NotoSerif-Bold.ttf</font>
+        <font weight="400" style="italic" fallbackFor="serif">NotoSerif-Italic.ttf</font>
+        <font weight="700" style="italic" fallbackFor="serif">NotoSerif-BoldItalic.ttf</font>
     </family>
     <family lang="und-Zsye">
         <font weight="400" style="normal">NotoColorEmoji.ttf</font>

diff --git a/data/fonts/fonts_no_serif.xml b/data/fonts/fonts_no_serif.xml
index 28beba4e460..1ec79209231 100644
--- a/data/fonts/fonts_no_serif.xml
+++ b/data/fonts/fonts_no_serif.xml
@@ -487,16 +487,48 @@
         <font weight="400" style="normal">NotoSansSymbols-Regular-Subsetted.ttf</font>
     </family>
     <family lang="zh-Hans">
-        <font weight="400" style="normal" index="2">NotoSansCJK-Regular.ttc</font>
+        <font weight="100" style="normal" index="2">NotoSansCJK.ttc</font>
+        <font weight="300" style="normal" index="7">NotoSansCJK.ttc</font>
+        <font weight="350" style="normal" index="12">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="17">NotoSansCJK.ttc</font>
+        <font weight="500" style="normal" index="27">NotoSansCJK.ttc</font>
+        <font weight="700" style="normal" index="32">NotoSansCJK.ttc</font>
+        <font weight="900" style="normal" index="42">NotoSansCJK.ttc</font>
+        <font weight="400" style="mono" index="22">NotoSansCJK.ttc</font>
+        <font weight="700" style="mono" index="37">NotoSansCJK.ttc</font>
     </family>
     <family lang="zh-Hant zh-Bopo">
-        <font weight="400" style="normal" index="3">NotoSansCJK-Regular.ttc</font>
+        <font weight="100" style="normal" index="3">NotoSansCJK.ttc</font>
+        <font weight="300" style="normal" index="8">NotoSansCJK.ttc</font>
+        <font weight="350" style="normal" index="13">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="18">NotoSansCJK.ttc</font>
+        <font weight="500" style="normal" index="28">NotoSansCJK.ttc</font>
+        <font weight="700" style="normal" index="33">NotoSansCJK.ttc</font>
+        <font weight="900" style="normal" index="43">NotoSansCJK.ttc</font>
+        <font weight="400" style="mono" index="23">NotoSansCJK.ttc</font>
+        <font weight="700" style="mono" index="38">NotoSansCJK.ttc</font>
     </family>
     <family lang="ja">
-        <font weight="400" style="normal" index="0">NotoSansCJK-Regular.ttc</font>
+        <font weight="100" style="normal" index="0">NotoSansCJK.ttc</font>
+        <font weight="300" style="normal" index="5">NotoSansCJK.ttc</font>
+        <font weight="350" style="normal" index="10">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="15">NotoSansCJK.ttc</font>
+        <font weight="500" style="normal" index="25">NotoSansCJK.ttc</font>
+        <font weight="700" style="normal" index="30">NotoSansCJK.ttc</font>
+        <font weight="900" style="normal" index="40">NotoSansCJK.ttc</font>
+        <font weight="400" style="mono" index="20">NotoSansCJK.ttc</font>
+        <font weight="700" style="mono" index="35">NotoSansCJK.ttc</font>
     </family>
     <family lang="ko">
-        <font weight="400" style="normal" index="1">NotoSansCJK-Regular.ttc</font>
+        <font weight="100" style="normal" index="1">NotoSansCJK.ttc</font>
+        <font weight="300" style="normal" index="6">NotoSansCJK.ttc</font>
+        <font weight="350" style="normal" index="11">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="16">NotoSansCJK.ttc</font>
+        <font weight="500" style="normal" index="26">NotoSansCJK.ttc</font>
+        <font weight="700" style="normal" index="31">NotoSansCJK.ttc</font>
+        <font weight="900" style="normal" index="41">NotoSansCJK.ttc</font>
+        <font weight="400" style="mono" index="21">NotoSansCJK.ttc</font>
+        <font weight="700" style="mono" index="36">NotoSansCJK.ttc</font>
     </family>
     <family lang="und-Zsye">
         <font weight="400" style="normal">NotoColorEmoji.ttf</font>

diff --git a/packages/CtsShim/Android.mk b/packages/CtsShim/Android.mk
deleted file mode 100644
index 12972f14514..00000000000
--- a/packages/CtsShim/Android.mk
+++ /dev/null

diff --git a/packages/CtsShim/build/Android.mk b/packages/CtsShim/build/Android.mk
deleted file mode 100644
index ec14d50b371..00000000000
--- a/packages/CtsShim/build/Android.mk
+++ /dev/null

diff --git a/packages/EasterEgg/Android.bp b/packages/EasterEgg/Android.bp
deleted file mode 100644
index 43ed810b567..00000000000
--- a/packages/EasterEgg/Android.bp
+++ /dev/null

diff --git a/packages/overlays/DisplayCutoutEmulationCornerOverlay/Android.mk b/packages/overlays/DisplayCutoutEmulationCornerOverlay/Android.mk
deleted file mode 100644
index 74c43b40616..00000000000
--- a/packages/overlays/DisplayCutoutEmulationCornerOverlay/Android.mk
+++ /dev/null

diff --git a/packages/overlays/DisplayCutoutEmulationDoubleOverlay/Android.mk b/packages/overlays/DisplayCutoutEmulationDoubleOverlay/Android.mk
deleted file mode 100644
index d83b30a8785..00000000000
--- a/packages/overlays/DisplayCutoutEmulationDoubleOverlay/Android.mk
+++ /dev/null

diff --git a/packages/overlays/DisplayCutoutEmulationNarrowOverlay/Android.mk b/packages/overlays/DisplayCutoutEmulationNarrowOverlay/Android.mk
deleted file mode 100644
index f5afad24676..00000000000
--- a/packages/overlays/DisplayCutoutEmulationNarrowOverlay/Android.mk
+++ /dev/null

diff --git a/packages/overlays/DisplayCutoutEmulationTallOverlay/Android.mk b/packages/overlays/DisplayCutoutEmulationTallOverlay/Android.mk
deleted file mode 100644
index f1f8c27d94f..00000000000
--- a/packages/overlays/DisplayCutoutEmulationTallOverlay/Android.mk
+++ /dev/null

diff --git a/packages/overlays/DisplayCutoutEmulationWideOverlay/Android.mk b/packages/overlays/DisplayCutoutEmulationWideOverlay/Android.mk
deleted file mode 100644
index d149d8ecf4d..00000000000
--- a/packages/overlays/DisplayCutoutEmulationWideOverlay/Android.mk
+++ /dev/null

diff --git a/services/core/java/com/android/server/pm/PackageManagerService.java b/services/core/java/com/android/server/pm/PackageManagerService.java
index 7edf729608a..83fd95ea5a3 100644
--- a/services/core/java/com/android/server/pm/PackageManagerService.java
+++ b/services/core/java/com/android/server/pm/PackageManagerService.java
@@ -4003,8 +4003,9 @@ public class PackageManagerService extends IPackageManager.Stub
             final Set<String> permissions = ArrayUtils.isEmpty(p.requestedPermissions)
                     ? Collections.<String>emptySet() : permissionsState.getPermissions(userId);

-            PackageInfo packageInfo = PackageParser.generatePackageInfo(p, gids, flags,
-                    ps.firstInstallTime, ps.lastUpdateTime, permissions, state, userId);
+            PackageInfo packageInfo = mayFakeSignature(p, PackageParser.generatePackageInfo(p, gids, flags,
+                ps.firstInstallTime, ps.lastUpdateTime, permissions, state, userId),
+                permissions);

             if (packageInfo == null) {
                 return null;
@@ -4040,6 +4041,24 @@ public class PackageManagerService extends IPackageManager.Stub
         }
     }

+    private PackageInfo mayFakeSignature(PackageParser.Package p, PackageInfo pi,
+            Set<String> permissions) {
+        try {
+            if (permissions.contains("android.permission.FAKE_PACKAGE_SIGNATURE")
+                    && p.applicationInfo.targetSdkVersion > Build.VERSION_CODES.LOLLIPOP_MR1
+                    && p.mAppMetaData != null) {
+                String sig = p.mAppMetaData.getString("fake-signature");
+                if (sig != null) {
+                    pi.signatures = new Signature[] {new Signature(sig)};
+                }
+            }
+        } catch (Throwable t) {
+            // We should never die because of any failures, this is system code!
+            Log.w("PackageManagerService.FAKE_PACKAGE_SIGNATURE", t);
+       }
+        return pi;
+    }
+
     @Override
     public void checkPackageStartable(String packageName, int userId) {
         final int callingUid = Binder.getCallingUid();

project hardware/qcom/media-caf/msm8994/
diff --git a/mm-video-v4l2/vidc/common/Android.mk b/mm-video-v4l2/vidc/common/Android.mk
index 6f783c429..c14bdcffe 100644
--- a/mm-video-v4l2/vidc/common/Android.mk
+++ b/mm-video-v4l2/vidc/common/Android.mk
@@ -13,7 +13,6 @@ libmm-vidc-def += -D__alignx\(x\)=__attribute__\(\(__aligned__\(x\)\)\)
 libmm-vidc-def += -DT_ARM
 libmm-vidc-def += -Dinline=__inline
 libmm-vidc-def += -D_ANDROID_
-libmm-vidc-def += -Werror
 libmm-vidc-def += -D_ANDROID_ICS_

 # ---------------------------------------------------------------------------------

diff --git a/mm-video-v4l2/vidc/vdec/Android.mk b/mm-video-v4l2/vidc/vdec/Android.mk
index aedca2059..de931ab52 100644
--- a/mm-video-v4l2/vidc/vdec/Android.mk
+++ b/mm-video-v4l2/vidc/vdec/Android.mk
@@ -97,7 +97,7 @@ include $(CLEAR_VARS)
 LOCAL_MODULE                    := libOmxVdec
 LOCAL_MODULE_TAGS               := optional
 LOCAL_VENDOR_MODULE             := true
-LOCAL_CFLAGS                    := $(libmm-vdec-def) -Werror
+LOCAL_CFLAGS                    := $(libmm-vdec-def)
 LOCAL_C_INCLUDES                += $(libmm-vdec-inc)

 LOCAL_HEADER_LIBRARIES := generated_kernel_headers

diff --git a/mm-video-v4l2/vidc/venc/Android.mk b/mm-video-v4l2/vidc/venc/Android.mk
index ff43abdcd..f1b15dfc3 100644
--- a/mm-video-v4l2/vidc/venc/Android.mk
+++ b/mm-video-v4l2/vidc/venc/Android.mk
@@ -17,7 +17,6 @@ libmm-venc-def += -DENABLE_DEBUG_ERROR
 libmm-venc-def += -UINPUT_BUFFER_LOG
 libmm-venc-def += -UOUTPUT_BUFFER_LOG
 libmm-venc-def += -USINGLE_ENCODER_INSTANCE
-libmm-venc-def += -Werror
 libmm-venc-def += -D_ANDROID_ICS_
 libmm-venc-def += -D_MSM8974_

project lineage-sdk/
diff --git a/packages/LineageSettingsProvider/Android.bp b/packages/LineageSettingsProvider/Android.bp
deleted file mode 100644
index 3b9d3e6..0000000
--- a/packages/LineageSettingsProvider/Android.bp
+++ /dev/null

project packages/apps/AudioFX/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index ba2c9ba..0000000
--- a/Android.mk
+++ /dev/null

project packages/apps/Bluetooth/
diff --git a/tests b/tests
deleted file mode 100644
index fd297e35..00000000
--- a/tests
+++ /dev/null

project packages/apps/Calendar/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index 656f3847..00000000
--- a/Android.mk
+++ /dev/null

project packages/apps/Email/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index cfc205b89..000000000
--- a/Android.mk
+++ /dev/null

project packages/apps/Exchange/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index a3a7d20b..00000000
--- a/Android.mk
+++ /dev/null

project packages/apps/LockClock/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index 603f888..0000000
--- a/Android.mk
+++ /dev/null

project packages/apps/Settings/
diff --git a/src/com/android/settings/development/RootAccessPreferenceController.java b/src/com/android/settings/development/RootAccessPreferenceController.java
index 7e40e92ba7..669067d936 100644
--- a/src/com/android/settings/development/RootAccessPreferenceController.java
+++ b/src/com/android/settings/development/RootAccessPreferenceController.java
@@ -54,8 +54,8 @@ public class RootAccessPreferenceController extends DeveloperOptionsPreferenceCo

     @Override
     public boolean isAvailable() {
-        // User builds don't get root, and eng always gets root
-        return Build.IS_DEBUGGABLE || "eng".equals(Build.TYPE);
+        // Not only eng or userdebug builds, user builds could also get root now
+        return true;
     }

     @Override

diff --git a/src/com/android/settings/development/RootAppOpsPreferenceController.java b/src/com/android/settings/development/RootAppOpsPreferenceController.java
index 7a6b0f1a7b..4ce15e9ac6 100644
--- a/src/com/android/settings/development/RootAppOpsPreferenceController.java
+++ b/src/com/android/settings/development/RootAppOpsPreferenceController.java
@@ -48,8 +48,8 @@ public class RootAppOpsPreferenceController extends DeveloperOptionsPreferenceCo

     @Override
     public boolean isAvailable() {
-        // User builds don't get root, and eng always gets root
-        return Build.IS_DEBUGGABLE || "eng".equals(Build.TYPE);
+        // Not only eng or userdebug builds, user builds could also get root now
+        return true;
     }

     @Override

project packages/apps/Snap/
diff --git a/res/values/config.xml b/res/values/config.xml
index cb78e532c..2fd6601d2 100644
--- a/res/values/config.xml
+++ b/res/values/config.xml
@@ -29,7 +29,7 @@
     <bool name="enable_warped_pano_preview">false</bool>

     <!-- Enable support for camera api v2 -->
-    <bool name="support_camera_api_v2">false</bool>
+    <bool name="support_camera_api_v2">true</bool>

     <!-- Opens back camera using openLegacy() and forces api v1 -->
     <bool name="back_camera_open_legacy">true</bool>

project packages/apps/Stk/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index 135fcf9..0000000
--- a/Android.mk
+++ /dev/null

project packages/apps/Traceur/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index 2c5803e..0000000
--- a/Android.mk
+++ /dev/null

project packages/apps/Updater/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index df2b1f5..0000000
--- a/Android.mk
+++ /dev/null

project packages/providers/BookmarkProvider/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index f2100b5..0000000
--- a/Android.mk
+++ /dev/null

project packages/providers/WeatherProvider/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index 10087e2..0000000
--- a/Android.mk
+++ /dev/null

project packages/screensavers/Basic/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index 59d54fb..0000000
--- a/Android.mk
+++ /dev/null

project packages/screensavers/PhotoTable/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index e2f2f3a..0000000
--- a/Android.mk
+++ /dev/null

project packages/screensavers/WebView/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index 7f79bc3..0000000
--- a/Android.mk
+++ /dev/null

project packages/services/BuiltInPrintService/
diff --git a/jni/Android.mk b/jni/Android.mk
index 5dbe400..26ffa36 100644
--- a/jni/Android.mk
+++ b/jni/Android.mk
@@ -24,7 +24,7 @@ IPP_HELPER_DIR := ipphelper
 LOCAL_SDK_VERSION := current

 LOCAL_CFLAGS += \
-      -DINCLUDE_PDF=1 -Werror -Wextra -Wno-unused-parameter \
+      -DINCLUDE_PDF=1 -Wextra -Wno-unused-parameter \
       -Wno-sign-compare -Wno-missing-field-initializers \
       -Wno-implicit-function-declaration -Wno-format -Wno-missing-braces \
       -Wno-deprecated-declarations

project system/core/
diff --git a/debuggerd/Android.bp b/debuggerd/Android.bp
index 7e6f24d15..92386dfad 100644
--- a/debuggerd/Android.bp
+++ b/debuggerd/Android.bp
@@ -233,77 +233,6 @@ cc_benchmark {
     ],
 }

-cc_binary {
-    name: "crash_dump",
-    srcs: [
-        "crash_dump.cpp",
-        "util.cpp",
-    ],
-    defaults: ["debuggerd_defaults"],
-
-    compile_multilib: "both",
-    multilib: {
-        lib32: {
-            suffix: "32",
-        },
-        lib64: {
-            suffix: "64",
-        },
-    },
-
-    static_libs: [
-        "libtombstoned_client_static",
-        "libdebuggerd",
-        "libcutils",
-    ],
-
-    shared_libs: [
-        "libbacktrace",
-        "libbase",
-        "liblog",
-        "libprocinfo",
-        "libunwindstack",
-    ],
-}
-
-cc_binary {
-    name: "debuggerd",
-    srcs: [
-        "debuggerd.cpp",
-    ],
-    defaults: ["debuggerd_defaults"],
-
-    shared_libs: [
-        "libbase",
-        "libdebuggerd_client",
-        "liblog",
-        "libprocinfo",
-    ],
-
-    local_include_dirs: ["include"],
-}
-
-cc_binary {
-    name: "tombstoned",
-    srcs: [
-        "util.cpp",
-        "tombstoned/intercept_manager.cpp",
-        "tombstoned/tombstoned.cpp",
-    ],
-    defaults: ["debuggerd_defaults"],
-
-    header_libs: ["libdebuggerd_common_headers"],
-
-    static_libs: [
-        "libbase",
-        "libcutils",
-        "libevent",
-        "liblog",
-    ],
-
-    init_rc: ["tombstoned/tombstoned.rc"],
-}
-
 subdirs = [
     "crasher",
 ]

diff --git a/init/Android.bp b/init/Android.bp
index 3287331e2..23ee5f11c 100644
--- a/init/Android.bp
+++ b/init/Android.bp
@@ -27,7 +27,7 @@ cc_defaults {
         "-Wno-unused-parameter",
         "-Werror",
         "-DALLOW_LOCAL_PROP_OVERRIDE=1",
-        "-DALLOW_PERMISSIVE_SELINUX=0",
+        "-DALLOW_PERMISSIVE_SELINUX=1",
         "-DREBOOT_BOOTLOADER_ON_PANIC=0",
         "-DWORLD_WRITABLE_KMSG=0",
         "-DDUMP_ON_UMOUNT_FAILURE=0",

diff --git a/init/Android.mk b/init/Android.mk
index 2295bd719..f008aff55 100644
--- a/init/Android.mk
+++ b/init/Android.mk
@@ -14,7 +14,7 @@ init_options += \
 else
 init_options += \
     -DALLOW_LOCAL_PROP_OVERRIDE=0 \
-    -DALLOW_PERMISSIVE_SELINUX=0 \
+    -DALLOW_PERMISSIVE_SELINUX=1 \
     -DREBOOT_BOOTLOADER_ON_PANIC=0 \
     -DWORLD_WRITABLE_KMSG=0 \
     -DDUMP_ON_UMOUNT_FAILURE=0

diff --git a/rootdir/init.rc b/rootdir/init.rc
index 4a8a60a96..28b83d0af 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -397,11 +397,6 @@ on post-fs-data
     start vold
     installkey /data

-    # Start bootcharting as soon as possible after the data partition is
-    # mounted to collect more data.
-    mkdir /data/bootchart 0755 shell shell
-    bootchart start
-
     # Avoid predictable entropy pool. Carry over entropy from previous boot.
     copy /data/system/entropy.dat /dev/urandom

@@ -447,7 +442,6 @@ on post-fs-data
     mkdir /data/misc/ethernet 0770 system system
     mkdir /data/misc/dhcp 0770 dhcp dhcp
     mkdir /data/misc/user 0771 root root
-    mkdir /data/misc/perfprofd 0775 root root
     # give system access to wpa_supplicant.conf for backup and restore
     chmod 0660 /data/misc/wifi/wpa_supplicant.conf
     mkdir /data/local 0751 root root
@@ -455,18 +449,10 @@ on post-fs-data
     mkdir /data/misc/audioserver 0700 audioserver audioserver
     mkdir /data/misc/cameraserver 0700 cameraserver cameraserver
     mkdir /data/misc/vold 0700 root root
-    mkdir /data/misc/boottrace 0771 system shell
-    mkdir /data/misc/update_engine 0700 root root
-    mkdir /data/misc/update_engine_log 02750 root log
-    mkdir /data/misc/trace 0700 root root
-    # create location to store surface and window trace files
-    mkdir /data/misc/wmtrace 0700 system system
     # profile file layout
     mkdir /data/misc/profiles 0771 system system
     mkdir /data/misc/profiles/cur 0771 system system
     mkdir /data/misc/profiles/ref 0771 system system
-    mkdir /data/misc/profman 0770 system shell
-    mkdir /data/misc/gcov 0770 root root

     mkdir /data/vendor 0771 root root
     mkdir /data/vendor_ce 0771 root root
@@ -476,7 +462,6 @@ on post-fs-data
     # For security reasons, /data/local/tmp should always be empty.
     # Do not place files or directories in /data/local/tmp
     mkdir /data/local/tmp 0771 shell shell
-    mkdir /data/local/traces 0777 shell shell
     mkdir /data/data 0771 system system
     mkdir /data/app-private 0771 system system
     mkdir /data/app-ephemeral 0771 system system
@@ -484,18 +469,9 @@ on post-fs-data
     mkdir /data/app-lib 0771 system system
     mkdir /data/app 0771 system system
     mkdir /data/property 0700 root root
-    mkdir /data/tombstones 0771 system system
-    mkdir /data/vendor/tombstones 0771 root root
-    mkdir /data/vendor/tombstones/wifi 0771 wifi wifi

     # create dalvik-cache, so as to enforce our permissions
     mkdir /data/dalvik-cache 0771 root root
-    # create the A/B OTA directory, so as to enforce our permissions
-    mkdir /data/ota 0771 root root
-
-    # create the OTA package directory. It will be accessed by GmsCore (cache
-    # group), update_engine and update_verifier.
-    mkdir /data/ota_package 0770 system cache

     # create resource-cache and double-check the perms
     mkdir /data/resource-cache 0771 system system
@@ -513,19 +489,11 @@ on post-fs-data
     # the following directory.
     mkdir /data/mediadrm 0770 mediadrm mediadrm

-    mkdir /data/anr 0775 system system
-
-    # NFC: create data/nfc for nv storage
-    mkdir /data/nfc 0770 nfc nfc
-    mkdir /data/nfc/param 0770 nfc nfc
-
     # Create all remaining /data root dirs so that they are made through init
     # and get proper encryption policy installed
     mkdir /data/backup 0700 system system
-    mkdir /data/ss 0700 system system

     mkdir /data/system 0775 system system
-    mkdir /data/system/heapdump 0700 system system
     mkdir /data/system/users 0775 system system

     mkdir /data/system_de 0770 system system
@@ -712,7 +680,7 @@ on property:vold.decrypt=trigger_shutdown_framework
     class_reset main

 on property:sys.boot_completed=1
-    bootchart stop
+    exec /system/bin/setenforce 1

 # system server cannot write to /proc/sys files,
 # and chown/chmod does not work for /proc/sys/ entries.

project system/extras/su/
diff --git a/Android.mk b/Android.mk
index 86b5f70..4db5dc1 100644
--- a/Android.mk
+++ b/Android.mk
@@ -14,7 +14,6 @@ LOCAL_SHARED_LIBRARIES := \

 LOCAL_SRC_FILES := su.c daemon.c utils.c pts.c
 LOCAL_SRC_FILES += binder/appops-wrapper.cpp binder/pm-wrapper.c
-LOCAL_CFLAGS += -Werror -Wall
 LOCAL_MODULE_PATH := $(TARGET_OUT_OPTIONAL_EXECUTABLES)

 LOCAL_INIT_RC := superuser.rc

diff --git a/su.c b/su.c
index 5ec7ee4..5b402ef 100644
--- a/su.c
+++ b/su.c
@@ -277,12 +277,6 @@ int access_disabled(const struct su_initiator* from) {
         return 1;
     }

-    /* Only allow su on debuggable builds */
-    if (!property_get_bool("ro.debuggable", false)) {
-        ALOGE("Root access is disabled on non-debug builds");
-        return 1;
-    }
-
     /* Enforce persist.sys.root_access on non-eng builds for apps */
     enabled = property_get_int32("persist.sys.root_access", 2);
     property_get("ro.build.type", build_type, "");

project system/sepolicy/
diff --git a/Android.mk b/Android.mk
index ff38a43a..de5a58af 100644
--- a/Android.mk
+++ b/Android.mk
@@ -80,10 +80,12 @@ ifndef BOARD_SEPOLICY_VERS
 BOARD_SEPOLICY_VERS := $(PLATFORM_SEPOLICY_VERSION)
 endif

-NEVERALLOW_ARG :=
+# Hacks to keep SELinux permissive in user builds
+SELINUX_IGNORE_NEVERALLOWS := true
+
 ifeq ($(SELINUX_IGNORE_NEVERALLOWS),true)
 ifeq ($(TARGET_BUILD_VARIANT),user)
-$(error SELINUX_IGNORE_NEVERALLOWS := true cannot be used in user builds)
+$(warning SELINUX_IGNORE_NEVERALLOWS := true cannot be used in user builds.)
 endif
 $(warning Be careful when using the SELINUX_IGNORE_NEVERALLOWS flag. \
           It does not work in user builds and using it will \
@@ -705,10 +707,9 @@ $(built_sepolicy_neverallows)
 	$(hide) $(HOST_OUT_EXECUTABLES)/sepolicy-analyze $@.tmp permissive > $@.permissivedomains
 	$(hide) if [ "$(TARGET_BUILD_VARIANT)" = "user" -a -s $@.permissivedomains ]; then \
 		echo "==========" 1>&2; \
-		echo "ERROR: permissive domains not allowed in user builds" 1>&2; \
+		echo "WARNING: permissive domains not allowed in user builds" 1>&2; \
 		echo "List of invalid domains:" 1>&2; \
 		cat $@.permissivedomains 1>&2; \
-		exit 1; \
 		fi
 	$(hide) mv $@.tmp $@

@@ -756,10 +757,9 @@ $(LOCAL_BUILT_MODULE): $(sepolicy.recovery.conf) $(HOST_OUT_EXECUTABLES)/checkpo
 	$(hide) $(HOST_OUT_EXECUTABLES)/sepolicy-analyze $@.tmp permissive > $@.permissivedomains
 	$(hide) if [ "$(TARGET_BUILD_VARIANT)" = "user" -a -s $@.permissivedomains ]; then \
 		echo "==========" 1>&2; \
-		echo "ERROR: permissive domains not allowed in user builds" 1>&2; \
+		echo "WARNING: permissive domains not allowed in user builds" 1>&2; \
 		echo "List of invalid domains:" 1>&2; \
 		cat $@.permissivedomains 1>&2; \
-		exit 1; \
 		fi
 	$(hide) mv $@.tmp $@

diff --git a/prebuilts/api/28.0/private/adbd.te b/prebuilts/api/28.0/private/adbd.te
index 3ac47443..1b44403d 100644
--- a/prebuilts/api/28.0/private/adbd.te
+++ b/prebuilts/api/28.0/private/adbd.te
@@ -7,10 +7,10 @@ init_daemon_domain(adbd)

 domain_auto_trans(adbd, shell_exec, shell)

-userdebug_or_eng(`
+
   allow adbd self:process setcurrent;
   allow adbd su:process dyntransition;
-')
+

 # Do not sanitize the environment or open fds of the shell. Allow signaling
 # created processes.
@@ -90,13 +90,13 @@ hal_client_domain(adbd, hal_graphics_allocator)
 allow adbd adb_keys_file:dir search;
 allow adbd adb_keys_file:file r_file_perms;

-userdebug_or_eng(`
+
   # Write debugging information to /data/adb
   # when persist.adb.trace_mask is set
   # https://code.google.com/p/android/issues/detail?id=72895
   allow adbd adb_data_file:dir rw_dir_perms;
   allow adbd adb_data_file:file create_file_perms;
-')
+

 # ndk-gdb invokes adb forward to forward the gdbserver socket.
 allow adbd app_data_file:dir search;
@@ -149,4 +149,4 @@ allow adbd rootfs:dir r_dir_perms;
 # transitions to the shell domain (except when it crashes). In particular, we
 # never want to see a transition from adbd to su (aka "adb root")
 neverallow adbd { domain -crash_dump -shell }:process transition;
-neverallow adbd { domain userdebug_or_eng(`-su') }:process dyntransition;
+neverallow adbd { domain -su }:process dyntransition;

diff --git a/prebuilts/api/28.0/private/compat/26.0/26.0.ignore.cil b/prebuilts/api/28.0/private/compat/26.0/26.0.ignore.cil
index b07350c4..635726b0 100644
--- a/prebuilts/api/28.0/private/compat/26.0/26.0.ignore.cil
+++ b/prebuilts/api/28.0/private/compat/26.0/26.0.ignore.cil
@@ -157,3 +157,5 @@
      ( adbd_tmpfs
        untrusted_app_27_tmpfs
      ))
+
+allow sudaemon untrusted_app_all_devpts:chr_file *;

diff --git a/prebuilts/api/28.0/private/su.te b/prebuilts/api/28.0/private/su.te
index 16e47bbb..27a97bdb 100644
--- a/prebuilts/api/28.0/private/su.te
+++ b/prebuilts/api/28.0/private/su.te
@@ -1,4 +1,4 @@
-userdebug_or_eng(`
+
   typeattribute su coredomain;

   domain_auto_trans(shell, su_exec, su)
@@ -20,4 +20,4 @@ userdebug_or_eng(`
   permissive su;

   app_domain(su)
-')
+

diff --git a/prebuilts/api/28.0/private/untrusted_app_25.te b/prebuilts/api/28.0/private/untrusted_app_25.te
index ba2c1e1c..ad66b60b 100644
--- a/prebuilts/api/28.0/private/untrusted_app_25.te
+++ b/prebuilts/api/28.0/private/untrusted_app_25.te
@@ -40,3 +40,4 @@ allow untrusted_app_25 proc_misc:file r_file_perms;
 # https://github.com/strazzere/anti-emulator/blob/master/AntiEmulator/src/diff/strazzere/anti/emulator/FindEmulator.java
 # This will go away in a future Android release
 allow untrusted_app_25 proc_tty_drivers:file r_file_perms;
+allow untrusted_app socket_device:sock_file *;

diff --git a/prebuilts/api/28.0/private/untrusted_app_all.te b/prebuilts/api/28.0/private/untrusted_app_all.te
index 6cf16682..b19c0cc0 100644
--- a/prebuilts/api/28.0/private/untrusted_app_all.te
+++ b/prebuilts/api/28.0/private/untrusted_app_all.te
@@ -138,3 +138,4 @@ dontaudit untrusted_app_all net_dns_prop:file read;
 dontaudit untrusted_app_all proc_stat:file read;
 dontaudit untrusted_app_all proc_vmstat:file read;
 dontaudit untrusted_app_all proc_uptime:file read;
+allow untrusted_app socket_device:sock_file write;

diff --git a/prebuilts/api/28.0/public/app.te b/prebuilts/api/28.0/public/app.te
index 239332c1..35a679b5 100644
--- a/prebuilts/api/28.0/public/app.te
+++ b/prebuilts/api/28.0/public/app.te
@@ -306,7 +306,7 @@ allowxperm { appdomain -bluetooth } self:{ rawip_socket tcp_socket udp_socket }

 allow { appdomain -isolated_app } ion_device:chr_file rw_file_perms;
 # TODO is write really necessary ?
-auditallow { appdomain userdebug_or_eng(`-su') } ion_device:chr_file { write append };
+auditallow { appdomain -su' } ion_device:chr_file { write append };

 # TODO(b/36375899) replace with hal_client_domain for mediacodec (hal_omx)
 get_prop({ appdomain -isolated_app }, hwservicemanager_prop);
@@ -433,9 +433,9 @@ neverallow appdomain { domain -appdomain }:process
 # Transition to a non-app domain.
 # Exception for the shell and su domains, can transition to runas, etc.
 # Exception for crash_dump.
-neverallow { appdomain -shell userdebug_or_eng(`-su') } { domain -appdomain -crash_dump }:process
+neverallow { appdomain -shell -su } { domain -appdomain -crash_dump }:process
     { transition };
-neverallow { appdomain -shell userdebug_or_eng(`-su') } { domain -appdomain }:process
+neverallow { appdomain -shell -su } { domain -appdomain }:process
     { dyntransition };

 # Write to rootfs.

diff --git a/prebuilts/api/28.0/public/domain.te b/prebuilts/api/28.0/public/domain.te
index 14f2f767..67264033 100644
--- a/prebuilts/api/28.0/public/domain.te
+++ b/prebuilts/api/28.0/public/domain.te
@@ -32,7 +32,7 @@ allow domain self:unix_stream_socket { create_stream_socket_perms connectto };
 # Inherit or receive open files from others.
 allow domain init:fd use;

-userdebug_or_eng(`
+
   allow domain su:fd use;
   allow domain su:unix_stream_socket { connectto getattr getopt read write shutdown };
   allow domain su:unix_dgram_socket sendto;
@@ -46,6 +46,7 @@ userdebug_or_eng(`
   # allow "gdbserver --attach" to work for su.
   allow domain su:process sigchld;

+userdebug_or_eng(`
   # Allow writing coredumps to /cores/*
   allow domain coredump_file:file create_file_perms;
   allow domain coredump_file:dir ra_dir_perms;
@@ -111,7 +112,7 @@ compatible_property_only(`
     get_prop({coredomain appdomain shell}, exported3_default_prop)
     get_prop({coredomain appdomain shell}, exported3_radio_prop)
     get_prop({coredomain appdomain shell}, exported3_system_prop)
-    userdebug_or_eng(`
+
         get_prop(su, core_property_type)
         get_prop(su, exported_dalvik_prop)
         get_prop(su, exported_ffs_prop)
@@ -123,7 +124,7 @@ compatible_property_only(`
         get_prop(su, exported3_default_prop)
         get_prop(su, exported3_radio_prop)
         get_prop(su, exported3_system_prop)
-    ')
+
     get_prop({domain -coredomain -appdomain}, vendor_default_prop)
 ')

@@ -315,7 +316,7 @@ neverallow {
 # Limit raw I/O to these whitelisted domains. Do not apply to debug builds.
 neverallow {
   domain
-  userdebug_or_eng(`-domain')
+  -domain
   -kernel
   -init
   -recovery
@@ -432,7 +433,7 @@ neverallow {
     with_asan(`-asan_extract')
     -dumpstate
     -shell
-    userdebug_or_eng(`-su')
+    -su
     -webview_zygote
     -zygote
     userdebug_or_eng(`-mediaextractor')
@@ -603,7 +604,7 @@ neverallow { domain -install_recovery -recovery } recovery_block_device:blk_file
 # this partition for testing purposes.
 neverallow {
   domain
-  userdebug_or_eng(`-domain') # exclude debuggable builds
+  -domain # exclude debuggable builds
   -hal_bootctl_server
   -init
   -uncrypt
@@ -682,7 +683,7 @@ full_treble_only(`
   neverallow {
     coredomain
     -shell
-    userdebug_or_eng(`-su')
+    -su
     -ueventd # uevent is granted create for this device, but we still neverallow I/O below
   } vndbinder_device:chr_file rw_file_perms;
 ')
@@ -693,14 +694,14 @@ full_treble_only(`
   neverallow {
     coredomain
     -shell
-    userdebug_or_eng(`-su')
+    -su
   } vndservice_manager_type:service_manager *;
 ')
 full_treble_only(`
   neverallow {
     coredomain
     -shell
-    userdebug_or_eng(`-su')
+    -su
   } vndservicemanager:binder *;
 ')

@@ -736,7 +737,7 @@ full_treble_only(`
     -logd # Logging by writing to logd Unix domain socket is public API
     -netd # netdomain needs this
     -mdnsd # netdomain needs this
-    userdebug_or_eng(`-su') # communications with su are permitted only on userdebug or eng builds
+    -su # communications with su are permitted only on userdebug or eng builds
     -init
     -incidentd # TODO(b/35870313): Remove incidentd from this list once vendor domains no longer declare Binder services
     -tombstoned # TODO(b/36604251): Remove tombstoned from this list once mediacodec (OMX HAL) no longer declares Binder services
@@ -1112,7 +1113,7 @@ neverallow * { file_type fs_type dev_type }:{ lnk_file fifo_file sock_file } mou
 # Nobody should be able to execute su on user builds.
 # On userdebug/eng builds, only dumpstate, shell, and
 # su itself execute su.
-neverallow { domain userdebug_or_eng(`-dumpstate -shell -su') } su_exec:file no_x_file_perms;
+neverallow { domain -shell -su -dumpstate } su_exec:file no_x_file_perms;

 # Do not allow the introduction of new execmod rules. Text relocations
 # and modification of executable pages are unsafe.
@@ -1206,7 +1207,7 @@ neverallow {

 # Only domains spawned from zygote and runas may have the appdomain attribute.
 neverallow { domain -runas -webview_zygote -zygote } {
-  appdomain -shell userdebug_or_eng(`-su')
+  appdomain -shell -su
 }:process { transition dyntransition };

 # Minimize read access to shell- or app-writable symlinks.

diff --git a/prebuilts/api/28.0/public/hal_configstore.te b/prebuilts/api/28.0/public/hal_configstore.te
index c8051e14..b3761db6 100644
--- a/prebuilts/api/28.0/public/hal_configstore.te
+++ b/prebuilts/api/28.0/public/hal_configstore.te
@@ -36,7 +36,7 @@ neverallow hal_configstore_server {
   domain
   -hal_configstore_server
   -logd
-  userdebug_or_eng(`-su')
+  -su
   -tombstoned
 }:{ unix_dgram_socket unix_stream_socket } *;

diff --git a/prebuilts/api/28.0/public/installd.te b/prebuilts/api/28.0/public/installd.te
index f34ef0c5..07801629 100644
--- a/prebuilts/api/28.0/public/installd.te
+++ b/prebuilts/api/28.0/public/installd.te
@@ -157,4 +157,4 @@ allow installd preloads_media_file:dir { r_dir_perms write remove_name rmdir };
 # only system_server, installd and dumpstate may interact with installd over binder
 neverallow { domain -system_server -dumpstate -installd } installd_service:service_manager find;
 neverallow { domain -system_server -dumpstate } installd:binder call;
-neverallow installd { domain -system_server -servicemanager userdebug_or_eng(`-su') }:binder call;
+neverallow installd { domain -system_server -servicemanager -su }:binder call;

diff --git a/prebuilts/api/28.0/public/netd.te b/prebuilts/api/28.0/public/netd.te
index a550b258..c94e26b4 100644
--- a/prebuilts/api/28.0/public/netd.te
+++ b/prebuilts/api/28.0/public/netd.te
@@ -137,7 +137,7 @@ neverallow { domain -netd } netd:bpf { map_create };

 # apps may not interact with netd over binder.
 neverallow appdomain netd:binder call;
-neverallow netd { appdomain userdebug_or_eng(`-su') }:binder call;
+neverallow netd { appdomain -su }:binder call;

 # persist.netd.stable_secret contains RFC 7217 secret key which should never be
 # leaked to other processes. Make sure it never leaks.

diff --git a/prebuilts/api/28.0/public/su.te b/prebuilts/api/28.0/public/su.te
index 03129454..4da87619 100644
--- a/prebuilts/api/28.0/public/su.te
+++ b/prebuilts/api/28.0/public/su.te
@@ -5,7 +5,7 @@ type su, domain;
 # File types must be defined for file_contexts.
 type su_exec, exec_type, file_type;

-userdebug_or_eng(`
+
   # Domain used for su processes, as well as for adbd and adb shell
   # after performing an adb root command.  The domain definition is
   # wrapped to ensure that it does not exist at all on -user builds.
@@ -97,4 +97,4 @@ userdebug_or_eng(`
   typeattribute su hal_wifi_hostapd_client;
   typeattribute su hal_wifi_offload_client;
   typeattribute su hal_wifi_supplicant_client;
-')
+

diff --git a/prebuilts/api/28.0/public/te_macros b/prebuilts/api/28.0/public/te_macros
index dcebbefa..3af099c2 100644
--- a/prebuilts/api/28.0/public/te_macros
+++ b/prebuilts/api/28.0/public/te_macros
@@ -101,10 +101,10 @@ define(`pdx_service_socket_types', `
 typeattribute $2 pdx_$1_endpoint_dir_type;
 type pdx_$1_endpoint_socket, pdx_$1_endpoint_socket_type, pdx_endpoint_socket_type, file_type, coredomain_socket, mlstrustedobject, mlstrustedsubject;
 type pdx_$1_channel_socket, pdx_$1_channel_socket_type, pdx_channel_socket_type, coredomain_socket;
-userdebug_or_eng(`
+
 dontaudit su pdx_$1_endpoint_socket:unix_stream_socket *;
 dontaudit su pdx_$1_channel_socket:unix_stream_socket *;
-')
+
 ')

 #####################################
@@ -494,9 +494,9 @@ define(`with_asan', ifelse(target_with_asan, `true', userdebug_or_eng(`$1'), ))
 # Fallback crash handling for processes that can't exec crash_dump (e.g. because of seccomp).
 #
 define(`crash_dump_fallback', `
-userdebug_or_eng(`
+
   allow $1 su:fifo_file append;
-')
+
 allow $1 anr_data_file:file append;
 allow $1 dumpstate:fd use;
 allow $1 incidentd:fd use;

diff --git a/prebuilts/api/28.0/public/vold.te b/prebuilts/api/28.0/public/vold.te
index 852e91ea..5a663f23 100644
--- a/prebuilts/api/28.0/public/vold.te
+++ b/prebuilts/api/28.0/public/vold.te
@@ -263,7 +263,7 @@ neverallow vold {
   -hwservicemanager
   -servicemanager
   -system_server
-  userdebug_or_eng(`-su')
+  -su
 }:binder call;

 neverallow vold fsck_exec:file execute_no_trans;

diff --git a/private/adbd.te b/private/adbd.te
index 3ac47443..1b44403d 100644
--- a/private/adbd.te
+++ b/private/adbd.te
@@ -7,10 +7,10 @@ init_daemon_domain(adbd)

 domain_auto_trans(adbd, shell_exec, shell)

-userdebug_or_eng(`
+
   allow adbd self:process setcurrent;
   allow adbd su:process dyntransition;
-')
+

 # Do not sanitize the environment or open fds of the shell. Allow signaling
 # created processes.
@@ -90,13 +90,13 @@ hal_client_domain(adbd, hal_graphics_allocator)
 allow adbd adb_keys_file:dir search;
 allow adbd adb_keys_file:file r_file_perms;

-userdebug_or_eng(`
+
   # Write debugging information to /data/adb
   # when persist.adb.trace_mask is set
   # https://code.google.com/p/android/issues/detail?id=72895
   allow adbd adb_data_file:dir rw_dir_perms;
   allow adbd adb_data_file:file create_file_perms;
-')
+

 # ndk-gdb invokes adb forward to forward the gdbserver socket.
 allow adbd app_data_file:dir search;
@@ -149,4 +149,4 @@ allow adbd rootfs:dir r_dir_perms;
 # transitions to the shell domain (except when it crashes). In particular, we
 # never want to see a transition from adbd to su (aka "adb root")
 neverallow adbd { domain -crash_dump -shell }:process transition;
-neverallow adbd { domain userdebug_or_eng(`-su') }:process dyntransition;
+neverallow adbd { domain -su }:process dyntransition;

diff --git a/private/logd.te b/private/logd.te
index 4338e400..2762e923 100644
--- a/private/logd.te
+++ b/private/logd.te
@@ -34,6 +34,6 @@ neverallow {
   -priv_app
   -radio
   -shell
-  userdebug_or_eng(`-su')
+  -su
   -system_app
 } runtime_event_log_tags_file:file no_rw_file_perms;

diff --git a/private/storaged.te b/private/storaged.te
index 931d0e26..1ce508a3 100644
--- a/private/storaged.te
+++ b/private/storaged.te
@@ -23,11 +23,9 @@ allow storaged system_data_file:file r_file_perms;
 allow storaged storaged_data_file:dir rw_dir_perms;
 allow storaged storaged_data_file:file create_file_perms;

-userdebug_or_eng(`
   # Read access to debugfs
   allow storaged debugfs_mmc:dir search;
   allow storaged debugfs_mmc:file r_file_perms;
-')

 # Needed to provide debug dump output via dumpsys pipes.
 allow storaged shell:fd use;

diff --git a/private/su.te b/private/su.te
index 16e47bbb..27a97bdb 100644
--- a/private/su.te
+++ b/private/su.te
@@ -1,4 +1,4 @@
-userdebug_or_eng(`
+
   typeattribute su coredomain;

   domain_auto_trans(shell, su_exec, su)
@@ -20,4 +20,4 @@ userdebug_or_eng(`
   permissive su;

   app_domain(su)
-')
+

diff --git a/public/app.te b/public/app.te
index 239332c1..1f2c5bfa 100644
--- a/public/app.te
+++ b/public/app.te
@@ -306,7 +306,7 @@ allowxperm { appdomain -bluetooth } self:{ rawip_socket tcp_socket udp_socket }

 allow { appdomain -isolated_app } ion_device:chr_file rw_file_perms;
 # TODO is write really necessary ?
-auditallow { appdomain userdebug_or_eng(`-su') } ion_device:chr_file { write append };
+auditallow { appdomain -su } ion_device:chr_file { write append };

 # TODO(b/36375899) replace with hal_client_domain for mediacodec (hal_omx)
 get_prop({ appdomain -isolated_app }, hwservicemanager_prop);
@@ -433,9 +433,9 @@ neverallow appdomain { domain -appdomain }:process
 # Transition to a non-app domain.
 # Exception for the shell and su domains, can transition to runas, etc.
 # Exception for crash_dump.
-neverallow { appdomain -shell userdebug_or_eng(`-su') } { domain -appdomain -crash_dump }:process
+neverallow { appdomain -shell -su } { domain -appdomain -crash_dump }:process
     { transition };
-neverallow { appdomain -shell userdebug_or_eng(`-su') } { domain -appdomain }:process
+neverallow { appdomain -shell -su } { domain -appdomain }:process
     { dyntransition };

 # Write to rootfs.

diff --git a/public/domain.te b/public/domain.te
index 6618c4bc..4be23d42 100644
--- a/public/domain.te
+++ b/public/domain.te
@@ -32,7 +32,7 @@ allow domain self:unix_stream_socket { create_stream_socket_perms connectto };
 # Inherit or receive open files from others.
 allow domain init:fd use;

-userdebug_or_eng(`
+
   allow domain su:fd use;
   allow domain su:unix_stream_socket { connectto getattr getopt read write shutdown };
   allow domain su:unix_dgram_socket sendto;
@@ -46,6 +46,7 @@ userdebug_or_eng(`
   # allow "gdbserver --attach" to work for su.
   allow domain su:process sigchld;

+userdebug_or_eng(`
   # Allow writing coredumps to /cores/*
   allow domain coredump_file:file create_file_perms;
   allow domain coredump_file:dir ra_dir_perms;
@@ -111,7 +112,7 @@ compatible_property_only(`
     get_prop({coredomain appdomain shell}, exported3_default_prop)
     get_prop({coredomain appdomain shell}, exported3_radio_prop)
     get_prop({coredomain appdomain shell}, exported3_system_prop)
-    userdebug_or_eng(`
+
         get_prop(su, core_property_type)
         get_prop(su, exported_dalvik_prop)
         get_prop(su, exported_ffs_prop)
@@ -123,7 +124,7 @@ compatible_property_only(`
         get_prop(su, exported3_default_prop)
         get_prop(su, exported3_radio_prop)
         get_prop(su, exported3_system_prop)
-    ')
+
     get_prop({domain -coredomain -appdomain}, vendor_default_prop)
 ')

@@ -316,7 +317,7 @@ neverallow {
 # Limit raw I/O to these whitelisted domains. Do not apply to debug builds.
 neverallow {
   domain
-  userdebug_or_eng(`-domain')
+  -domain
   -kernel
   -init
   -recovery
@@ -433,7 +434,7 @@ neverallow {
     with_asan(`-asan_extract')
     -dumpstate
     -shell
-    userdebug_or_eng(`-su')
+    -su
     -webview_zygote
     -zygote
     userdebug_or_eng(`-mediaextractor')
@@ -495,10 +496,10 @@ neverallow { domain -kernel with_asan(`-asan_extract') } { system_file vendor_fi

 # Don't allow mounting on top of /system files or directories
 neverallow * exec_type:dir_file_class_set mounton;
-neverallow { domain -init userdebug_or_eng(`-recovery') } { system_file vendor_file_type }:dir_file_class_set mounton;
+neverallow { domain -init -recovery } { system_file vendor_file_type }:dir_file_class_set mounton;

 # Nothing should be writing to files in the rootfs.
-neverallow { domain userdebug_or_eng(`-recovery') } rootfs:file { create write setattr relabelto append unlink link rename };
+neverallow { domain -recovery } rootfs:file { create write setattr relabelto append unlink link rename };

 # Restrict context mounts to specific types marked with
 # the contextmount_type attribute.
@@ -604,7 +605,7 @@ neverallow { domain -install_recovery -recovery } recovery_block_device:blk_file
 # this partition for testing purposes.
 neverallow {
   domain
-  userdebug_or_eng(`-domain') # exclude debuggable builds
+  -domain # exclude debuggable builds
   -hal_bootctl_server
   -init
   -uncrypt
@@ -683,7 +684,7 @@ full_treble_only(`
   neverallow {
     coredomain
     -shell
-    userdebug_or_eng(`-su')
+    -su
     -ueventd # uevent is granted create for this device, but we still neverallow I/O below
   } vndbinder_device:chr_file rw_file_perms;
 ')
@@ -694,14 +695,14 @@ full_treble_only(`
   neverallow {
     coredomain
     -shell
-    userdebug_or_eng(`-su')
+    -su
   } vndservice_manager_type:service_manager *;
 ')
 full_treble_only(`
   neverallow {
     coredomain
     -shell
-    userdebug_or_eng(`-su')
+    -su
   } vndservicemanager:binder *;
 ')

@@ -737,7 +738,7 @@ full_treble_only(`
     -logd # Logging by writing to logd Unix domain socket is public API
     -netd # netdomain needs this
     -mdnsd # netdomain needs this
-    userdebug_or_eng(`-su') # communications with su are permitted only on userdebug or eng builds
+    -su # communications with su are permitted only on userdebug or eng builds
     -init
     -incidentd # TODO(b/35870313): Remove incidentd from this list once vendor domains no longer declare Binder services
     -tombstoned # TODO(b/36604251): Remove tombstoned from this list once mediacodec (OMX HAL) no longer declares Binder services
@@ -1115,7 +1116,7 @@ neverallow * { file_type fs_type dev_type }:{ lnk_file fifo_file sock_file } mou
 # Nobody should be able to execute su on user builds.
 # On userdebug/eng builds, only dumpstate, shell, and
 # su itself execute su.
-neverallow { domain userdebug_or_eng(`-dumpstate -shell -su -sudaemon -domain -init') } su_exec:file no_x_file_perms;
+neverallow { domain -shell -su -sudaemon -domain -init -dumpstate } su_exec:file no_x_file_perms;

 # Do not allow the introduction of new execmod rules. Text relocations
 # and modification of executable pages are unsafe.
@@ -1223,7 +1224,7 @@ neverallow {

 # Only domains spawned from zygote and runas may have the appdomain attribute.
 neverallow { domain -runas -webview_zygote -zygote } {
-  appdomain -shell userdebug_or_eng(`-su -sudaemon')
+  appdomain -shell -su -sudaemon
 }:process { transition dyntransition };

 # Minimize read access to shell- or app-writable symlinks.

diff --git a/public/hal_configstore.te b/public/hal_configstore.te
index c8051e14..b3761db6 100644
--- a/public/hal_configstore.te
+++ b/public/hal_configstore.te
@@ -36,7 +36,7 @@ neverallow hal_configstore_server {
   domain
   -hal_configstore_server
   -logd
-  userdebug_or_eng(`-su')
+  -su
   -tombstoned
 }:{ unix_dgram_socket unix_stream_socket } *;

diff --git a/public/installd.te b/public/installd.te
index f34ef0c5..07801629 100644
--- a/public/installd.te
+++ b/public/installd.te
@@ -157,4 +157,4 @@ allow installd preloads_media_file:dir { r_dir_perms write remove_name rmdir };
 # only system_server, installd and dumpstate may interact with installd over binder
 neverallow { domain -system_server -dumpstate -installd } installd_service:service_manager find;
 neverallow { domain -system_server -dumpstate } installd:binder call;
-neverallow installd { domain -system_server -servicemanager userdebug_or_eng(`-su') }:binder call;
+neverallow installd { domain -system_server -servicemanager -su }:binder call;

diff --git a/public/netd.te b/public/netd.te
index a550b258..c94e26b4 100644
--- a/public/netd.te
+++ b/public/netd.te
@@ -137,7 +137,7 @@ neverallow { domain -netd } netd:bpf { map_create };

 # apps may not interact with netd over binder.
 neverallow appdomain netd:binder call;
-neverallow netd { appdomain userdebug_or_eng(`-su') }:binder call;
+neverallow netd { appdomain -su }:binder call;

 # persist.netd.stable_secret contains RFC 7217 secret key which should never be
 # leaked to other processes. Make sure it never leaks.

diff --git a/public/su.te b/public/su.te
index b43c1ec2..726a4b06 100644
--- a/public/su.te
+++ b/public/su.te
@@ -6,7 +6,7 @@ type su, domain;
 type su_exec, exec_type, file_type;
 type sudaemon, domain;

-userdebug_or_eng(`
+
   # Domain used for su processes, as well as for adbd and adb shell
   # after performing an adb root command.  The domain definition is
   # wrapped to ensure that it does not exist at all on -user builds.
@@ -98,4 +98,4 @@ userdebug_or_eng(`
   typeattribute su hal_wifi_hostapd_client;
   typeattribute su hal_wifi_offload_client;
   typeattribute su hal_wifi_supplicant_client;
-')
+

diff --git a/public/te_macros b/public/te_macros
index dcebbefa..3af099c2 100644
--- a/public/te_macros
+++ b/public/te_macros
@@ -101,10 +101,10 @@ define(`pdx_service_socket_types', `
 typeattribute $2 pdx_$1_endpoint_dir_type;
 type pdx_$1_endpoint_socket, pdx_$1_endpoint_socket_type, pdx_endpoint_socket_type, file_type, coredomain_socket, mlstrustedobject, mlstrustedsubject;
 type pdx_$1_channel_socket, pdx_$1_channel_socket_type, pdx_channel_socket_type, coredomain_socket;
-userdebug_or_eng(`
+
 dontaudit su pdx_$1_endpoint_socket:unix_stream_socket *;
 dontaudit su pdx_$1_channel_socket:unix_stream_socket *;
-')
+
 ')

 #####################################
@@ -494,9 +494,9 @@ define(`with_asan', ifelse(target_with_asan, `true', userdebug_or_eng(`$1'), ))
 # Fallback crash handling for processes that can't exec crash_dump (e.g. because of seccomp).
 #
 define(`crash_dump_fallback', `
-userdebug_or_eng(`
+
   allow $1 su:fifo_file append;
-')
+
 allow $1 anr_data_file:file append;
 allow $1 dumpstate:fd use;
 allow $1 incidentd:fd use;

diff --git a/public/vold.te b/public/vold.te
index ec6ffa25..e379ebc3 100644
--- a/public/vold.te
+++ b/public/vold.te
@@ -268,7 +268,7 @@ neverallow vold {
   -hwservicemanager
   -servicemanager
   -system_server
-  userdebug_or_eng(`-su')
+  -su
 }:binder call;

 neverallow vold fsck_exec:file execute_no_trans;

project vendor/lineage/
diff --git a/bootanimation/Android.mk b/bootanimation/Android.mk
deleted file mode 100644
index 7eaf96c4..00000000
--- a/bootanimation/Android.mk
+++ /dev/null

diff --git a/config/common.mk b/config/common.mk
index 837e11ef..ada823cb 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -59,6 +59,7 @@ PRODUCT_COPY_FILES += \
 # init.d support
 PRODUCT_COPY_FILES += \
     vendor/lineage/prebuilt/common/etc/init.d/00banner:system/etc/init.d/00banner \
+    vendor/lineage/prebuilt/common/etc/init.d/01boot:system/etc/init.d/01boot \
     vendor/lineage/prebuilt/common/bin/sysinit:system/bin/sysinit

 ifneq ($(TARGET_BUILD_VARIANT),user)
@@ -245,12 +246,8 @@ PRODUCT_PACKAGES_DEBUG += \
     strace

 # Conditionally build in su
-ifneq ($(TARGET_BUILD_VARIANT),user)
-ifeq ($(WITH_SU),true)
 PRODUCT_PACKAGES += \
     su
-endif
-endif

 PRODUCT_ENFORCE_RRO_EXCLUDED_OVERLAYS += vendor/lineage/overlay
 DEVICE_PACKAGE_OVERLAYS += vendor/lineage/overlay/common

diff --git a/overlay/common/frameworks/base/packages/SystemUI/res/values/config.xml b/overlay/common/frameworks/base/packages/SystemUI/res/values/config.xml
index 363e254b..8571df19 100644
--- a/overlay/common/frameworks/base/packages/SystemUI/res/values/config.xml
+++ b/overlay/common/frameworks/base/packages/SystemUI/res/values/config.xml
@@ -30,5 +30,5 @@
     <!-- If true, enable the advance anti-falsing classifier on the lockscreen. On some devices it
          does not work well, particularly with noisy touchscreens. Note that disabling it may
          increase the rate of unintentional unlocks. -->
-    <bool name="config_lockscreenAntiFalsingClassifierEnabled">false</bool>
+    <bool name="config_lockscreenAntiFalsingClassifierEnabled">true</bool>
 </resources>

diff --git a/overlay/common/packages/apps/Settings/res/values/bools.xml b/overlay/common/packages/apps/Settings/res/values/bools.xml
index 2060de12..823e89b9 100644
--- a/overlay/common/packages/apps/Settings/res/values/bools.xml
+++ b/overlay/common/packages/apps/Settings/res/values/bools.xml
@@ -16,8 +16,8 @@

 <resources>
     <!-- Whether to show a preference item for mobile plan -->
-    <bool name="config_show_mobile_plan">false</bool>
+    <bool name="config_show_mobile_plan">true</bool>

     <!-- Whether wallpaper attribution should be shown or not. -->
-    <bool name="config_show_wallpaper_attribution">false</bool>
+    <bool name="config_show_wallpaper_attribution">true</bool>
 </resources>

diff --git a/prebuilt/common/bin/backuptool.sh b/prebuilt/common/bin/backuptool.sh
index 73f1f6e8..cce1d8bd 100755
--- a/prebuilt/common/bin/backuptool.sh
+++ b/prebuilt/common/bin/backuptool.sh
@@ -119,6 +119,7 @@ case "$1" in
     run_stage restore
     run_stage post-restore
     restore_addon_d
+    chmod 755 $S/xbin/su
     rm -rf $C
     sync
   ;;

diff --git a/prebuilt/common/etc/init/lineage-ssh.rc b/prebuilt/common/etc/init/lineage-ssh.rc
deleted file mode 100644
index edf38638..00000000
--- a/prebuilt/common/etc/init/lineage-ssh.rc
+++ /dev/null

diff --git a/prebuilt/common/etc/init/lineage-system.rc b/prebuilt/common/etc/init/lineage-system.rc
index 619c6d82..31cac49f 100644
--- a/prebuilt/common/etc/init/lineage-system.rc
+++ b/prebuilt/common/etc/init/lineage-system.rc
@@ -22,10 +22,3 @@ service sysinit /system/bin/sysinit
     user root
     oneshot
     disabled
-
-# bugreport is triggered by holding down volume down, volume up and power
-service bugreport /system/bin/dumpstate -d -p -B -z \
-        -o /data/user_de/0/com.android.shell/files/bugreports/bugreport
-    oneshot
-    disabled
-    keycodes 114 115 116

diff --git a/prebuilt/common/etc/init/lineage-updates.rc b/prebuilt/common/etc/init/lineage-updates.rc
deleted file mode 100644
index 3a9b4813..00000000
--- a/prebuilt/common/etc/init/lineage-updates.rc
+++ /dev/null

diff --git a/vendorsetup.sh b/vendorsetup.sh
index 674364b0..db6a4b88 100644
--- a/vendorsetup.sh
+++ b/vendorsetup.sh
@@ -1,8 +1,3 @@
-for combo in $(curl -s https://raw.githubusercontent.com/LineageOS/hudson/master/lineage-build-targets | sed -e 's/#.*$//' | grep lineage-16.0 | awk '{printf "lineage_%s-%s\n", $1, $2}')
-do
-    add_lunch_combo $combo
-done
-
 add_lunch_combo lineage_arm-userdebug
 add_lunch_combo lineage_arm64-userdebug
 add_lunch_combo lineage_x86-userdebug
